{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "961898db",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import xml.etree.ElementTree as ET\n",
    "import os\n",
    "from urllib.request import urlretrieve as download\n",
    "from glob import glob as g\n",
    "from Bio.Blast import NCBIWWW, NCBIXML\n",
    "from mypdb import PDB_file as mypdb\n",
    "from Bio.Blast.Applications import NcbipsiblastCommandline\n",
    "from time import time as t\n",
    "from tqdm import tqdm\n",
    "\n",
    "\n",
    "import glob\n",
    "import xml.etree.ElementTree as ET\n",
    "from collections import defaultdict\n",
    "\n",
    "import re\n",
    "import MDAnalysis as mda\n",
    "from time import time as t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ffd8d4e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Class to handle and download the hits from a blast search\n",
    "\n",
    "\n",
    "\n",
    "def merge_dicts(dict1, dict2):\n",
    "    for k2, v2 in dict2.items():\n",
    "        if k2 in dict1:\n",
    "            if isinstance(dict1[k2], list):\n",
    "                dict1[k2].append(v2)\n",
    "            else:\n",
    "                dict1[k2] = [dict1[k2], v2]\n",
    "        else:\n",
    "            print(f\"New Key : {k2}\")\n",
    "            dict1[k2] = v2\n",
    "    return dict1\n",
    "\n",
    "class hit:\n",
    "    def __init__(self, xml_hit):\n",
    "        self.hit_num = xml_hit[0].text\n",
    "        pdb_info = xml_hit[1].text.split(\"|\")[1:]\n",
    "        self.pdb_id = pdb_info[0]\n",
    "        self.pdb_chain = pdb_info[1]\n",
    "        self.description = xml_hit[2].text\n",
    "        data = [[y.tag.replace(\"Hsp_\", \"\"), y.text] for y in xml_hit[5][0]]\n",
    "        self.data = dict(data)\n",
    "        self.pdb = None\n",
    "        self._not_exists = False\n",
    "\n",
    "    @staticmethod\n",
    "    def download2(code, pdir=None):\n",
    "        base_url = \"https://files.rcsb.org/download\"\n",
    "        pdb_url = f\"{base_url}/{code}.pdb\"\n",
    "        f_p = os.path.join(pdir, f\"{code}.pdb\")\n",
    "        try:\n",
    "            download(pdb_url, f_p)\n",
    "            return f_p  # Return the file path if succeeded\n",
    "        except Exception:\n",
    "            print(f\"File {code} not found.\")\n",
    "            return None  # Return None if failed\n",
    "\n",
    "    def _assign(self, f_p):\n",
    "        self.pdb_path = f_p\n",
    "        self.pdb = mypdb(f_p)\n",
    "        \n",
    "    def check_for_pdb(self, pdir=None):\n",
    "        if pdir is None:\n",
    "            pdir = \"./pdbs/\"\n",
    "        elif pdir[-1] != \"/\":\n",
    "            pdir += \"/\"\n",
    "        if not os.path.isdir(pdir):\n",
    "            os.makedirs(pdir, exist_ok=True)\n",
    "        matches = g(pdir + f\"{self.pdb_id}*\")\n",
    "        if matches:\n",
    "            if not self.pdb:\n",
    "                self._assign(matches[0])\n",
    "            return True\n",
    "        else:\n",
    "            return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ab2139fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Class to handle the blast search\n",
    "class blast:\n",
    "    def __init__(self, sequence, name, database, path, program=\"blastp\", hitlen=10000):\n",
    "        self.sequence = sequence\n",
    "        self.program = program\n",
    "        self.database = database\n",
    "        self.path = path\n",
    "        self.name = name\n",
    "        self.hitlen = hitlen\n",
    "\n",
    "        if program == \"blastp\" and (self.path and os.path.isfile(self.path)):\n",
    "            self.parse_search()\n",
    "        elif program == \"psiblast\":\n",
    "            self.psiblast_search()\n",
    "        else:\n",
    "            self.bsearch()\n",
    "\n",
    "    def bsearch(self):\n",
    "        # Perform the initial BLASTP search\n",
    "        if self.program == \"blastp\":\n",
    "            print(\"Searching BLASTP...\")\n",
    "            t1 = t()\n",
    "            self.results = NCBIWWW.qblast(self.program, self.database, self.sequence, hitlist_size=self.hitlen)\n",
    "            t2 = t()\n",
    "            print(f\"BLASTP took {round(t2-t1,4)} seconds\")\n",
    "            if not self.path:\n",
    "                self.path = f\"{self.name}-blast.xml\"\n",
    "            with open(self.path, \"w\") as output_xml:\n",
    "                output_xml.write(self.results.read())\n",
    "            self.parse_search()\n",
    "\n",
    "    def psiblast_search(self):\n",
    "        # Run local PSI-BLAST using NcbipsiblastCommandline\n",
    "        print(f\"Running PSI-BLAST on {self.name}...\")\n",
    "        input_fasta = f\"{self.name}.fasta\"\n",
    "        with open(input_fasta, \"w\") as f:\n",
    "            f.write(f\">query\\n{self.sequence}\\n\")\n",
    "\n",
    "        psiblast_cline = NcbipsiblastCommandline(\n",
    "            query=input_fasta,\n",
    "            db=\"/home/marmatt/ncbi-blast-2.16.0+/bin/pdbaa\",\n",
    "            evalue=10,\n",
    "            num_iterations=3,\n",
    "            out_ascii_pssm=f\"{self.name}.pssm\",\n",
    "            out=f\"{self.name}-psiblast.xml\",\n",
    "            outfmt=5\n",
    "        )\n",
    "        stdout, stderr = psiblast_cline()\n",
    "        if stderr:\n",
    "            print(f\"PSI-BLAST ERROR: {stderr}\")\n",
    "        else:\n",
    "            print(\"PSI-BLAST search completed.\")\n",
    "            self.parse_search(xml_file=f\"{self.name}-psiblast.xml\")\n",
    "\n",
    "    def parse_search(self, xml_file=None):\n",
    "        xml_file = xml_file or self.path\n",
    "        if not xml_file:\n",
    "            raise Exception(\"No XML file path provided.\")\n",
    "        t1 = t()\n",
    "        tree = ET.parse(xml_file)\n",
    "        iteration = tree.findall(\"./BlastOutput_iterations/Iteration/\")\n",
    "        self.query_length = iteration[3].text\n",
    "        hits = [hit(x) for x in iteration[-2]] #returns hit objects\n",
    "        self.hits = hits\n",
    "        mega_dict = hits[0].data\n",
    "        for x in hits[1:]:\n",
    "            mega_dict = merge_dicts(mega_dict, x.data)\n",
    "        mega_dict[\"PDB ID\"] = [x.pdb_id for x in hits]\n",
    "        mega_dict[\"Chain\"] = [x.pdb_chain for x in hits]\n",
    "        mega_dict[\"Description\"] = [x.description for x in hits]\n",
    "        self.df = pd.DataFrame.from_dict(mega_dict)\n",
    "        print(self.df)\n",
    "        t2 = t()\n",
    "        print(f\"Time taken to parse {t2-t1}\")\n",
    "    \n",
    "    def download_pdbs(self, pdir=None):\n",
    "        default_dir = \"./PDBs\"\n",
    "        pdir = os.path.abspath(pdir if pdir else default_dir)\n",
    "        if not os.path.isdir(pdir):\n",
    "            os.makedirs(pdir, exist_ok=True)\n",
    "        \n",
    "        files = [os.path.splitext(f)[0] for f in os.listdir(pdir)]\n",
    "        hit_bar = tqdm(self.hits, desc=\"Processing Hits\")\n",
    "        \n",
    "        for x in hit_bar:\n",
    "            if x.pdb_id not in files:\n",
    "                hit_bar.set_description(f\"Downloading {x}\")\n",
    "                \n",
    "                try:\n",
    "                    file_path = x.download2(x.pdb_id, pdir=pdir)\n",
    "\n",
    "                    if file_path:\n",
    "                        x._assign(file_path)\n",
    "                    else:\n",
    "                        # In case download2 does not return a valid path\n",
    "                        raise Exception(\"Download failed\")\n",
    "                except Exception as e:\n",
    "                    # Print a message if download fails or file path is invalid\n",
    "                    print(f\"Structure {x.pdb_id} was not found...\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e81f2b43",
   "metadata": {},
   "source": [
    "## Actually running the blast search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "125b6d45",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Searching BLASTP...\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[5], line 22\u001b[0m\n\u001b[1;32m     20\u001b[0m     bs \u001b[38;5;241m=\u001b[39m blast(braf_fasta, name, database\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpdb\u001b[39m\u001b[38;5;124m\"\u001b[39m, path\u001b[38;5;241m=\u001b[39mres_path)\n\u001b[1;32m     21\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m---> 22\u001b[0m     bs \u001b[38;5;241m=\u001b[39m \u001b[43mblast\u001b[49m\u001b[43m(\u001b[49m\u001b[43mbraf_fasta\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mname\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdatabase\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mpdb\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpath\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[1;32m     24\u001b[0m bs\u001b[38;5;241m.\u001b[39mdownload_pdbs()\n",
      "Cell \u001b[0;32mIn[4], line 16\u001b[0m, in \u001b[0;36mblast.__init__\u001b[0;34m(self, sequence, name, database, path, program, hitlen)\u001b[0m\n\u001b[1;32m     14\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpsiblast_search()\n\u001b[1;32m     15\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m---> 16\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbsearch\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn[4], line 23\u001b[0m, in \u001b[0;36mblast.bsearch\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     21\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSearching BLASTP...\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     22\u001b[0m t1 \u001b[38;5;241m=\u001b[39m t()\n\u001b[0;32m---> 23\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mresults \u001b[38;5;241m=\u001b[39m \u001b[43mNCBIWWW\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mqblast\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mprogram\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdatabase\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msequence\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhitlist_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhitlen\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     24\u001b[0m t2 \u001b[38;5;241m=\u001b[39m t()\n\u001b[1;32m     25\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mBLASTP took \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mround\u001b[39m(t2\u001b[38;5;241m-\u001b[39mt1,\u001b[38;5;241m4\u001b[39m)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m seconds\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/Blast/NCBIWWW.py:268\u001b[0m, in \u001b[0;36mqblast\u001b[0;34m(program, database, sequence, url_base, auto_format, composition_based_statistics, db_genetic_code, endpoints, entrez_query, expect, filter, gapcosts, genetic_code, hitlist_size, i_thresh, layout, lcase_mask, matrix_name, nucl_penalty, nucl_reward, other_advanced, perc_ident, phi_pattern, query_file, query_believe_defline, query_from, query_to, searchsp_eff, service, threshold, ungapped_alignment, word_size, short_query, alignments, alignment_view, descriptions, entrez_links_new_window, expect_low, expect_high, format_entrez_query, format_object, format_type, ncbi_gi, results_file, show_overview, megablast, template_type, template_length, username, password)\u001b[0m\n\u001b[1;32m    266\u001b[0m wait \u001b[38;5;241m=\u001b[39m qblast\u001b[38;5;241m.\u001b[39m_previous \u001b[38;5;241m+\u001b[39m delay \u001b[38;5;241m-\u001b[39m current\n\u001b[1;32m    267\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m wait \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[0;32m--> 268\u001b[0m     \u001b[43mtime\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msleep\u001b[49m\u001b[43m(\u001b[49m\u001b[43mwait\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    269\u001b[0m     qblast\u001b[38;5;241m.\u001b[39m_previous \u001b[38;5;241m=\u001b[39m current \u001b[38;5;241m+\u001b[39m wait\n\u001b[1;32m    270\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "\n",
    "\"\"\"\n",
    "Here we perform a blast search on:\n",
    " 1.   The BRAF monomer\n",
    "\"\"\"\n",
    "with open(\"./fastas.txt\") as f:\n",
    "    fastas = f.readlines()\n",
    "\n",
    "braf_fasta = fastas[1]\n",
    "name = \"braf\"\n",
    "\n",
    "# Directory for storing blast search results\n",
    "output_dir = \"./blast_search\"\n",
    "# Ensure the directory exists\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "\n",
    "res_path = os.path.join(output_dir, f\"{name}-blast.xml\")  # Path for the results file\n",
    "\n",
    "# Check for saved results in the specified directory\n",
    "if os.path.exists(res_path):\n",
    "    bs = blast(braf_fasta, name, database=\"pdb\", path=res_path)\n",
    "else:\n",
    "    bs = blast(braf_fasta, name, database=\"pdb\", path=None)\n",
    "\n",
    "bs.download_pdbs()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b84c34d",
   "metadata": {},
   "source": [
    "## Counting the number of pdb files in the directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2c13cfea",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 2168 PDB files in the directory 'PDBs'.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "import glob\n",
    "def count_pdb_files(directory):\n",
    "    # Ensure the directory path ends with a slash\n",
    "    directory = os.path.join(directory, '')\n",
    "\n",
    "    # Use glob to find all .pdb files in the directory\n",
    "    pdb_files = glob.glob(os.path.join(directory, '*.pdb'))\n",
    "\n",
    "    # Return the count of .pdb files\n",
    "    return len(pdb_files)\n",
    "\n",
    "# Specify the directory\n",
    "pdb_directory = 'PDBs'\n",
    "\n",
    "# Get the count of PDB files\n",
    "pdb_count = count_pdb_files(pdb_directory)\n",
    "\n",
    "print(f\"There are {pdb_count} PDB files in the directory '{pdb_directory}'.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af58070b",
   "metadata": {},
   "source": [
    "## Counting the number of pdb hits in the xml file and checking which ones have not been downloaded\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c0522d85",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 2220 total PDB hits in the file 'braf-blast.xml'.\n",
      "There are 52 PDB IDs in the XML not found in the directory 'PDBs':\n",
      "{'9CD7', '8XFY', '9AXX', '8GXQ', '7Z37', '9BHI', '9D8F', '8ATL', '8XKP', '8VXF', '8Y22', '9AXA', '8PYI', '9AXH', '7OZY', '8C12', '9CMZ', '8YKI', '8XN7', '8W1L', '9ESA', '6Z83', '8VXE', '6Q38', '9C1R', '9F6X', '7Q6H', '8BR5', '8W2X', '9MHB', '8XFM', '9AXM', '7Z6I', '7PI4', '8XRR', '9D8Z', '8EJ4', '9BI8', '9EDY', '8V5I', '9AXC', '8VB5', '8VXD', '9AY7', '8ACM', '7P5Z', '8ATB', '9H46', '9H8C', '8V5H', '8YHF', '9BFB'}\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "def count_pdb_files(directory):\n",
    "    # Use glob to find all .pdb files in the directory\n",
    "    pdb_files = glob.glob(os.path.join(directory, '*.pdb'))\n",
    "\n",
    "    # Extract the base filenames (without extension) to compare with PDB IDs\n",
    "    pdb_file_ids = {os.path.splitext(os.path.basename(f))[0] for f in pdb_files}\n",
    "\n",
    "    return pdb_file_ids\n",
    "\n",
    "def find_unique_and_duplicate_pdb_hit_ids(xml_file):\n",
    "    # Parse the XML file\n",
    "    tree = ET.parse(xml_file)\n",
    "    root = tree.getroot()\n",
    "\n",
    "    # Dictionary to count occurrences of each PDB hit ID\n",
    "    hit_id_counts = defaultdict(int)\n",
    "\n",
    "    # Iterate over all Hit elements in the XML\n",
    "    for hit in root.findall('.//Hit'):\n",
    "        # Extract the Hit_id text\n",
    "        hit_id = hit.find('Hit_id').text\n",
    "\n",
    "        # Assuming the Hit_id format is 'pdb|PDB_ID|Chain', extract the PDB_ID\n",
    "        pdb_id = hit_id.split('|')[1]\n",
    "\n",
    "        # Increment the count for this PDB_ID\n",
    "        hit_id_counts[pdb_id] += 1\n",
    "\n",
    "    # Find all PDB IDs (unique and duplicates)\n",
    "    all_hit_ids = set(hit_id_counts.keys())\n",
    "\n",
    "    return all_hit_ids\n",
    "\n",
    "# Specify the directory and XML file\n",
    "pdb_directory = 'PDBs'\n",
    "xml_file = 'braf-blast.xml'\n",
    "\n",
    "# Get the PDB file IDs from the directory\n",
    "pdb_file_ids = count_pdb_files(pdb_directory)\n",
    "\n",
    "# Get all PDB hit IDs from the XML\n",
    "all_pdb_ids = find_unique_and_duplicate_pdb_hit_ids(xml_file)\n",
    "\n",
    "# Calculate the number of total PDB hits\n",
    "total_pdb_hits = len(all_pdb_ids)\n",
    "\n",
    "# Find PDB IDs in XML that are not in the directory\n",
    "missing_pdb_ids = all_pdb_ids - pdb_file_ids\n",
    "\n",
    "print(f\"There are {total_pdb_hits} total PDB hits in the file '{xml_file}'.\")\n",
    "print(f\"There are {len(missing_pdb_ids)} PDB IDs in the XML not found in the directory '{pdb_directory}':\")\n",
    "print(missing_pdb_ids)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "952b9660",
   "metadata": {},
   "source": [
    "## Here we are stripping the downloaded pdb files to only contain the chain of interest\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4fc269f2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6UAN_chainD.pdb     end_CA.pdb\t Results\t\t target.rsr\n",
      "blast_search\t    fastas.txt\t seqres_info.txt\t target.sch\n",
      "braf-blast.xml\t    mypdb.py\t seqres_sequences.fasta  target.V99990001\n",
      "BRAF-pipeline.png   PDBs\t start_CA.pdb\t\t template.pdb\n",
      "BRAFProtocol.ipynb  __pycache__  target.D00000001\n",
      "clean_pdb\t    README.md\t target.ini\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1A9U.pdb with 2950 atoms.\n",
      "File Results/activation_segments/unaligned/1A9U_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             10-APR-98   1A9U\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MAP KINASE P38;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN ACTIVATED PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 OTHER_DETAILS: SB203580 PYRIDINYLIMIDAZOLE\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     J.RAINGEAUD,S.GUPTA,J.S.ROGERS,M.DICKENS,J.HAN,R.J.ULEVITCH,\n",
      "REMARK     1  AUTH 2 R.J.DAVIS\n",
      "REMARK     1  TITL   PRO-INFLAMMATORY CYTOKINES AND ENVIRONMENTAL STRESS\n",
      "REMARK     CAUSE\n",
      "REMARK     1  TITL 2 P38 MITOGEN-ACTIVATED PROTEIN KINASE ACTIVATION BY DUAL\n",
      "REMARK     1  TITL 3 PHOSPHORYLATION ON TYROSINE AND THREONINE\n",
      "REMARK     1  REF    J.BIOL.CHEM.                  V. 270  7420 1995\n",
      "\n",
      "File Results/activation_segments/unaligned/1A9U_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             10-APR-98   1A9U\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MAP KINASE P38;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN ACTIVATED PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 OTHER_DETAILS: SB203580 PYRIDINYLIMIDAZOLE\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     J.RAINGEAUD,S.GUPTA,J.S.ROGERS,M.DICKENS,J.HAN,R.J.ULEVITCH,\n",
      "REMARK     1  AUTH 2 R.J.DAVIS\n",
      "REMARK     1  TITL   PRO-INFLAMMATORY CYTOKINES AND ENVIRONMENTAL STRESS\n",
      "REMARK     CAUSE\n",
      "REMARK     1  TITL 2 P38 MITOGEN-ACTIVATED PROTEIN KINASE ACTIVATION BY DUAL\n",
      "REMARK     1  TITL 3 PHOSPHORYLATION ON TYROSINE AND THREONINE\n",
      "REMARK     1  REF    J.BIOL.CHEM.                  V. 270  7420 1995\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1AD5.pdb A Results/activation_segments/unaligned/1AD5_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1AD5.pdb with 8572 atoms.\n",
      "File Results/activation_segments/unaligned/1AD5_A.pdb before post_process, first few lines:\n",
      "HEADER    TYROSINE-PROTEIN KINASE                 20-FEB-97   1AD5\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: HAEMATOPOETIC CELL KINASE HCK;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: SH3-SH2-KINASE-REGULATORY TAIL;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.8\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.60\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 18.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 3.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : 10000000.000\n",
      "\n",
      "File Results/activation_segments/unaligned/1AD5_A.pdb after post_process, first few lines:\n",
      "HEADER    TYROSINE-PROTEIN KINASE                 20-FEB-97   1AD5\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: HAEMATOPOETIC CELL KINASE HCK;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: SH3-SH2-KINASE-REGULATORY TAIL;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.8\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.60\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 18.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 3.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : 10000000.000\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1AGW.pdb A Results/activation_segments/unaligned/1AGW_A.pdb\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/marmatt/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/coordinates/PDB.py:1151: UserWarning: Found no information for attr: 'formalcharges' Using default value of '0'\n",
      "  warnings.warn(\"Found no information for attr: '{}'\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1AGW.pdb with 4607 atoms.\n",
      "File Results/activation_segments/unaligned/1AGW_A.pdb before post_process, first few lines:\n",
      "HEADER    PROTEIN KINASE                          25-MAR-97   1AGW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FGF RECEPTOR 1;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: FGFR1K;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   M.MOHAMMADI,J.SCHLESSINGER,S.R.HUBBARD\n",
      "REMARK     1  TITL   STRUCTURE OF THE FGF RECEPTOR TYROSINE KINASE DOMAIN\n",
      "REMARK     REVEALS\n",
      "REMARK     1  TITL 2 A NOVEL AUTOINHIBITORY MECHANISM\n",
      "REMARK     1  REF    CELL(CAMBRIDGE,MASS.)         V.  86   577 1996\n",
      "REMARK     1  REFN                   ISSN 0092-8674\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "\n",
      "File Results/activation_segments/unaligned/1AGW_A.pdb after post_process, first few lines:\n",
      "HEADER    PROTEIN KINASE                          25-MAR-97   1AGW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FGF RECEPTOR 1;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: FGFR1K;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   M.MOHAMMADI,J.SCHLESSINGER,S.R.HUBBARD\n",
      "REMARK     1  TITL   STRUCTURE OF THE FGF RECEPTOR TYROSINE KINASE DOMAIN\n",
      "REMARK     REVEALS\n",
      "REMARK     1  TITL 2 A NOVEL AUTOINHIBITORY MECHANISM\n",
      "REMARK     1  REF    CELL(CAMBRIDGE,MASS.)         V.  86   577 1996\n",
      "REMARK     1  REFN                   ISSN 0092-8674\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1APM.pdb E Results/activation_segments/unaligned/1APM_E.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1APM.pdb with 3075 atoms.\n",
      "File Results/activation_segments/unaligned/1APM_E.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE(PHOSPHOTRANSFERASE)         18-JAN-93   1APM\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 EC: 2.7.1.37;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: PEPTIDE INHIBITOR PKI(5-24);\n",
      "COMPND    8 CHAIN: I;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.ZHENG,D.R.KNIGHTON,L.F.TENEYCK,R.KARLSSON,N.-H.XUONG,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT OF CAMP-\n",
      "REMARK     DEPENDENT\n",
      "REMARK     1  TITL 2 PROTEIN KINASE COMPLEXED WITH MG/ATP AND PEPTIDE\n",
      "REMARK     INHIBITOR\n",
      "REMARK     1  REF    BIOCHEMISTRY                  V.  32  2154 1993\n",
      "\n",
      "File Results/activation_segments/unaligned/1APM_E.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE(PHOSPHOTRANSFERASE)         18-JAN-93   1APM\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 EC: 2.7.1.37;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: PEPTIDE INHIBITOR PKI(5-24);\n",
      "COMPND    8 CHAIN: I;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.ZHENG,D.R.KNIGHTON,L.F.TENEYCK,R.KARLSSON,N.-H.XUONG,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT OF CAMP-\n",
      "REMARK     DEPENDENT\n",
      "REMARK     1  TITL 2 PROTEIN KINASE COMPLEXED WITH MG/ATP AND PEPTIDE\n",
      "REMARK     INHIBITOR\n",
      "REMARK     1  REF    BIOCHEMISTRY                  V.  32  2154 1993\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1AQ1.pdb A Results/activation_segments/unaligned/1AQ1_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1AQ1.pdb with 2390 atoms.\n",
      "File Results/activation_segments/unaligned/1AQ1_A.pdb before post_process, first few lines:\n",
      "HEADER    PROTEIN KINASE                          05-AUG-97   1AQ1\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN-DEPENDENT PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: CDK2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: COMPLEX WITH CYCLIN A OR CYCLIN E\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   H.L.DE\n",
      "REMARK     BONDT,J.ROSENBLATT,J.JANCARIK,H.D.JONES,D.O.MORGAN,\n",
      "REMARK     1  AUTH 2 S.H.KIM\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF CYCLIN-DEPENDENT KINASE 2\n",
      "REMARK     1  REF    NATURE                        V. 363   595 1993\n",
      "REMARK     1  REFN                   ISSN 0028-0836\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1AQ1_A.pdb after post_process, first few lines:\n",
      "HEADER    PROTEIN KINASE                          05-AUG-97   1AQ1\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN-DEPENDENT PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: CDK2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: COMPLEX WITH CYCLIN A OR CYCLIN E\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   H.L.DE\n",
      "REMARK     BONDT,J.ROSENBLATT,J.JANCARIK,H.D.JONES,D.O.MORGAN,\n",
      "REMARK     1  AUTH 2 S.H.KIM\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF CYCLIN-DEPENDENT KINASE 2\n",
      "REMARK     1  REF    NATURE                        V. 363   595 1993\n",
      "REMARK     1  REFN                   ISSN 0028-0836\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1ATP.pdb E Results/activation_segments/unaligned/1ATP_E.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1ATP.pdb with 3070 atoms.\n",
      "File Results/activation_segments/unaligned/1ATP_E.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE(PHOSPHOTRANSFERASE)         08-JAN-93   1ATP\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 EC: 2.7.1.37;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: PEPTIDE INHIBITOR PKI(5-24);\n",
      "COMPND    8 CHAIN: I;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.R.KNIGHTON,S.M.BELL,J.ZHENG,L.F.TENEYCK,N.-H.XUONG,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   2.0 ANGSTROM REFINED CRYSTAL STRUCTURE OF THE CATALYTIC\n",
      "REMARK     1  TITL 2 SUBUNIT OF CAMP-DEPENDENT PROTEIN KINASE COMPLEXED WITH\n",
      "REMARK     A\n",
      "REMARK     1  TITL 3 PEPTIDE INHIBITOR AND DETERGENT\n",
      "REMARK     1  REF    TO BE PUBLISHED\n",
      "\n",
      "File Results/activation_segments/unaligned/1ATP_E.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE(PHOSPHOTRANSFERASE)         08-JAN-93   1ATP\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 EC: 2.7.1.37;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: PEPTIDE INHIBITOR PKI(5-24);\n",
      "COMPND    8 CHAIN: I;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.R.KNIGHTON,S.M.BELL,J.ZHENG,L.F.TENEYCK,N.-H.XUONG,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   2.0 ANGSTROM REFINED CRYSTAL STRUCTURE OF THE CATALYTIC\n",
      "REMARK     1  TITL 2 SUBUNIT OF CAMP-DEPENDENT PROTEIN KINASE COMPLEXED WITH\n",
      "REMARK     A\n",
      "REMARK     1  TITL 3 PEPTIDE INHIBITOR AND DETERGENT\n",
      "REMARK     1  REF    TO BE PUBLISHED\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1B38.pdb A Results/activation_segments/unaligned/1B38_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1B38.pdb with 2576 atoms.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/marmatt/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/topology/guessers.py:146: UserWarning: Failed to guess the mass for the following atom types: MN\n",
      "  warnings.warn(\"Failed to guess the mass for the following atom types: {}\".format(atom_type))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File Results/activation_segments/unaligned/1B38_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             17-DEC-98   1B38\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTEIN (CELL DIVISION PROTEIN KINASE 2);\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: INTACT;\n",
      "COMPND    5 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-\n",
      "REMARK     GAHMEN,J.BRANDSEN,H.D.JONES,D.O.MORGAN,L.MEIJER,\n",
      "REMARK     1  AUTH 2 J.VESELY,S.H.KIM\n",
      "REMARK     1  TITL   MULTIPLE MODES OF LIGAND RECOGNITION: CRYSTAL STRUCTURES\n",
      "REMARK     OF\n",
      "REMARK     1  TITL 2 CYCLIN-DEPENDENT PROTEIN KINASE 2 IN COMPLEX WITH ATP\n",
      "REMARK     AND\n",
      "REMARK     1  TITL 3 TWO INHIBITORS, OLOMOUCINE AND ISOPENTENYLADENINE\n",
      "REMARK     1  REF    PROTEINS                      V.  22   378 1995\n",
      "\n",
      "File Results/activation_segments/unaligned/1B38_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             17-DEC-98   1B38\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTEIN (CELL DIVISION PROTEIN KINASE 2);\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: INTACT;\n",
      "COMPND    5 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-\n",
      "REMARK     GAHMEN,J.BRANDSEN,H.D.JONES,D.O.MORGAN,L.MEIJER,\n",
      "REMARK     1  AUTH 2 J.VESELY,S.H.KIM\n",
      "REMARK     1  TITL   MULTIPLE MODES OF LIGAND RECOGNITION: CRYSTAL STRUCTURES\n",
      "REMARK     OF\n",
      "REMARK     1  TITL 2 CYCLIN-DEPENDENT PROTEIN KINASE 2 IN COMPLEX WITH ATP\n",
      "REMARK     AND\n",
      "REMARK     1  TITL 3 TWO INHIBITORS, OLOMOUCINE AND ISOPENTENYLADENINE\n",
      "REMARK     1  REF    PROTEINS                      V.  22   378 1995\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1B6C.pdb B Results/activation_segments/unaligned/1B6C_B.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1B6C.pdb with 13840 atoms.\n",
      "File Results/activation_segments/unaligned/1B6C_B.pdb before post_process, first few lines:\n",
      "HEADER    COMPLEX (ISOMERASE/PROTEIN KINASE)      13-JAN-99   1B6C\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FK506-BINDING PROTEIN;\n",
      "COMPND    3 CHAIN: A, C, E, G;\n",
      "COMPND    4 SYNONYM: FKBP12;\n",
      "COMPND    5 EC: 5.2.1.8;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: TGF-B SUPERFAMILY RECEPTOR TYPE I;\n",
      "COMPND    9 CHAIN: B, D, F, H;\n",
      "COMPND    10 FRAGMENT: CYTOPLASMIC PORTION;\n",
      "COMPND    11 SYNONYM: SERINE/THREONINE-PROTEIN KINASE RECEPTOR R4;\n",
      "COMPND    12 EC: 2.7.1.37;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 0.3C\n",
      "\n",
      "File Results/activation_segments/unaligned/1B6C_B.pdb after post_process, first few lines:\n",
      "HEADER    COMPLEX (ISOMERASE/PROTEIN KINASE)      13-JAN-99   1B6C\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FK506-BINDING PROTEIN;\n",
      "COMPND    3 CHAIN: A, C, E, G;\n",
      "COMPND    4 SYNONYM: FKBP12;\n",
      "COMPND    5 EC: 5.2.1.8;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: TGF-B SUPERFAMILY RECEPTOR TYPE I;\n",
      "COMPND    9 CHAIN: B, D, F, H;\n",
      "COMPND    10 FRAGMENT: CYTOPLASMIC PORTION;\n",
      "COMPND    11 SYNONYM: SERINE/THREONINE-PROTEIN KINASE RECEPTOR R4;\n",
      "COMPND    12 EC: 2.7.1.37;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 0.3C\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1BI7.pdb A Results/activation_segments/unaligned/1BI7_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1BI7.pdb with 3087 atoms.\n",
      "File Results/activation_segments/unaligned/1BI7_A.pdb before post_process, first few lines:\n",
      "HEADER    COMPLEX (KINASE/ANTI-ONCOGENE)          22-JUN-98   1BI7\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN-DEPENDENT KINASE 6;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: CDK6;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: MULTIPLE TUMOR SUPPRESSOR;\n",
      "COMPND    9 CHAIN: B;\n",
      "COMPND    10 SYNONYM: P16INK4A, MTS1;\n",
      "COMPND    11 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    3.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.8\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1BI7_A.pdb after post_process, first few lines:\n",
      "HEADER    COMPLEX (KINASE/ANTI-ONCOGENE)          22-JUN-98   1BI7\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN-DEPENDENT KINASE 6;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: CDK6;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: MULTIPLE TUMOR SUPPRESSOR;\n",
      "COMPND    9 CHAIN: B;\n",
      "COMPND    10 SYNONYM: P16INK4A, MTS1;\n",
      "COMPND    11 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    3.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.8\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1BKX.pdb A Results/activation_segments/unaligned/1BKX_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1BKX.pdb with 2832 atoms.\n",
      "File Results/activation_segments/unaligned/1BKX_A.pdb before post_process, first few lines:\n",
      "HEADER    COMPLEX (PHOSPHOTRANSFERASE/ADENOSINE)  01-JUL-97   1BKX\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: CAPK, PKA;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   N.NARAYANA,S.COX,S.SHALTIEL,S.S.TAYLOR,N.XUONG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF A POLYHISTIDINE-TAGGED RECOMBINANT\n",
      "REMARK     1  TITL 2 CATALYTIC SUBUNIT OF CAMP-DEPENDENT PROTEIN KINASE\n",
      "REMARK     COMPLEXED\n",
      "REMARK     1  TITL 3 WITH THE PEPTIDE INHIBITOR PKI(5-24) AND ADENOSINE\n",
      "REMARK     1  REF    BIOCHEMISTRY                  V.  36  4438 1997\n",
      "REMARK     1  REFN                   ISSN 0006-2960\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   J.ZHENG,D.R.KNIGHTON,L.F.TEN EYCK,R.KARLSSON,N.XUONG,\n",
      "\n",
      "File Results/activation_segments/unaligned/1BKX_A.pdb after post_process, first few lines:\n",
      "HEADER    COMPLEX (PHOSPHOTRANSFERASE/ADENOSINE)  01-JUL-97   1BKX\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: CAPK, PKA;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   N.NARAYANA,S.COX,S.SHALTIEL,S.S.TAYLOR,N.XUONG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF A POLYHISTIDINE-TAGGED RECOMBINANT\n",
      "REMARK     1  TITL 2 CATALYTIC SUBUNIT OF CAMP-DEPENDENT PROTEIN KINASE\n",
      "REMARK     COMPLEXED\n",
      "REMARK     1  TITL 3 WITH THE PEPTIDE INHIBITOR PKI(5-24) AND ADENOSINE\n",
      "REMARK     1  REF    BIOCHEMISTRY                  V.  36  4438 1997\n",
      "REMARK     1  REFN                   ISSN 0006-2960\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   J.ZHENG,D.R.KNIGHTON,L.F.TEN EYCK,R.KARLSSON,N.XUONG,\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1BMK.pdb A Results/activation_segments/unaligned/1BMK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1BMK.pdb with 2950 atoms.\n",
      "File Results/activation_segments/unaligned/1BMK_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             23-JUL-98   1BMK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTEIN (MAP KINASE P38);\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN ACTIVATED PROTEIN KINASE;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MUTATION: YES;\n",
      "COMPND    7 OTHER_DETAILS: SB218655 PYRIDINYLIMIDAZOLE\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     J.RAINGEAUD,S.GUPTA,J.S.ROGERS,M.DICKENS,J.HAN,R.J.ULEVITCH,\n",
      "REMARK     1  AUTH 2 R.J.DAVIS\n",
      "REMARK     1  TITL   PRO-INFLAMMATORY CYTOKINES AND ENVIRONMENTAL STRESS\n",
      "REMARK     CAUSE\n",
      "REMARK     1  TITL 2 P38 MITOGEN- ACTIVATED PROTEIN KINASE ACTIVATION BY DUAL\n",
      "REMARK     1  TITL 3 PHOSPHORYLATION ON TYROSINE AND THREONINE\n",
      "REMARK     1  REF    J.BIOL.CHEM.                  V. 270  7420 1995\n",
      "REMARK     1  REFN                   ISSN 0021-9258\n",
      "\n",
      "File Results/activation_segments/unaligned/1BMK_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             23-JUL-98   1BMK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTEIN (MAP KINASE P38);\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN ACTIVATED PROTEIN KINASE;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MUTATION: YES;\n",
      "COMPND    7 OTHER_DETAILS: SB218655 PYRIDINYLIMIDAZOLE\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     J.RAINGEAUD,S.GUPTA,J.S.ROGERS,M.DICKENS,J.HAN,R.J.ULEVITCH,\n",
      "REMARK     1  AUTH 2 R.J.DAVIS\n",
      "REMARK     1  TITL   PRO-INFLAMMATORY CYTOKINES AND ENVIRONMENTAL STRESS\n",
      "REMARK     CAUSE\n",
      "REMARK     1  TITL 2 P38 MITOGEN- ACTIVATED PROTEIN KINASE ACTIVATION BY DUAL\n",
      "REMARK     1  TITL 3 PHOSPHORYLATION ON TYROSINE AND THREONINE\n",
      "REMARK     1  REF    J.BIOL.CHEM.                  V. 270  7420 1995\n",
      "REMARK     1  REFN                   ISSN 0021-9258\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1BYG.pdb A Results/activation_segments/unaligned/1BYG_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1BYG.pdb with 2140 atoms.\n",
      "File Results/activation_segments/unaligned/1BYG_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             14-OCT-98   1BYG\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTEIN (C-TERMINAL SRC KINASE);\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: CSK;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 OTHER_DETAILS: THE C-TERMINAL HISTIDINE TAG WAS NOT OBSERVED IN THE\n",
      "COMPND    9 DENSITY AND IS NOT PRESENT IN THE SEQRES.\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1BYG_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             14-OCT-98   1BYG\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTEIN (C-TERMINAL SRC KINASE);\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: CSK;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 OTHER_DETAILS: THE C-TERMINAL HISTIDINE TAG WAS NOT OBSERVED IN THE\n",
      "COMPND    9 DENSITY AND IS NOT PRESENT IN THE SEQRES.\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CDK.pdb A Results/activation_segments/unaligned/1CDK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CDK.pdb with 6361 atoms.\n",
      "File Results/activation_segments/unaligned/1CDK_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       04-JUL-94   1CDK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PROTEIN KINASE A;\n",
      "COMPND    5 EC: 2.7.11.11;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: PROTEIN KINASE INHIBITOR;\n",
      "COMPND    8 CHAIN: I, J;\n",
      "COMPND    9 SYNONYM: PKI(5-24);\n",
      "COMPND    10 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.ZHENG,E.A.TRAFNY,D.R.KNIGHTON,N.-H.XUONG,S.S.TAYLOR,\n",
      "REMARK     1  AUTH 2 L.F.TEN EYCK,J.M.SOWADSKI\n",
      "REMARK     1  TITL   2.2 ANGSTROM REFINED CRYSTAL STRUCTURE OF THE CATALYTIC\n",
      "REMARK     1  TITL 2 SUBUNIT OF CAMP-DEPENDENT PROTEIN KINASE COMPLEXED WITH\n",
      "REMARK     1  TITL 3 MNATP AND A PEPTIDE INHIBITOR\n",
      "REMARK     1  REF    ACTA CRYSTALLOGR.,SECT.D      V.  49   362 1993\n",
      "\n",
      "File Results/activation_segments/unaligned/1CDK_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       04-JUL-94   1CDK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PROTEIN KINASE A;\n",
      "COMPND    5 EC: 2.7.11.11;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: PROTEIN KINASE INHIBITOR;\n",
      "COMPND    8 CHAIN: I, J;\n",
      "COMPND    9 SYNONYM: PKI(5-24);\n",
      "COMPND    10 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.ZHENG,E.A.TRAFNY,D.R.KNIGHTON,N.-H.XUONG,S.S.TAYLOR,\n",
      "REMARK     1  AUTH 2 L.F.TEN EYCK,J.M.SOWADSKI\n",
      "REMARK     1  TITL   2.2 ANGSTROM REFINED CRYSTAL STRUCTURE OF THE CATALYTIC\n",
      "REMARK     1  TITL 2 SUBUNIT OF CAMP-DEPENDENT PROTEIN KINASE COMPLEXED WITH\n",
      "REMARK     1  TITL 3 MNATP AND A PEPTIDE INHIBITOR\n",
      "REMARK     1  REF    ACTA CRYSTALLOGR.,SECT.D      V.  49   362 1993\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CKI.pdb A Results/activation_segments/unaligned/1CKI_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CKI.pdb with 4823 atoms.\n",
      "File Results/activation_segments/unaligned/1CKI_A.pdb before post_process, first few lines:\n",
      "HEADER    PHOSPHOTRANSFERASE                      25-AUG-95   1CKI\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE I DELTA;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 EC: 2.7.1.-;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   R.XU,G.CARMEL,R.M.SWEET,J.KURET,X.CHENG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF CASEIN KINASE-1, A PHOSPHATE-\n",
      "REMARK     DIRECTED\n",
      "REMARK     1  TITL 2 PROTEIN KINASE\n",
      "REMARK     1  REF    EMBO J.                       V.  14  1015 1995\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   P.R.GRAVES,D.W.HAAS,C.H.HAGEDORN,A.A.DEPAOLI-\n",
      "REMARK     ROACH,P.J.ROACH\n",
      "REMARK     1  TITL   MOLECULAR CLONING, EXPRESSION, AND CHARACTERIZATION OF A\n",
      "\n",
      "File Results/activation_segments/unaligned/1CKI_A.pdb after post_process, first few lines:\n",
      "HEADER    PHOSPHOTRANSFERASE                      25-AUG-95   1CKI\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE I DELTA;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 EC: 2.7.1.-;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   R.XU,G.CARMEL,R.M.SWEET,J.KURET,X.CHENG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF CASEIN KINASE-1, A PHOSPHATE-\n",
      "REMARK     DIRECTED\n",
      "REMARK     1  TITL 2 PROTEIN KINASE\n",
      "REMARK     1  REF    EMBO J.                       V.  14  1015 1995\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   P.R.GRAVES,D.W.HAAS,C.H.HAGEDORN,A.A.DEPAOLI-\n",
      "REMARK     ROACH,P.J.ROACH\n",
      "REMARK     1  TITL   MOLECULAR CLONING, EXPRESSION, AND CHARACTERIZATION OF A\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CM8.pdb A Results/activation_segments/unaligned/1CM8_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CM8.pdb with 5442 atoms.\n",
      "File Results/activation_segments/unaligned/1CM8_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             17-MAY-99   1CM8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PHOSPHORYLATED MAP KINASE P38-GAMMA;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: STRESS-ACTIVATED PROTEIN KINASE-3, ERK6, ERK5;\n",
      "COMPND    5 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 0.2\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.40\n",
      "\n",
      "File Results/activation_segments/unaligned/1CM8_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             17-MAY-99   1CM8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PHOSPHORYLATED MAP KINASE P38-GAMMA;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: STRESS-ACTIVATED PROTEIN KINASE-3, ERK6, ERK5;\n",
      "COMPND    5 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 0.2\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.40\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CMK.pdb E Results/activation_segments/unaligned/1CMK_E.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CMK.pdb with 3048 atoms.\n",
      "File Results/activation_segments/unaligned/1CMK_E.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       18-NOV-93   1CMK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 EC: 2.7.11.11;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    8 CHAIN: I;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   R.KARLSSON,J.ZHENG,N.-H.XUONG,S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   THE CRYSTAL STRUCTURE OF THE MAMMALIAN CATALYTIC SUBUNIT\n",
      "REMARK     OF\n",
      "REMARK     1  TITL 2 CAMP-DEPENDENT PROTEIN KINASE AND A DI-IODINATED\n",
      "REMARK     PKI(5-24)\n",
      "REMARK     1  TITL 3 INHIBITOR PEPTIDE DISPLAYS AN OPEN CONFORMATION\n",
      "REMARK     1  REF    ACTA CRYSTALLOGR.,SECT.D      V.  49   381 1993\n",
      "\n",
      "File Results/activation_segments/unaligned/1CMK_E.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       18-NOV-93   1CMK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 EC: 2.7.11.11;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    8 CHAIN: I;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   R.KARLSSON,J.ZHENG,N.-H.XUONG,S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   THE CRYSTAL STRUCTURE OF THE MAMMALIAN CATALYTIC SUBUNIT\n",
      "REMARK     OF\n",
      "REMARK     1  TITL 2 CAMP-DEPENDENT PROTEIN KINASE AND A DI-IODINATED\n",
      "REMARK     PKI(5-24)\n",
      "REMARK     1  TITL 3 INHIBITOR PEPTIDE DISPLAYS AN OPEN CONFORMATION\n",
      "REMARK     1  REF    ACTA CRYSTALLOGR.,SECT.D      V.  49   381 1993\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CSN.pdb A Results/activation_segments/unaligned/1CSN_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CSN.pdb with 2531 atoms.\n",
      "File Results/activation_segments/unaligned/1CSN_A.pdb before post_process, first few lines:\n",
      "HEADER    PHOSPHOTRANSFERASE                      25-APR-95   1CSN\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE-1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 EC: 2.7.1.-\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   G.CARMEL,B.LEICHUS,X.CHENG\n",
      "REMARK     1  TITL   EXPRESSION, PURIFICATION, CRYSTALLIZATION, AND\n",
      "REMARK     PRELIMINARY\n",
      "REMARK     1  TITL 2 X-RAY ANALYSIS OF CASEIN KINASE-1 FROM\n",
      "REMARK     SCHIZOSACCHAROMYCES\n",
      "REMARK     1  TITL 3 POMBE\n",
      "REMARK     1  REF    J.BIOL.CHEM.                  V. 269  7304 1994\n",
      "REMARK     1  REFN                   ISSN 0021-9258\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1CSN_A.pdb after post_process, first few lines:\n",
      "HEADER    PHOSPHOTRANSFERASE                      25-APR-95   1CSN\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE-1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 EC: 2.7.1.-\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   G.CARMEL,B.LEICHUS,X.CHENG\n",
      "REMARK     1  TITL   EXPRESSION, PURIFICATION, CRYSTALLIZATION, AND\n",
      "REMARK     PRELIMINARY\n",
      "REMARK     1  TITL 2 X-RAY ANALYSIS OF CASEIN KINASE-1 FROM\n",
      "REMARK     SCHIZOSACCHAROMYCES\n",
      "REMARK     1  TITL 3 POMBE\n",
      "REMARK     1  REF    J.BIOL.CHEM.                  V. 269  7304 1994\n",
      "REMARK     1  REFN                   ISSN 0021-9258\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CTP.pdb E Results/activation_segments/unaligned/1CTP_E.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1CTP.pdb with 2732 atoms.\n",
      "File Results/activation_segments/unaligned/1CTP_E.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       08-APR-93   1CTP\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 EC: 2.7.1.37;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    8 CHAIN: I;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.ZHENG,D.R.KNIGHTON,N.-H.XUONG,L.F.T.EYCK,R.KARLSSON,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   CRYSTAL-STRUCTURE OF THE CATALYTIC SUBUNIT OF CAMP-\n",
      "REMARK     DEPENDENT\n",
      "REMARK     1  TITL 2 PROTEIN-KINASE COMPLEXED WITH A MGATP AND PEPTIDE\n",
      "REMARK     INHIBITOR\n",
      "REMARK     1  REF    BIOCHEMISTRY                  V.  32  2154 1993\n",
      "\n",
      "File Results/activation_segments/unaligned/1CTP_E.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       08-APR-93   1CTP\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 EC: 2.7.1.37;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    8 CHAIN: I;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.ZHENG,D.R.KNIGHTON,N.-H.XUONG,L.F.T.EYCK,R.KARLSSON,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   CRYSTAL-STRUCTURE OF THE CATALYTIC SUBUNIT OF CAMP-\n",
      "REMARK     DEPENDENT\n",
      "REMARK     1  TITL 2 PROTEIN-KINASE COMPLEXED WITH A MGATP AND PEPTIDE\n",
      "REMARK     INHIBITOR\n",
      "REMARK     1  REF    BIOCHEMISTRY                  V.  32  2154 1993\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1DAW.pdb A Results/activation_segments/unaligned/1DAW_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1DAW.pdb with 3051 atoms.\n",
      "File Results/activation_segments/unaligned/1DAW_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             01-NOV-99   1DAW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTEIN KINASE CK2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,L.A.PINNA,O.G.ISSINGER,D.SCHOMBURG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT OF PROTEIN\n",
      "REMARK     KINASE\n",
      "REMARK     1  TITL 2 CK2 FROM ZEA MAYS AT 2.1 A RESOLUTION\n",
      "REMARK     1  REF    EMBO J.                       V.  17  2451 1998\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "REMARK     1  DOI    10.1093/EMBOJ/17.9.2451\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   B.GUERRA,K.NIEFIND,L.A.PINNA,D.SCHOMBURG,O.G.ISSINGER\n",
      "REMARK     1  TITL   EXPRESSION, PURIFICATION AND CRYSTALLIZATION OF THE\n",
      "\n",
      "File Results/activation_segments/unaligned/1DAW_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             01-NOV-99   1DAW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTEIN KINASE CK2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,L.A.PINNA,O.G.ISSINGER,D.SCHOMBURG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT OF PROTEIN\n",
      "REMARK     KINASE\n",
      "REMARK     1  TITL 2 CK2 FROM ZEA MAYS AT 2.1 A RESOLUTION\n",
      "REMARK     1  REF    EMBO J.                       V.  17  2451 1998\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "REMARK     1  DOI    10.1093/EMBOJ/17.9.2451\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   B.GUERRA,K.NIEFIND,L.A.PINNA,D.SCHOMBURG,O.G.ISSINGER\n",
      "REMARK     1  TITL   EXPRESSION, PURIFICATION AND CRYSTALLIZATION OF THE\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1DI9.pdb A Results/activation_segments/unaligned/1DI9_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1DI9.pdb with 2859 atoms.\n",
      "File Results/activation_segments/unaligned/1DI9_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             29-NOV-99   1DI9\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: P38 KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 EC: 2.7.1.-;\n",
      "COMPND    5 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.60\n",
      "\n",
      "File Results/activation_segments/unaligned/1DI9_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             29-NOV-99   1DI9\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: P38 KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 EC: 2.7.1.-;\n",
      "COMPND    5 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.60\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1DS5.pdb A Results/activation_segments/unaligned/1DS5_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1DS5.pdb with 11932 atoms.\n",
      "File Results/activation_segments/unaligned/1DS5_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             07-JAN-00   1DS5\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A, B, C, D;\n",
      "COMPND    4 SYNONYM: CK2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CASEIN KINASE, BETA CHAIN;\n",
      "COMPND    9 CHAIN: E, F, G, H;\n",
      "COMPND    10 FRAGMENT: RESIDUES 181-203;\n",
      "COMPND    11 SYNONYM: CK2;\n",
      "COMPND    12 EC: 2.7.1.37;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,L.A.PINNA,O.G.ISSINGER,D.SCHOMBURG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT OF PROTEIN\n",
      "REMARK     KINASE\n",
      "\n",
      "File Results/activation_segments/unaligned/1DS5_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             07-JAN-00   1DS5\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A, B, C, D;\n",
      "COMPND    4 SYNONYM: CK2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CASEIN KINASE, BETA CHAIN;\n",
      "COMPND    9 CHAIN: E, F, G, H;\n",
      "COMPND    10 FRAGMENT: RESIDUES 181-203;\n",
      "COMPND    11 SYNONYM: CK2;\n",
      "COMPND    12 EC: 2.7.1.37;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,L.A.PINNA,O.G.ISSINGER,D.SCHOMBURG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT OF PROTEIN\n",
      "REMARK     KINASE\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1E9H.pdb A Results/activation_segments/unaligned/1E9H_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1E9H.pdb with 9355 atoms.\n",
      "File Results/activation_segments/unaligned/1E9H_A.pdb before post_process, first few lines:\n",
      "HEADER    CELL CYCLE                              16-OCT-00   1E9H\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, C;\n",
      "COMPND    4 SYNONYM: CYCLIN-DEPENDENT KINASE 2, CDK2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: PHOSPHORYLATED ON THR 160 CHAINS A AND C ARE ASYMETRIC\n",
      "COMPND    8 UNIT COPIES;\n",
      "COMPND    9 MOL_ID: 2;\n",
      "COMPND    10 MOLECULE: CYCLIN A3;\n",
      "COMPND    11 CHAIN: B, D;\n",
      "COMPND    12 SYNONYM: A3;\n",
      "COMPND    13 ENGINEERED: YES;\n",
      "COMPND    14 OTHER_DETAILS: TRUNCATED FRAGMENT OF CYCLIN A. IN COMPLEX WITH CDK2\n",
      "COMPND    15 CHAINS B AND D ARE ASYMETRIC UNIT COPIES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1E9H_A.pdb after post_process, first few lines:\n",
      "HEADER    CELL CYCLE                              16-OCT-00   1E9H\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, C;\n",
      "COMPND    4 SYNONYM: CYCLIN-DEPENDENT KINASE 2, CDK2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: PHOSPHORYLATED ON THR 160 CHAINS A AND C ARE ASYMETRIC\n",
      "COMPND    8 UNIT COPIES;\n",
      "COMPND    9 MOL_ID: 2;\n",
      "COMPND    10 MOLECULE: CYCLIN A3;\n",
      "COMPND    11 CHAIN: B, D;\n",
      "COMPND    12 SYNONYM: A3;\n",
      "COMPND    13 ENGINEERED: YES;\n",
      "COMPND    14 OTHER_DETAILS: TRUNCATED FRAGMENT OF CYCLIN A. IN COMPLEX WITH CDK2\n",
      "COMPND    15 CHAINS B AND D ARE ASYMETRIC UNIT COPIES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FMK.pdb A Results/activation_segments/unaligned/1FMK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FMK.pdb with 4006 atoms.\n",
      "File Results/activation_segments/unaligned/1FMK_A.pdb before post_process, first few lines:\n",
      "HEADER    PHOSPHOTRANSFERASE                      24-JAN-97   1FMK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE SRC;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: RESIDUES 86-836, CONTAINING SH2, SH3, KINASE 2 DOMAINS AND\n",
      "COMPND    5 C-TERMINAL TAIL;\n",
      "COMPND    6 SYNONYM: C-SRC, P60-SRC;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.50\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 20.00\n",
      "\n",
      "File Results/activation_segments/unaligned/1FMK_A.pdb after post_process, first few lines:\n",
      "HEADER    PHOSPHOTRANSFERASE                      24-JAN-97   1FMK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE SRC;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: RESIDUES 86-836, CONTAINING SH2, SH3, KINASE 2 DOMAINS AND\n",
      "COMPND    5 C-TERMINAL TAIL;\n",
      "COMPND    6 SYNONYM: C-SRC, P60-SRC;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.50\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 20.00\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FOT.pdb A Results/activation_segments/unaligned/1FOT_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FOT.pdb with 2497 atoms.\n",
      "File Results/activation_segments/unaligned/1FOT_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             28-AUG-00   1FOT\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE TYPE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: N-TERMINUS TRUNCATED TPK1;\n",
      "COMPND    5 SYNONYM: TPK1 DELTA;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.KURET,J.W.PFLUGRATH\n",
      "REMARK     1  TITL   CRYSTALLIZATION AND PRELIMINARY X-RAY ANALYSIS OF THE\n",
      "REMARK     1  TITL 2 CAMP-DEPENDE PROTEIN KINASE CATALYTIC SUBUNIT FROM\n",
      "REMARK     1  TITL 3 SACCHAROMYCES CEREVISIAE\n",
      "REMARK     1  REF    BIOCHEMISTRY                  V.  30 10595 1991\n",
      "REMARK     1  REFN                   ISSN 0006-2960\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1FOT_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             28-AUG-00   1FOT\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE TYPE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: N-TERMINUS TRUNCATED TPK1;\n",
      "COMPND    5 SYNONYM: TPK1 DELTA;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.KURET,J.W.PFLUGRATH\n",
      "REMARK     1  TITL   CRYSTALLIZATION AND PRELIMINARY X-RAY ANALYSIS OF THE\n",
      "REMARK     1  TITL 2 CAMP-DEPENDE PROTEIN KINASE CATALYTIC SUBUNIT FROM\n",
      "REMARK     1  TITL 3 SACCHAROMYCES CEREVISIAE\n",
      "REMARK     1  REF    BIOCHEMISTRY                  V.  30 10595 1991\n",
      "REMARK     1  REFN                   ISSN 0006-2960\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FPU.pdb A Results/activation_segments/unaligned/1FPU_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FPU.pdb with 4451 atoms.\n",
      "File Results/activation_segments/unaligned/1FPU_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             31-AUG-00   1FPU\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: P150, C-ABL;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1FPU_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             31-AUG-00   1FPU\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: P150, C-ABL;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FQ1.pdb B Results/activation_segments/unaligned/1FQ1_B.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FQ1.pdb with 3853 atoms.\n",
      "File Results/activation_segments/unaligned/1FQ1_B.pdb before post_process, first few lines:\n",
      "HEADER    HYDROLASE/TRANSFERASE                   01-SEP-00   1FQ1\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN-DEPENDENT KINASE INHIBITOR 3;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 EC: 3.1.3.48;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MUTATION: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    9 CHAIN: B;\n",
      "COMPND    10 EC: 2.7.11.22;\n",
      "COMPND    11 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   R.Y.C.POON,T.HUNTER\n",
      "REMARK     1  TITL   DEPHOSPHORYLATION OF CDK2 THR160 BY THE CYCLIN DEPENDENT\n",
      "REMARK     1  TITL 2 KINASE-INTERACTING PHOSPHATASE KAP IN THE ABSENCE OF\n",
      "REMARK     CYCLIN\n",
      "REMARK     1  REF    SCIENCE                       V. 270    90 1995\n",
      "\n",
      "File Results/activation_segments/unaligned/1FQ1_B.pdb after post_process, first few lines:\n",
      "HEADER    HYDROLASE/TRANSFERASE                   01-SEP-00   1FQ1\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN-DEPENDENT KINASE INHIBITOR 3;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 EC: 3.1.3.48;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MUTATION: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    9 CHAIN: B;\n",
      "COMPND    10 EC: 2.7.11.22;\n",
      "COMPND    11 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   R.Y.C.POON,T.HUNTER\n",
      "REMARK     1  TITL   DEPHOSPHORYLATION OF CDK2 THR160 BY THE CYCLIN DEPENDENT\n",
      "REMARK     1  TITL 2 KINASE-INTERACTING PHOSPHATASE KAP IN THE ABSENCE OF\n",
      "REMARK     CYCLIN\n",
      "REMARK     1  REF    SCIENCE                       V. 270    90 1995\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FVR.pdb A Results/activation_segments/unaligned/1FVR_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1FVR.pdb with 5201 atoms.\n",
      "File Results/activation_segments/unaligned/1FVR_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             20-SEP-00   1FVR\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE TIE-2;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1FVR_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             20-SEP-00   1FVR\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE TIE-2;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GAG.pdb A Results/activation_segments/unaligned/1GAG_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GAG.pdb with 2520 atoms.\n",
      "File Results/activation_segments/unaligned/1GAG_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       29-NOV-00   1GAG\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR, TYROSINE KINASE DOMAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: IR;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES;\n",
      "COMPND    9 MOL_ID: 2;\n",
      "COMPND    10 MOLECULE: BISUBSTRATE PEPTIDE INHIBITOR;\n",
      "COMPND    11 CHAIN: B;\n",
      "COMPND    12 ENGINEERED: YES;\n",
      "COMPND    13 OTHER_DETAILS: PHE107 OF PEPTIDE 13-MER MODIFIED BY ATP GAMMA S PLUS\n",
      "COMPND    14 FOUR-ATOM LINKER, RESIDUE 112\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   S.R.HUBBARD\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE ACTIVATED INSULIN RECEPTOR\n",
      "\n",
      "File Results/activation_segments/unaligned/1GAG_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       29-NOV-00   1GAG\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR, TYROSINE KINASE DOMAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: IR;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES;\n",
      "COMPND    9 MOL_ID: 2;\n",
      "COMPND    10 MOLECULE: BISUBSTRATE PEPTIDE INHIBITOR;\n",
      "COMPND    11 CHAIN: B;\n",
      "COMPND    12 ENGINEERED: YES;\n",
      "COMPND    13 OTHER_DETAILS: PHE107 OF PEPTIDE 13-MER MODIFIED BY ATP GAMMA S PLUS\n",
      "COMPND    14 FOUR-ATOM LINKER, RESIDUE 112\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   S.R.HUBBARD\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE ACTIVATED INSULIN RECEPTOR\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GII.pdb A Results/activation_segments/unaligned/1GII_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GII.pdb with 2340 atoms.\n",
      "File Results/activation_segments/unaligned/1GII_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-FEB-01   1GII\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: CYCLIN DEPENDENT KINASE 2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 98.0\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.00\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 8.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 2.000\n",
      "\n",
      "File Results/activation_segments/unaligned/1GII_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-FEB-01   1GII\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: CYCLIN DEPENDENT KINASE 2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 98.0\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.00\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 8.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 2.000\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GJO.pdb A Results/activation_segments/unaligned/1GJO_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GJO.pdb with 2377 atoms.\n",
      "File Results/activation_segments/unaligned/1GJO_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             31-JUL-01   1GJO\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FIBROBLAST GROWTH FACTOR RECEPTOR 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN RESIDUES 465-768;\n",
      "COMPND    5 SYNONYM: FGFR-2, KERATINOCYTE;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 98\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.40\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 40.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 0.000\n",
      "\n",
      "File Results/activation_segments/unaligned/1GJO_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             31-JUL-01   1GJO\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FIBROBLAST GROWTH FACTOR RECEPTOR 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN RESIDUES 465-768;\n",
      "COMPND    5 SYNONYM: FGFR-2, KERATINOCYTE;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 98\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.40\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 40.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 0.000\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GOL.pdb A Results/activation_segments/unaligned/1GOL_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GOL.pdb with 3012 atoms.\n",
      "File Results/activation_segments/unaligned/1GOL_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             23-JAN-96   1GOL\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: EXTRACELLULAR REGULATED KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MAP KINASE ERK2;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.ZHANG,F.ZHANG,D.EBERT,M.H.COBB,E.J.GOLDSMITH\n",
      "REMARK     1  TITL   ACTIVITY OF THE MAP KINASE ERK2 IS CONTROLLED BY A\n",
      "REMARK     FLEXIBLE\n",
      "REMARK     1  TITL 2 SURFACE LOOP\n",
      "REMARK     1  REF    STRUCTURE                     V.   3   299 1995\n",
      "REMARK     1  REFN                   ISSN 0969-2126\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   F.ZHANG,A.STRAND,D.ROBBINS,M.H.COBB,E.J.GOLDSMITH\n",
      "REMARK     1  TITL   ATOMIC STRUCTURE OF THE MAP KINASE ERK2 AT 2.3 A\n",
      "\n",
      "File Results/activation_segments/unaligned/1GOL_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             23-JAN-96   1GOL\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: EXTRACELLULAR REGULATED KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MAP KINASE ERK2;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   J.ZHANG,F.ZHANG,D.EBERT,M.H.COBB,E.J.GOLDSMITH\n",
      "REMARK     1  TITL   ACTIVITY OF THE MAP KINASE ERK2 IS CONTROLLED BY A\n",
      "REMARK     FLEXIBLE\n",
      "REMARK     1  TITL 2 SURFACE LOOP\n",
      "REMARK     1  REF    STRUCTURE                     V.   3   299 1995\n",
      "REMARK     1  REFN                   ISSN 0969-2126\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   F.ZHANG,A.STRAND,D.ROBBINS,M.H.COBB,E.J.GOLDSMITH\n",
      "REMARK     1  TITL   ATOMIC STRUCTURE OF THE MAP KINASE ERK2 AT 2.3 A\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GY3.pdb A Results/activation_segments/unaligned/1GY3_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GY3.pdb with 9284 atoms.\n",
      "File Results/activation_segments/unaligned/1GY3_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE SUBSTRATE       19-APR-02   1GY3\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, C;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 OTHER_DETAILS: PHOSPHORYLATED ON THR160;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CYCLIN A2;\n",
      "COMPND    9 CHAIN: B, D;\n",
      "COMPND    10 FRAGMENT: RESIDUES 175-432;\n",
      "COMPND    11 ENGINEERED: YES;\n",
      "COMPND    12 MOL_ID: 3;\n",
      "COMPND    13 MOLECULE: SUBSTRATE PEPTIDE;\n",
      "COMPND    14 CHAIN: E, F;\n",
      "COMPND    15 ENGINEERED: YES;\n",
      "COMPND    16 OTHER_DETAILS: SEQUENCE HHASPRK\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "\n",
      "File Results/activation_segments/unaligned/1GY3_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE SUBSTRATE       19-APR-02   1GY3\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, C;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 OTHER_DETAILS: PHOSPHORYLATED ON THR160;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CYCLIN A2;\n",
      "COMPND    9 CHAIN: B, D;\n",
      "COMPND    10 FRAGMENT: RESIDUES 175-432;\n",
      "COMPND    11 ENGINEERED: YES;\n",
      "COMPND    12 MOL_ID: 3;\n",
      "COMPND    13 MOLECULE: SUBSTRATE PEPTIDE;\n",
      "COMPND    14 CHAIN: E, F;\n",
      "COMPND    15 ENGINEERED: YES;\n",
      "COMPND    16 OTHER_DETAILS: SEQUENCE HHASPRK\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GZ8.pdb A Results/activation_segments/unaligned/1GZ8_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GZ8.pdb with 2731 atoms.\n",
      "File Results/activation_segments/unaligned/1GZ8_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             17-MAY-02   1GZ8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE, CYCLIN-DEPENDENT PROTEIN KINASE 2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: COMPLEX WITH CYCLIN A OR CYCLIN E\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.0\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1GZ8_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             17-MAY-02   1GZ8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE, CYCLIN-DEPENDENT PROTEIN KINASE 2;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: COMPLEX WITH CYCLIN A OR CYCLIN E\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.0\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GZK.pdb A Results/activation_segments/unaligned/1GZK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GZK.pdb with 2481 atoms.\n",
      "File Results/activation_segments/unaligned/1GZK_A.pdb before post_process, first few lines:\n",
      "HEADER    KINASE                                  23-MAY-02   1GZK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN, RESIDUES (146 - 460);\n",
      "COMPND    5 SYNONYM: PROTEIN KINASE AKT-2, PROTEIN KINASE B BETA, RAC-PK-BETA,\n",
      "COMPND    6 PKB BETA;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "\n",
      "File Results/activation_segments/unaligned/1GZK_A.pdb after post_process, first few lines:\n",
      "HEADER    KINASE                                  23-MAY-02   1GZK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN, RESIDUES (146 - 460);\n",
      "COMPND    5 SYNONYM: PROTEIN KINASE AKT-2, PROTEIN KINASE B BETA, RAC-PK-BETA,\n",
      "COMPND    6 PKB BETA;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GZN.pdb A Results/activation_segments/unaligned/1GZN_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1GZN.pdb with 2445 atoms.\n",
      "File Results/activation_segments/unaligned/1GZN_A.pdb before post_process, first few lines:\n",
      "HEADER    KINASE                                  24-MAY-02   1GZN\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN AND HYDROPHOBIC MOTIF, RESIDUES 146-480;\n",
      "COMPND    5 SYNONYM: PROTEIN KINASE AKT-2, PROTEIN KINASE B BETA, RAC-PK-BETA,\n",
      "COMPND    6 PKB BETA;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.1\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "\n",
      "File Results/activation_segments/unaligned/1GZN_A.pdb after post_process, first few lines:\n",
      "HEADER    KINASE                                  24-MAY-02   1GZN\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN AND HYDROPHOBIC MOTIF, RESIDUES 146-480;\n",
      "COMPND    5 SYNONYM: PROTEIN KINASE AKT-2, PROTEIN KINASE B BETA, RAC-PK-BETA,\n",
      "COMPND    6 PKB BETA;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.1\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1H01.pdb A Results/activation_segments/unaligned/1H01_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1H01.pdb with 2661 atoms.\n",
      "File Results/activation_segments/unaligned/1H01_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             10-JUN-02   1H01\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN-DEPENDENT KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE, CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    5 EC: 2.7.11.22;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-GAHMEN,H.DE BONDT,S.-H.KIM\n",
      "REMARK     1  TITL   HIGH-RESOLUTION CRYSTAL STRUCTURES OF HUMAN\n",
      "REMARK     1  TITL 2 CYCLIN-DEPENDENT KINASE 2 WITH AND WITHOUT ATP:\n",
      "REMARK     1  TITL 3 BOUND WATERS AND NATURAL LIGAND AS A GUIDE FOR\n",
      "REMARK     1  TITL 4 INHIBITOR DESIGN\n",
      "REMARK     1  REF    J.MED.CHEM.                   V.  39  4540 1996\n",
      "REMARK     1  REFN                   ISSN 0022-2623\n",
      "REMARK     1  PMID   8917641\n",
      "REMARK     1  DOI    10.1021/JM960402A\n",
      "REMARK     2\n",
      "\n",
      "File Results/activation_segments/unaligned/1H01_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             10-JUN-02   1H01\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN-DEPENDENT KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE, CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    5 EC: 2.7.11.22;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-GAHMEN,H.DE BONDT,S.-H.KIM\n",
      "REMARK     1  TITL   HIGH-RESOLUTION CRYSTAL STRUCTURES OF HUMAN\n",
      "REMARK     1  TITL 2 CYCLIN-DEPENDENT KINASE 2 WITH AND WITHOUT ATP:\n",
      "REMARK     1  TITL 3 BOUND WATERS AND NATURAL LIGAND AS A GUIDE FOR\n",
      "REMARK     1  TITL 4 INHIBITOR DESIGN\n",
      "REMARK     1  REF    J.MED.CHEM.                   V.  39  4540 1996\n",
      "REMARK     1  REFN                   ISSN 0022-2623\n",
      "REMARK     1  PMID   8917641\n",
      "REMARK     1  DOI    10.1021/JM960402A\n",
      "REMARK     2\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1H1P.pdb A Results/activation_segments/unaligned/1H1P_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1H1P.pdb with 9388 atoms.\n",
      "File Results/activation_segments/unaligned/1H1P_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             21-JUL-02   1H1P\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, C;\n",
      "COMPND    4 SYNONYM: CYCLIN-DEPENDENT KINASE 2, P33 PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: PHOSPHORYLATED ON THR160;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CYCLIN A2;\n",
      "COMPND    10 CHAIN: B, D;\n",
      "COMPND    11 SYNONYM: CYCLIN A, CYCLIN A3;\n",
      "COMPND    12 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "\n",
      "File Results/activation_segments/unaligned/1H1P_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             21-JUL-02   1H1P\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, C;\n",
      "COMPND    4 SYNONYM: CYCLIN-DEPENDENT KINASE 2, P33 PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: PHOSPHORYLATED ON THR160;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CYCLIN A2;\n",
      "COMPND    10 CHAIN: B, D;\n",
      "COMPND    11 SYNONYM: CYCLIN A, CYCLIN A3;\n",
      "COMPND    12 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1H1W.pdb A Results/activation_segments/unaligned/1H1W_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1H1W.pdb with 2643 atoms.\n",
      "File Results/activation_segments/unaligned/1H1W_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             23-JUL-02   1H1W\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: 3-PHOSPHOINOSITIDE DEPENDENT PROTEIN KINASE-1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE CATALYTIC DOMAIN, RESIDUES 71-359;\n",
      "COMPND    5 SYNONYM: HPDK1;\n",
      "COMPND    6 EC: 2.7.11.1;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 OTHER_DETAILS: ACTIVATION LOOP PHOSPHORYLATION (SER241)\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "\n",
      "File Results/activation_segments/unaligned/1H1W_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             23-JUL-02   1H1W\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: 3-PHOSPHOINOSITIDE DEPENDENT PROTEIN KINASE-1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE CATALYTIC DOMAIN, RESIDUES 71-359;\n",
      "COMPND    5 SYNONYM: HPDK1;\n",
      "COMPND    6 EC: 2.7.11.1;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 OTHER_DETAILS: ACTIVATION LOOP PHOSPHORYLATION (SER241)\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1H4L.pdb A Results/activation_segments/unaligned/1H4L_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1H4L.pdb with 6935 atoms.\n",
      "File Results/activation_segments/unaligned/1H4L_A.pdb before post_process, first few lines:\n",
      "HEADER    KINASE/KINASE ACTIVATOR                 11-MAY-01   1H4L\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 5;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: TAU PROTEIN KINASE II CATALYTIC SUBUNIT, CDK5;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: CYCLIN-DEPENDENT KINASE 5 ACTIVATOR;\n",
      "COMPND    8 CHAIN: D, E;\n",
      "COMPND    9 FRAGMENT: RESIDUES 147-293;\n",
      "COMPND    10 SYNONYM: CDK5 ACTIVATOR 1, CYCLIN-DEPENDENT KINASE 5 REGULATORY\n",
      "COMPND    11 SUBUNIT 1, TAU PROTEIN KINASE II 23 KDA SUBUNIT, TPKII REGULATORY\n",
      "COMPND    12 SUBUNIT, P23, P25, P35;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.65 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "\n",
      "File Results/activation_segments/unaligned/1H4L_A.pdb after post_process, first few lines:\n",
      "HEADER    KINASE/KINASE ACTIVATOR                 11-MAY-01   1H4L\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 5;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: TAU PROTEIN KINASE II CATALYTIC SUBUNIT, CDK5;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: CYCLIN-DEPENDENT KINASE 5 ACTIVATOR;\n",
      "COMPND    8 CHAIN: D, E;\n",
      "COMPND    9 FRAGMENT: RESIDUES 147-293;\n",
      "COMPND    10 SYNONYM: CDK5 ACTIVATOR 1, CYCLIN-DEPENDENT KINASE 5 REGULATORY\n",
      "COMPND    11 SUBUNIT 1, TAU PROTEIN KINASE II 23 KDA SUBUNIT, TPKII REGULATORY\n",
      "COMPND    12 SUBUNIT, P23, P25, P35;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.65 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1I44.pdb A Results/activation_segments/unaligned/1I44_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1I44.pdb with 2378 atoms.\n",
      "File Results/activation_segments/unaligned/1I44_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             19-FEB-01   1I44\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: IR;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 0.5\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH AND HUBER, 1991\n",
      "\n",
      "File Results/activation_segments/unaligned/1I44_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             19-FEB-01   1I44\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: IR;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 0.5\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH AND HUBER, 1991\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1IA8.pdb A Results/activation_segments/unaligned/1IA8_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1IA8.pdb with 2370 atoms.\n",
      "File Results/activation_segments/unaligned/1IA8_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-MAR-01   1IA8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CHK1 CHECKPOINT KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CHK1KD (RESIDUES 1-289);\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.70 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.70\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 7.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 1.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : NULL\n",
      "\n",
      "File Results/activation_segments/unaligned/1IA8_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-MAR-01   1IA8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CHK1 CHECKPOINT KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CHK1KD (RESIDUES 1-289);\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.70 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.70\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 7.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 1.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : NULL\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1IAN.pdb A Results/activation_segments/unaligned/1IAN_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1IAN.pdb with 409 atoms.\n",
      "File Results/activation_segments/unaligned/1IAN_A.pdb before post_process, first few lines:\n",
      "HEADER    SERINE/THREONINE-PROTEIN KINASE         07-MAR-97   1IAN\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: P38 MAP KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: CSBP, RK, P38;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   A.CUENDA,J.ROUSE,Y.N.DOZA,R.MEIER,P.COHEN,T.F.GALLAGHER,\n",
      "REMARK     1  AUTH 2 P.R.YOUNG,J.C.LEE\n",
      "REMARK     1  TITL   SB 203580 IS A SPECIFIC INHIBITOR OF A MAP KINASE\n",
      "REMARK     HOMOLOGUE\n",
      "REMARK     1  TITL 2 WHICH IS STIMULATED BY CELLULAR STRESSES AND\n",
      "REMARK     INTERLEUKIN-1\n",
      "REMARK     1  REF    FEBS LETT.                    V. 364   229 1995\n",
      "REMARK     1  REFN                   ISSN 0014-5793\n",
      "REMARK     2\n",
      "\n",
      "File Results/activation_segments/unaligned/1IAN_A.pdb after post_process, first few lines:\n",
      "HEADER    SERINE/THREONINE-PROTEIN KINASE         07-MAR-97   1IAN\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: P38 MAP KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: CSBP, RK, P38;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   A.CUENDA,J.ROUSE,Y.N.DOZA,R.MEIER,P.COHEN,T.F.GALLAGHER,\n",
      "REMARK     1  AUTH 2 P.R.YOUNG,J.C.LEE\n",
      "REMARK     1  TITL   SB 203580 IS A SPECIFIC INHIBITOR OF A MAP KINASE\n",
      "REMARK     HOMOLOGUE\n",
      "REMARK     1  TITL 2 WHICH IS STIMULATED BY CELLULAR STRESSES AND\n",
      "REMARK     INTERLEUKIN-1\n",
      "REMARK     1  REF    FEBS LETT.                    V. 364   229 1995\n",
      "REMARK     1  REFN                   ISSN 0014-5793\n",
      "REMARK     2\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1IRK.pdb A Results/activation_segments/unaligned/1IRK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1IRK.pdb with 2585 atoms.\n",
      "File Results/activation_segments/unaligned/1IRK_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE (PHOSPHOTRANSFERASE)        02-JAN-95   1IRK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR TYROSINE KINASE DOMAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   L.WEI,S.R.HUBBARD,W.A.HENDRICKSON,L.ELLIS\n",
      "REMARK     1  TITL   EXPRESSION, CHARACTERIZATION AND CRYSTALLIZATION OF THE\n",
      "REMARK     1  TITL 2 CATALYTIC CORE OF THE HUMAN INSULIN RECEPTOR PROTEIN\n",
      "REMARK     1  TITL 3 TYROSINE KINASE DOMAIN\n",
      "REMARK     1  REF    TO BE PUBLISHED\n",
      "REMARK     1  REFN\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH\n",
      "REMARK     Y.EBINA,L.ELLIS,K.JARNAGIN,M.EDERY,L.GRAF,E.CLAUSER,J.-H.OU,\n",
      "REMARK     1  AUTH 2 F.MASIARZ,Y.W.KAN,I.D.GOLDFINE,R.A.ROTH,W.J.RUTTER\n",
      "REMARK     1  TITL   THE HUMAN INSULIN RECEPTOR CDNA: THE STRUCTURAL BASIS\n",
      "REMARK     FOR\n",
      "\n",
      "File Results/activation_segments/unaligned/1IRK_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE (PHOSPHOTRANSFERASE)        02-JAN-95   1IRK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR TYROSINE KINASE DOMAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   L.WEI,S.R.HUBBARD,W.A.HENDRICKSON,L.ELLIS\n",
      "REMARK     1  TITL   EXPRESSION, CHARACTERIZATION AND CRYSTALLIZATION OF THE\n",
      "REMARK     1  TITL 2 CATALYTIC CORE OF THE HUMAN INSULIN RECEPTOR PROTEIN\n",
      "REMARK     1  TITL 3 TYROSINE KINASE DOMAIN\n",
      "REMARK     1  REF    TO BE PUBLISHED\n",
      "REMARK     1  REFN\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH\n",
      "REMARK     Y.EBINA,L.ELLIS,K.JARNAGIN,M.EDERY,L.GRAF,E.CLAUSER,J.-H.OU,\n",
      "REMARK     1  AUTH 2 F.MASIARZ,Y.W.KAN,I.D.GOLDFINE,R.A.ROTH,W.J.RUTTER\n",
      "REMARK     1  TITL   THE HUMAN INSULIN RECEPTOR CDNA: THE STRUCTURAL BASIS\n",
      "REMARK     FOR\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1J3H.pdb A Results/activation_segments/unaligned/1J3H_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1J3H.pdb with 5002 atoms.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/marmatt/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/topology/guessers.py:146: UserWarning: Failed to guess the mass for the following atom types: HG\n",
      "  warnings.warn(\"Failed to guess the mass for the following atom types: {}\".format(atom_type))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File Results/activation_segments/unaligned/1J3H_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             31-JAN-03   1J3H\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PKAC-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   MADHUSUDAN,P.AKAMINE,N.H.XUONG,S.S.TAYLOR\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF A TRANSITION STATE MIMIC OF THE\n",
      "REMARK     1  TITL 2 CATALYTIC SUBUNIT OF CAMP-DEPENDENT PROTEIN KINASE\n",
      "REMARK     1  REF    NAT.STRUCT.BIOL.              V.   9   273 2002\n",
      "REMARK     1  REFN                   ISSN 1072-8368\n",
      "REMARK     1  DOI    10.1038/NSB780\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   R.KARLSSON,J.ZHENG,N.XUONG,S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   STRUCTURE OF THE MAMMALIAN CATALYTIC SUBUNIT OF\n",
      "REMARK     1  TITL 2 CAMP-DEPENDENT PROTEIN KINASE AND AN INHIBITOR PEPTIDE\n",
      "\n",
      "File Results/activation_segments/unaligned/1J3H_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             31-JAN-03   1J3H\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PKAC-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   MADHUSUDAN,P.AKAMINE,N.H.XUONG,S.S.TAYLOR\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF A TRANSITION STATE MIMIC OF THE\n",
      "REMARK     1  TITL 2 CATALYTIC SUBUNIT OF CAMP-DEPENDENT PROTEIN KINASE\n",
      "REMARK     1  REF    NAT.STRUCT.BIOL.              V.   9   273 2002\n",
      "REMARK     1  REFN                   ISSN 1072-8368\n",
      "REMARK     1  DOI    10.1038/NSB780\n",
      "REMARK     1 REFERENCE 2\n",
      "REMARK     1  AUTH   R.KARLSSON,J.ZHENG,N.XUONG,S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   STRUCTURE OF THE MAMMALIAN CATALYTIC SUBUNIT OF\n",
      "REMARK     1  TITL 2 CAMP-DEPENDENT PROTEIN KINASE AND AN INHIBITOR PEPTIDE\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JBP.pdb E Results/activation_segments/unaligned/1JBP_E.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JBP.pdb with 3069 atoms.\n",
      "File Results/activation_segments/unaligned/1JBP_E.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-JUN-01   1JBP\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, MUSCLE/BRAIN FORM;\n",
      "COMPND    9 CHAIN: S;\n",
      "COMPND    10 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    11 SYNONYM: PKI-ALPHA;\n",
      "COMPND    12 ENGINEERED: YES;\n",
      "COMPND    13 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.R.KNIGHTON,S.M.BELL,J.ZHENG,L.F.TEN EYCK,N.XUONG,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   2.0 A REFINED CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT\n",
      "\n",
      "File Results/activation_segments/unaligned/1JBP_E.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-JUN-01   1JBP\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, MUSCLE/BRAIN FORM;\n",
      "COMPND    9 CHAIN: S;\n",
      "COMPND    10 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    11 SYNONYM: PKI-ALPHA;\n",
      "COMPND    12 ENGINEERED: YES;\n",
      "COMPND    13 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.R.KNIGHTON,S.M.BELL,J.ZHENG,L.F.TEN EYCK,N.XUONG,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   2.0 A REFINED CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JNK.pdb A Results/activation_segments/unaligned/1JNK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JNK.pdb with 3016 atoms.\n",
      "File Results/activation_segments/unaligned/1JNK_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             03-JUN-98   1JNK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: C-JUN N-TERMINAL KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: RESIDUES 45 - 400;\n",
      "COMPND    5 SYNONYM: JNK3;\n",
      "COMPND    6 EC: 2.7.1.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.843\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.30\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 30.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 2.000\n",
      "\n",
      "File Results/activation_segments/unaligned/1JNK_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             03-JUN-98   1JNK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: C-JUN N-TERMINAL KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: RESIDUES 45 - 400;\n",
      "COMPND    5 SYNONYM: JNK3;\n",
      "COMPND    6 EC: 2.7.1.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.843\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.30\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 30.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 2.000\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JOW.pdb B Results/activation_segments/unaligned/1JOW_B.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JOW.pdb with 3986 atoms.\n",
      "File Results/activation_segments/unaligned/1JOW_B.pdb before post_process, first few lines:\n",
      "HEADER    CELL CYCLE/TRANSFERASE                  31-JUL-01   1JOW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN HOMOLOG;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: V-CYCLIN;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: CELL DIVISION PROTEIN KINASE 6;\n",
      "COMPND    8 CHAIN: B;\n",
      "COMPND    9 FRAGMENT: RESIDUES 1-308;\n",
      "COMPND    10 SYNONYM: CDK6, SERINE/THREONINE-PROTEIN KINASE PLSTIRE;\n",
      "COMPND    11 EC: 2.7.1.-;\n",
      "COMPND    12 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-GAHMEN,S.H.KIM\n",
      "REMARK     1  TITL   CRYSTALLIZATION OF A COMPLEX BETWEEN HUMAN CDK6 AND A\n",
      "REMARK     1  TITL 2 VIRUS-ENCODED CYCLIN IS CRITICALLY DEPENDENT ON THE\n",
      "REMARK     ADDITION\n",
      "\n",
      "File Results/activation_segments/unaligned/1JOW_B.pdb after post_process, first few lines:\n",
      "HEADER    CELL CYCLE/TRANSFERASE                  31-JUL-01   1JOW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CYCLIN HOMOLOG;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: V-CYCLIN;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: CELL DIVISION PROTEIN KINASE 6;\n",
      "COMPND    8 CHAIN: B;\n",
      "COMPND    9 FRAGMENT: RESIDUES 1-308;\n",
      "COMPND    10 SYNONYM: CDK6, SERINE/THREONINE-PROTEIN KINASE PLSTIRE;\n",
      "COMPND    11 EC: 2.7.1.-;\n",
      "COMPND    12 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-GAHMEN,S.H.KIM\n",
      "REMARK     1  TITL   CRYSTALLIZATION OF A COMPLEX BETWEEN HUMAN CDK6 AND A\n",
      "REMARK     1  TITL 2 VIRUS-ENCODED CYCLIN IS CRITICALLY DEPENDENT ON THE\n",
      "REMARK     ADDITION\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JPA.pdb A Results/activation_segments/unaligned/1JPA_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JPA.pdb with 4641 atoms.\n",
      "File Results/activation_segments/unaligned/1JPA_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             01-AUG-01   1JPA\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: NEURAL KINASE, NUK=EPH/ELK/ECK FAMILY RECEPTOR-LIKE\n",
      "COMPND    3 TYROSINE KINASE;\n",
      "COMPND    4 CHAIN: A, B;\n",
      "COMPND    5 FRAGMENT: PARTIAL JUXTAMEMBRANE, ENTIRE KINASE DOMAIN;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.91 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1JPA_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             01-AUG-01   1JPA\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: NEURAL KINASE, NUK=EPH/ELK/ECK FAMILY RECEPTOR-LIKE\n",
      "COMPND    3 TYROSINE KINASE;\n",
      "COMPND    4 CHAIN: A, B;\n",
      "COMPND    5 FRAGMENT: PARTIAL JUXTAMEMBRANE, ENTIRE KINASE DOMAIN;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.91 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JQH.pdb A Results/activation_segments/unaligned/1JQH_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JQH.pdb with 7292 atoms.\n",
      "File Results/activation_segments/unaligned/1JQH_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             07-AUG-01   1JQH\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: IGF-1 RECEPTOR KINASE;\n",
      "COMPND    3 CHAIN: A, B, C;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX 2000\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1JQH_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             07-AUG-01   1JQH\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: IGF-1 RECEPTOR KINASE;\n",
      "COMPND    3 CHAIN: A, B, C;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX 2000\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JWH.pdb A Results/activation_segments/unaligned/1JWH_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1JWH.pdb with 9221 atoms.\n",
      "File Results/activation_segments/unaligned/1JWH_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             04-SEP-01   1JWH\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE II, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: CK II, CATALYTIC SUBUNIT;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CASEIN KINASE II BETA CHAIN;\n",
      "COMPND    9 CHAIN: C, D;\n",
      "COMPND    10 SYNONYM: CK II, REGULATORY SUBUNIT;\n",
      "COMPND    11 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,I.ERMAKOWA,O.G.ISSINGER\n",
      "REMARK     1  TITL   CRYSTALLIZATION AND PRELIMINARY CHARACTERIZATION OF\n",
      "REMARK     CRYSTALS\n",
      "REMARK     1  TITL 2 OF HUMAN PROTEIN KINASE CK2\n",
      "REMARK     1  REF    ACTA CRYSTALLOGR.,SECT.D      V.  56  1680 2000\n",
      "\n",
      "File Results/activation_segments/unaligned/1JWH_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             04-SEP-01   1JWH\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE II, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: CK II, CATALYTIC SUBUNIT;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CASEIN KINASE II BETA CHAIN;\n",
      "COMPND    9 CHAIN: C, D;\n",
      "COMPND    10 SYNONYM: CK II, REGULATORY SUBUNIT;\n",
      "COMPND    11 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,I.ERMAKOWA,O.G.ISSINGER\n",
      "REMARK     1  TITL   CRYSTALLIZATION AND PRELIMINARY CHARACTERIZATION OF\n",
      "REMARK     CRYSTALS\n",
      "REMARK     1  TITL 2 OF HUMAN PROTEIN KINASE CK2\n",
      "REMARK     1  REF    ACTA CRYSTALLOGR.,SECT.D      V.  56  1680 2000\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1K2P.pdb A Results/activation_segments/unaligned/1K2P_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1K2P.pdb with 4178 atoms.\n",
      "File Results/activation_segments/unaligned/1K2P_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             28-SEP-01   1K2P\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE BTK;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: BRUTON'S TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.1\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.10\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 5.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 2.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : NULL\n",
      "\n",
      "File Results/activation_segments/unaligned/1K2P_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             28-SEP-01   1K2P\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE BTK;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: BRUTON'S TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.1\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.10\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 5.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 2.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : NULL\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1K3A.pdb A Results/activation_segments/unaligned/1K3A_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1K3A.pdb with 2548 atoms.\n",
      "File Results/activation_segments/unaligned/1K3A_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             02-OCT-01   1K3A\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN-LIKE GROWTH FACTOR 1 RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: BETA CHAIN, KINASE DOMAIN (RESIDUES 988-1286);\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: INSULIN RECEPTOR SUBSTRATE 1;\n",
      "COMPND    9 CHAIN: B;\n",
      "COMPND    10 SYNONYM: IRS-1;\n",
      "COMPND    11 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "\n",
      "File Results/activation_segments/unaligned/1K3A_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             02-OCT-01   1K3A\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN-LIKE GROWTH FACTOR 1 RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: BETA CHAIN, KINASE DOMAIN (RESIDUES 988-1286);\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: INSULIN RECEPTOR SUBSTRATE 1;\n",
      "COMPND    9 CHAIN: B;\n",
      "COMPND    10 SYNONYM: IRS-1;\n",
      "COMPND    11 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1K9A.pdb A Results/activation_segments/unaligned/1K9A_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1K9A.pdb with 21313 atoms.\n",
      "File Results/activation_segments/unaligned/1K9A_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             28-OCT-01   1K9A\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CARBOXYL-TERMINAL SRC KINASE;\n",
      "COMPND    3 CHAIN: A, B, C, D, E, F;\n",
      "COMPND    4 SYNONYM: TYROSINE-PROTEIN KINASE CSK;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1K9A_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             28-OCT-01   1K9A\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CARBOXYL-TERMINAL SRC KINASE;\n",
      "COMPND    3 CHAIN: A, B, C, D, E, F;\n",
      "COMPND    4 SYNONYM: TYROSINE-PROTEIN KINASE CSK;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1KSW.pdb A Results/activation_segments/unaligned/1KSW_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1KSW.pdb with 3691 atoms.\n",
      "File Results/activation_segments/unaligned/1KSW_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             14-JAN-02   1KSW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTO-ONCOGENE TYROSINE-PROTEIN KINASE SRC;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: SH3, SH2 AND KINASE DOMAINS;\n",
      "COMPND    5 SYNONYM: C-SRC;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.851\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.80\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 20.00\n",
      "\n",
      "File Results/activation_segments/unaligned/1KSW_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             14-JAN-02   1KSW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTO-ONCOGENE TYROSINE-PROTEIN KINASE SRC;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: SH3, SH2 AND KINASE DOMAINS;\n",
      "COMPND    5 SYNONYM: C-SRC;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.851\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.80\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 20.00\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1KWP.pdb A Results/activation_segments/unaligned/1KWP_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1KWP.pdb with 4899 atoms.\n",
      "File Results/activation_segments/unaligned/1KWP_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             30-JAN-02   1KWP\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MAP KINASE ACTIVATED PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: MAPKAP2;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1KWP_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             30-JAN-02   1KWP\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MAP KINASE ACTIVATED PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: MAPKAP2;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1L3R.pdb E Results/activation_segments/unaligned/1L3R_E.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1L3R.pdb with 3160 atoms.\n",
      "File Results/activation_segments/unaligned/1L3R_E.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             28-FEB-02   1L3R\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, MUSCLE/BRAIN FORM;\n",
      "COMPND    9 CHAIN: I;\n",
      "COMPND    10 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    11 SYNONYM: PKI-ALPHA;\n",
      "COMPND    12 ENGINEERED: YES;\n",
      "COMPND    13 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.R.KNIGHTON,S.M.BELL,J.ZHENG,L.F.TEN EYCK,N.XUONG,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   2.0 A REFINED CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT\n",
      "\n",
      "File Results/activation_segments/unaligned/1L3R_E.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             28-FEB-02   1L3R\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, MUSCLE/BRAIN FORM;\n",
      "COMPND    9 CHAIN: I;\n",
      "COMPND    10 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    11 SYNONYM: PKI-ALPHA;\n",
      "COMPND    12 ENGINEERED: YES;\n",
      "COMPND    13 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.R.KNIGHTON,S.M.BELL,J.ZHENG,L.F.TEN EYCK,N.XUONG,\n",
      "REMARK     1  AUTH 2 S.S.TAYLOR,J.M.SOWADSKI\n",
      "REMARK     1  TITL   2.0 A REFINED CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1LEW.pdb A Results/activation_segments/unaligned/1LEW_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1LEW.pdb with 2955 atoms.\n",
      "File Results/activation_segments/unaligned/1LEW_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             10-APR-02   1LEW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 14;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MAPK 14,CRK1,MITOGEN-ACTIVATED PROTEIN KINASE P38 ALPHA,MAP\n",
      "COMPND    5 KINASE P38 ALPHA;\n",
      "COMPND    6 EC: 2.7.11.24;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: MYOCYTE-SPECIFIC ENHANCER FACTOR 2A;\n",
      "COMPND    10 CHAIN: B;\n",
      "COMPND    11 SYNONYM: SERUM RESPONSE FACTOR-LIKE PROTEIN 1;\n",
      "COMPND    12 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "\n",
      "File Results/activation_segments/unaligned/1LEW_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             10-APR-02   1LEW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 14;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MAPK 14,CRK1,MITOGEN-ACTIVATED PROTEIN KINASE P38 ALPHA,MAP\n",
      "COMPND    5 KINASE P38 ALPHA;\n",
      "COMPND    6 EC: 2.7.11.24;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: MYOCYTE-SPECIFIC ENHANCER FACTOR 2A;\n",
      "COMPND    10 CHAIN: B;\n",
      "COMPND    11 SYNONYM: SERUM RESPONSE FACTOR-LIKE PROTEIN 1;\n",
      "COMPND    12 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1LUF.pdb A Results/activation_segments/unaligned/1LUF_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1LUF.pdb with 2296 atoms.\n",
      "File Results/activation_segments/unaligned/1LUF_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-MAY-02   1LUF\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MUSCLE-SPECIFIC TYROSINE KINASE RECEPTOR MUSK;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CYTOPLASMIC REGION (RESIDUES 526-868);\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.05 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.1\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1LUF_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-MAY-02   1LUF\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MUSCLE-SPECIFIC TYROSINE KINASE RECEPTOR MUSK;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CYTOPLASMIC REGION (RESIDUES 526-868);\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.05 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.1\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1M14.pdb A Results/activation_segments/unaligned/1M14_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1M14.pdb with 2469 atoms.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/marmatt/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/topology/guessers.py:146: UserWarning: Failed to guess the mass for the following atom types: AL\n",
      "  warnings.warn(\"Failed to guess the mass for the following atom types: {}\".format(atom_type))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File Results/activation_segments/unaligned/1M14_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             17-JUN-02   1M14\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: EPIDERMAL GROWTH FACTOR RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN (RESIDUES 671-998);\n",
      "COMPND    5 SYNONYM: RECEPTOR PROTEIN-TYROSINE KINASE ERBB-1;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 98.1\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.60\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 50.00\n",
      "\n",
      "File Results/activation_segments/unaligned/1M14_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             17-JUN-02   1M14\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: EPIDERMAL GROWTH FACTOR RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN (RESIDUES 671-998);\n",
      "COMPND    5 SYNONYM: RECEPTOR PROTEIN-TYROSINE KINASE ERBB-1;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 98.1\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.60\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 50.00\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1M2P.pdb A Results/activation_segments/unaligned/1M2P_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1M2P.pdb with 2933 atoms.\n",
      "File Results/activation_segments/unaligned/1M2P_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             25-JUN-02   1M2P\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE II, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATLYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: CKII;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1M2P_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             25-JUN-02   1M2P\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE II, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATLYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: CKII;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1M7N.pdb A Results/activation_segments/unaligned/1M7N_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1M7N.pdb with 4987 atoms.\n",
      "File Results/activation_segments/unaligned/1M7N_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-JUL-02   1M7N\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN-LIKE GROWTH FACTOR I RECEPTOR;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN (RESIDUES 974-1286);\n",
      "COMPND    5 SYNONYM: CD221 ANTIGEN;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.70 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1M7N_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-JUL-02   1M7N\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN-LIKE GROWTH FACTOR I RECEPTOR;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN (RESIDUES 974-1286);\n",
      "COMPND    5 SYNONYM: CD221 ANTIGEN;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.70 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1M7Q.pdb A Results/activation_segments/unaligned/1M7Q_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1M7Q.pdb with 2952 atoms.\n",
      "File Results/activation_segments/unaligned/1M7Q_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-JUL-02   1M7Q\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 14;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN-ACTIVATED PROTEIN KINASE P38, MAP KINASE P38,\n",
      "COMPND    5 CYTOKINE SUPPRESSIVE ANTI-INFLAMMATORY DRUG BINDING PROTEIN, CSAID\n",
      "COMPND    6 BINDING PROTEIN, CSBP, MAX-INTERACTING PROTEIN 2, MAP KINASE MXI2;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "File Results/activation_segments/unaligned/1M7Q_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-JUL-02   1M7Q\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 14;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN-ACTIVATED PROTEIN KINASE P38, MAP KINASE P38,\n",
      "COMPND    5 CYTOKINE SUPPRESSIVE ANTI-INFLAMMATORY DRUG BINDING PROTEIN, CSAID\n",
      "COMPND    6 BINDING PROTEIN, CSBP, MAX-INTERACTING PROTEIN 2, MAP KINASE MXI2;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MP8.pdb A Results/activation_segments/unaligned/1MP8_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MP8.pdb with 2278 atoms.\n",
      "File Results/activation_segments/unaligned/1MP8_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             11-SEP-02   1MP8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FOCAL ADHESION KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: FAK, FADK 1, PP125FAK, PROTEIN-TYROSINE KINASE 2;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.0\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1MP8_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             11-SEP-02   1MP8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FOCAL ADHESION KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: FAK, FADK 1, PP125FAK, PROTEIN-TYROSINE KINASE 2;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.0\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MQ4.pdb A Results/activation_segments/unaligned/1MQ4_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MQ4.pdb with 2211 atoms.\n",
      "File Results/activation_segments/unaligned/1MQ4_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             13-SEP-02   1MQ4\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: AURORA-RELATED KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: AURORA-A PROTEIN KINASE;\n",
      "COMPND    6 EC: 2.7.-.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1MQ4_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             13-SEP-02   1MQ4\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: AURORA-RELATED KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: AURORA-A PROTEIN KINASE;\n",
      "COMPND    6 EC: 2.7.-.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MQB.pdb A Results/activation_segments/unaligned/1MQB_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MQB.pdb with 4349 atoms.\n",
      "File Results/activation_segments/unaligned/1MQB_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             16-SEP-02   1MQB\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: EPHRIN TYPE-A RECEPTOR 2;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: EPHA2 RECEPTOR TYROSINE KINASE, TYROSINE-PROTEIN KINASE\n",
      "COMPND    6 RECEPTOR ECK, EPITHELIAL CELL KINASE;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1MQB_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             16-SEP-02   1MQB\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: EPHRIN TYPE-A RECEPTOR 2;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: EPHA2 RECEPTOR TYROSINE KINASE, TYROSINE-PROTEIN KINASE\n",
      "COMPND    6 RECEPTOR ECK, EPITHELIAL CELL KINASE;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MRU.pdb A Results/activation_segments/unaligned/1MRU_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MRU.pdb with 4217 atoms.\n",
      "File Results/activation_segments/unaligned/1MRU_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             18-SEP-02   1MRU\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROBABLE SERINE/THREONINE-PROTEIN KINASE PKNB;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 SYNONYM: REGULATORY PROKARYOTIC PROTEIN;\n",
      "COMPND    6 EC: 2.7.1.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    3.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1MRU_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             18-SEP-02   1MRU\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROBABLE SERINE/THREONINE-PROTEIN KINASE PKNB;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 SYNONYM: REGULATORY PROKARYOTIC PROTEIN;\n",
      "COMPND    6 EC: 2.7.1.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    3.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MRV.pdb A Results/activation_segments/unaligned/1MRV_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MRV.pdb with 2150 atoms.\n",
      "File Results/activation_segments/unaligned/1MRV_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             18-SEP-02   1MRV\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: RAC-PK-BETA; PROTEIN KINASE AKT-2; PROTEIN KINASE B, BETA;\n",
      "COMPND    6 PKB BETA;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "File Results/activation_segments/unaligned/1MRV_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             18-SEP-02   1MRV\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: RAC-PK-BETA; PROTEIN KINASE AKT-2; PROTEIN KINASE B, BETA;\n",
      "COMPND    6 PKB BETA;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MUO.pdb A Results/activation_segments/unaligned/1MUO_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1MUO.pdb with 2029 atoms.\n",
      "File Results/activation_segments/unaligned/1MUO_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             24-SEP-02   1MUO\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: AURORA-RELATED KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: AURORA-2 KINASE DOMAIN, RESIDUES 107-403;\n",
      "COMPND    5 SYNONYM: AURORA-2\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.90\n",
      "\n",
      "File Results/activation_segments/unaligned/1MUO_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             24-SEP-02   1MUO\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: AURORA-RELATED KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: AURORA-2 KINASE DOMAIN, RESIDUES 107-403;\n",
      "COMPND    5 SYNONYM: AURORA-2\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.90\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1NA7.pdb A Results/activation_segments/unaligned/1NA7_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1NA7.pdb with 2848 atoms.\n",
      "File Results/activation_segments/unaligned/1NA7_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             27-NOV-02   1NA7\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE II, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: CK II;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,L.A.PINNA,O.-G.ISSINGER,D.SCHOMBURG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT OF PROTEIN\n",
      "REMARK     KINASE\n",
      "REMARK     1  TITL 2 CK2 FROM ZEA MAYS AT 2.1 A RESOLUTION\n",
      "REMARK     1  REF    EMBO J.                       V.  17  2451 1998\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "REMARK     1  DOI    10.1093/EMBOJ/17.9.2451\n",
      "REMARK     1 REFERENCE 2\n",
      "\n",
      "File Results/activation_segments/unaligned/1NA7_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             27-NOV-02   1NA7\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE II, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: CK II;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,L.A.PINNA,O.-G.ISSINGER,D.SCHOMBURG\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF THE CATALYTIC SUBUNIT OF PROTEIN\n",
      "REMARK     KINASE\n",
      "REMARK     1  TITL 2 CK2 FROM ZEA MAYS AT 2.1 A RESOLUTION\n",
      "REMARK     1  REF    EMBO J.                       V.  17  2451 1998\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "REMARK     1  DOI    10.1093/EMBOJ/17.9.2451\n",
      "REMARK     1 REFERENCE 2\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1NXK.pdb A Results/activation_segments/unaligned/1NXK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1NXK.pdb with 9405 atoms.\n",
      "File Results/activation_segments/unaligned/1NXK_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             10-FEB-03   1NXK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MAP KINASE-ACTIVATED PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, B, C, D;\n",
      "COMPND    4 FRAGMENT: MK2;\n",
      "COMPND    5 SYNONYM: MAPK-ACTIVATED PROTEIN KINASE 2, MAPKAP KINASE 2, MAPKAPK-2;\n",
      "COMPND    6 EC: 2.7.1.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.70 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD WITH PHASES\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1NXK_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             10-FEB-03   1NXK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MAP KINASE-ACTIVATED PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, B, C, D;\n",
      "COMPND    4 FRAGMENT: MK2;\n",
      "COMPND    5 SYNONYM: MAPK-ACTIVATED PROTEIN KINASE 2, MAPKAP KINASE 2, MAPKAPK-2;\n",
      "COMPND    6 EC: 2.7.1.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.70 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD WITH PHASES\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1O6K.pdb A Results/activation_segments/unaligned/1O6K_A.pdb\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/marmatt/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/topology/guessers.py:146: UserWarning: Failed to guess the mass for the following atom types: SE\n",
      "  warnings.warn(\"Failed to guess the mass for the following atom types: {}\".format(atom_type))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1O6K.pdb with 2982 atoms.\n",
      "File Results/activation_segments/unaligned/1O6K_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-OCT-02   1O6K\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN, RESIDUES 146-481;\n",
      "COMPND    5 SYNONYM: RAC-PK-BETA, PROTEIN KINASE AKT-2, PROTEIN KINASE B, BETA,\n",
      "COMPND    6 PKB BETA, PROTEIN KINASE BETA, AKT2;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES;\n",
      "COMPND    10 MOL_ID: 2;\n",
      "COMPND    11 MOLECULE: GLYCOGEN SYNTHASE KINASE-3 BETA;\n",
      "COMPND    12 CHAIN: C;\n",
      "COMPND    13 FRAGMENT: PEPTIDE, RESIDUES 3-12;\n",
      "COMPND    14 SYNONYM: GSK-3 BETA;\n",
      "COMPND    15 EC: 2.7.1.37;\n",
      "COMPND    16 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.70 ANGSTROMS.\n",
      "\n",
      "File Results/activation_segments/unaligned/1O6K_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-OCT-02   1O6K\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN, RESIDUES 146-481;\n",
      "COMPND    5 SYNONYM: RAC-PK-BETA, PROTEIN KINASE AKT-2, PROTEIN KINASE B, BETA,\n",
      "COMPND    6 PKB BETA, PROTEIN KINASE BETA, AKT2;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES;\n",
      "COMPND    10 MOL_ID: 2;\n",
      "COMPND    11 MOLECULE: GLYCOGEN SYNTHASE KINASE-3 BETA;\n",
      "COMPND    12 CHAIN: C;\n",
      "COMPND    13 FRAGMENT: PEPTIDE, RESIDUES 3-12;\n",
      "COMPND    14 SYNONYM: GSK-3 BETA;\n",
      "COMPND    15 EC: 2.7.1.37;\n",
      "COMPND    16 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.70 ANGSTROMS.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1O6L.pdb A Results/activation_segments/unaligned/1O6L_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1O6L.pdb with 3118 atoms.\n",
      "File Results/activation_segments/unaligned/1O6L_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-OCT-02   1O6L\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN, RESIDUES 146 - 467;\n",
      "COMPND    5 SYNONYM: PROTEIN KINASE AKT-2, PROTEIN KINASE B BETA RAC-PK-BETA, PKB\n",
      "COMPND    6 BETA;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES;\n",
      "COMPND    10 OTHER_DETAILS: PKBBETA RESIDUES 146-467 FUSED TO EEQEMFEDFDYIADW;\n",
      "COMPND    11 MOL_ID: 2;\n",
      "COMPND    12 MOLECULE: GLYCOGEN SYNTHASE KINASE-3 BETA;\n",
      "COMPND    13 CHAIN: C;\n",
      "COMPND    14 FRAGMENT: PEPTIDE, RESIDUES 3-12;\n",
      "COMPND    15 SYNONYM: GSK-3 BETA;\n",
      "COMPND    16 EC: 2.7.1.37;\n",
      "COMPND    17 ENGINEERED: YES\n",
      "REMARK     2\n",
      "\n",
      "File Results/activation_segments/unaligned/1O6L_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-OCT-02   1O6L\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RAC-BETA SERINE/THREONINE PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN, RESIDUES 146 - 467;\n",
      "COMPND    5 SYNONYM: PROTEIN KINASE AKT-2, PROTEIN KINASE B BETA RAC-PK-BETA, PKB\n",
      "COMPND    6 BETA;\n",
      "COMPND    7 EC: 2.7.1.-;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES;\n",
      "COMPND    10 OTHER_DETAILS: PKBBETA RESIDUES 146-467 FUSED TO EEQEMFEDFDYIADW;\n",
      "COMPND    11 MOL_ID: 2;\n",
      "COMPND    12 MOLECULE: GLYCOGEN SYNTHASE KINASE-3 BETA;\n",
      "COMPND    13 CHAIN: C;\n",
      "COMPND    14 FRAGMENT: PEPTIDE, RESIDUES 3-12;\n",
      "COMPND    15 SYNONYM: GSK-3 BETA;\n",
      "COMPND    16 EC: 2.7.1.37;\n",
      "COMPND    17 ENGINEERED: YES\n",
      "REMARK     2\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1O6Y.pdb A Results/activation_segments/unaligned/1O6Y_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1O6Y.pdb with 2071 atoms.\n",
      "File Results/activation_segments/unaligned/1O6Y_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             21-OCT-02   1O6Y\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: SERINE/THREONINE-PROTEIN KINASE PKNB;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN, RESIDUES 1-279;\n",
      "COMPND    5 SYNONYM: PKNB;\n",
      "COMPND    6 EC: 2.7.11.1;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1O6Y_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             21-OCT-02   1O6Y\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: SERINE/THREONINE-PROTEIN KINASE PKNB;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN, RESIDUES 1-279;\n",
      "COMPND    5 SYNONYM: PKNB;\n",
      "COMPND    6 EC: 2.7.11.1;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OB3.pdb A Results/activation_segments/unaligned/1OB3_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OB3.pdb with 4632 atoms.\n",
      "File Results/activation_segments/unaligned/1OB3_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             24-JAN-03   1OB3\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION CONTROL PROTEIN 2 HOMOLOG;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PFPK5;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.90\n",
      "\n",
      "File Results/activation_segments/unaligned/1OB3_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             24-JAN-03   1OB3\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION CONTROL PROTEIN 2 HOMOLOG;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PFPK5;\n",
      "COMPND    5 EC: 2.7.1.-;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.90\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OGU.pdb A Results/activation_segments/unaligned/1OGU_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OGU.pdb with 8893 atoms.\n",
      "File Results/activation_segments/unaligned/1OGU_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             13-MAY-03   1OGU\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, C;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: PHOSPHORYLATED ON THR160;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CYCLIN A2;\n",
      "COMPND    10 CHAIN: B, D;\n",
      "COMPND    11 FRAGMENT: RESIDUES 174-432;\n",
      "COMPND    12 SYNONYM: CYCLIN A, CYCLIN A3;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "\n",
      "File Results/activation_segments/unaligned/1OGU_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             13-MAY-03   1OGU\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A, C;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 OTHER_DETAILS: PHOSPHORYLATED ON THR160;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CYCLIN A2;\n",
      "COMPND    10 CHAIN: B, D;\n",
      "COMPND    11 FRAGMENT: RESIDUES 174-432;\n",
      "COMPND    12 SYNONYM: CYCLIN A, CYCLIN A3;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OIR.pdb A Results/activation_segments/unaligned/1OIR_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OIR.pdb with 2439 atoms.\n",
      "File Results/activation_segments/unaligned/1OIR_A.pdb before post_process, first few lines:\n",
      "HEADER    KINASE                                  24-JUN-03   1OIR\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-GAHMEN,H.DE BONDT,S.-H.KIM\n",
      "REMARK     1  TITL   HIGH-RESOLUTION CRYSTAL STRUCTURES OF HUMAN\n",
      "REMARK     1  TITL 2 CYCLIN-DEPENDENT KINASE 2 WITH AND WITHOUT ATP:\n",
      "REMARK     1  TITL 3 BOUND WATERS AND NATURAL LIGAND AS A GUIDE FOR\n",
      "REMARK     1  TITL 4 INHIBITOR DESIGN\n",
      "REMARK     1  REF    J.MED.CHEM.                   V.  39  4540 1996\n",
      "REMARK     1  REFN                   ISSN 0022-2623\n",
      "REMARK     1  PMID   8917641\n",
      "REMARK     1  DOI    10.1021/JM960402A\n",
      "REMARK     2\n",
      "\n",
      "File Results/activation_segments/unaligned/1OIR_A.pdb after post_process, first few lines:\n",
      "HEADER    KINASE                                  24-JUN-03   1OIR\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-GAHMEN,H.DE BONDT,S.-H.KIM\n",
      "REMARK     1  TITL   HIGH-RESOLUTION CRYSTAL STRUCTURES OF HUMAN\n",
      "REMARK     1  TITL 2 CYCLIN-DEPENDENT KINASE 2 WITH AND WITHOUT ATP:\n",
      "REMARK     1  TITL 3 BOUND WATERS AND NATURAL LIGAND AS A GUIDE FOR\n",
      "REMARK     1  TITL 4 INHIBITOR DESIGN\n",
      "REMARK     1  REF    J.MED.CHEM.                   V.  39  4540 1996\n",
      "REMARK     1  REFN                   ISSN 0022-2623\n",
      "REMARK     1  PMID   8917641\n",
      "REMARK     1  DOI    10.1021/JM960402A\n",
      "REMARK     2\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OIT.pdb A Results/activation_segments/unaligned/1OIT_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OIT.pdb with 2416 atoms.\n",
      "File Results/activation_segments/unaligned/1OIT_A.pdb before post_process, first few lines:\n",
      "HEADER    KINASE                                  24-JUN-03   1OIT\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-GAHMEN,H.DE BONDT,S.-H.KIM\n",
      "REMARK     1  TITL   HIGH-RESOLUTION CRYSTAL STRUCTURES OF HUMAN CYCLIN-\n",
      "REMARK     DEPENDENT\n",
      "REMARK     1  TITL 2 KINASE 2 WITH AND WITHOUT ATP: BOUND WATERS AND NATURAL\n",
      "REMARK     1  TITL 3 LIGAND AS A GUIDE FOR INHIBITOR DESIGN\n",
      "REMARK     1  REF    J.MED.CHEM.                   V.  39  4540 1996\n",
      "REMARK     1  REFN                   ISSN 0022-2623\n",
      "REMARK     1  PMID   8917641\n",
      "REMARK     1  DOI    10.1021/JM960402A\n",
      "REMARK     2\n",
      "\n",
      "File Results/activation_segments/unaligned/1OIT_A.pdb after post_process, first few lines:\n",
      "HEADER    KINASE                                  24-JUN-03   1OIT\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CELL DIVISION PROTEIN KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: P33 PROTEIN KINASE;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   U.SCHULZE-GAHMEN,H.DE BONDT,S.-H.KIM\n",
      "REMARK     1  TITL   HIGH-RESOLUTION CRYSTAL STRUCTURES OF HUMAN CYCLIN-\n",
      "REMARK     DEPENDENT\n",
      "REMARK     1  TITL 2 KINASE 2 WITH AND WITHOUT ATP: BOUND WATERS AND NATURAL\n",
      "REMARK     1  TITL 3 LIGAND AS A GUIDE FOR INHIBITOR DESIGN\n",
      "REMARK     1  REF    J.MED.CHEM.                   V.  39  4540 1996\n",
      "REMARK     1  REFN                   ISSN 0022-2623\n",
      "REMARK     1  PMID   8917641\n",
      "REMARK     1  DOI    10.1021/JM960402A\n",
      "REMARK     2\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OKY.pdb A Results/activation_segments/unaligned/1OKY_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OKY.pdb with 2475 atoms.\n",
      "File Results/activation_segments/unaligned/1OKY_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             01-AUG-03   1OKY\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: 3-PHOSPHOINOSITIDE DEPENDENT PROTEIN KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN, RESIDUES 51-360;\n",
      "COMPND    5 SYNONYM: HPDK1;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1OKY_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             01-AUG-03   1OKY\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: 3-PHOSPHOINOSITIDE DEPENDENT PROTEIN KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN, RESIDUES 51-360;\n",
      "COMPND    5 SYNONYM: HPDK1;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OL5.pdb A Results/activation_segments/unaligned/1OL5_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OL5.pdb with 2607 atoms.\n",
      "File Results/activation_segments/unaligned/1OL5_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/CELL CYCLE                  06-AUG-03   1OL5\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: SERINE/THREONINE KINASE 6;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN, RESIDUES 122-403;\n",
      "COMPND    5 SYNONYM: AURORA-A, SERINE/THREONINE KINASE 15, AURORA/IPL1-RELATED\n",
      "COMPND    6 KINASE 1, AURORA-RELATED KINASE 1, HARK1, BREAST-TUMOR-AMPLIFIED\n",
      "COMPND    7 KINASE;\n",
      "COMPND    8 EC: 2.7.1.37;\n",
      "COMPND    9 ENGINEERED: YES;\n",
      "COMPND    10 OTHER_DETAILS: PHOSPHORYLATED ON THR287, THR288;\n",
      "COMPND    11 MOL_ID: 2;\n",
      "COMPND    12 MOLECULE: RESTRICTED EXPRESSION PROLIFERATION ASSOCIATED PROTEIN 100;\n",
      "COMPND    13 CHAIN: B;\n",
      "COMPND    14 FRAGMENT: N-TERMINAL, RESIDUES 1-43;\n",
      "COMPND    15 SYNONYM: TPX2, PROTEIN FLS353, DIL-2, P100, TARGETING PROTEIN FOR\n",
      "COMPND    16 XKLP2, HEPATOCELLULAR CARCINOMA-ASSOCIATED ANTIGEN 519;\n",
      "COMPND    17 ENGINEERED: YES\n",
      "REMARK     2\n",
      "\n",
      "File Results/activation_segments/unaligned/1OL5_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/CELL CYCLE                  06-AUG-03   1OL5\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: SERINE/THREONINE KINASE 6;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN, RESIDUES 122-403;\n",
      "COMPND    5 SYNONYM: AURORA-A, SERINE/THREONINE KINASE 15, AURORA/IPL1-RELATED\n",
      "COMPND    6 KINASE 1, AURORA-RELATED KINASE 1, HARK1, BREAST-TUMOR-AMPLIFIED\n",
      "COMPND    7 KINASE;\n",
      "COMPND    8 EC: 2.7.1.37;\n",
      "COMPND    9 ENGINEERED: YES;\n",
      "COMPND    10 OTHER_DETAILS: PHOSPHORYLATED ON THR287, THR288;\n",
      "COMPND    11 MOL_ID: 2;\n",
      "COMPND    12 MOLECULE: RESTRICTED EXPRESSION PROLIFERATION ASSOCIATED PROTEIN 100;\n",
      "COMPND    13 CHAIN: B;\n",
      "COMPND    14 FRAGMENT: N-TERMINAL, RESIDUES 1-43;\n",
      "COMPND    15 SYNONYM: TPX2, PROTEIN FLS353, DIL-2, P100, TARGETING PROTEIN FOR\n",
      "COMPND    16 XKLP2, HEPATOCELLULAR CARCINOMA-ASSOCIATED ANTIGEN 519;\n",
      "COMPND    17 ENGINEERED: YES\n",
      "REMARK     2\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OMW.pdb A Results/activation_segments/unaligned/1OMW_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OMW.pdb with 8151 atoms.\n",
      "File Results/activation_segments/unaligned/1OMW_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             26-FEB-03   1OMW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: G-PROTEIN COUPLED RECEPTOR KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: GRK2, BETA-ARK-1, BETA-ADRENERGIC RECEPTOR KINASE 1;\n",
      "COMPND    5 EC: 2.7.1.126;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: GUANINE NUCLEOTIDE-BINDING PROTEIN G(I)/G(S)/G(T) BETA\n",
      "COMPND    10 SUBUNIT 1;\n",
      "COMPND    11 CHAIN: B;\n",
      "COMPND    12 SYNONYM: TRANSDUCIN BETA CHAIN 1;\n",
      "COMPND    13 ENGINEERED: YES;\n",
      "COMPND    14 MOL_ID: 3;\n",
      "COMPND    15 MOLECULE: GUANINE NUCLEOTIDE-BINDING PROTEIN G(I)/G(S)/G(O) GAMMA-2\n",
      "COMPND    16 SUBUNIT;\n",
      "COMPND    17 CHAIN: G;\n",
      "COMPND    18 SYNONYM: G GAMMA-I;\n",
      "\n",
      "File Results/activation_segments/unaligned/1OMW_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             26-FEB-03   1OMW\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: G-PROTEIN COUPLED RECEPTOR KINASE 2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: GRK2, BETA-ARK-1, BETA-ADRENERGIC RECEPTOR KINASE 1;\n",
      "COMPND    5 EC: 2.7.1.126;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: GUANINE NUCLEOTIDE-BINDING PROTEIN G(I)/G(S)/G(T) BETA\n",
      "COMPND    10 SUBUNIT 1;\n",
      "COMPND    11 CHAIN: B;\n",
      "COMPND    12 SYNONYM: TRANSDUCIN BETA CHAIN 1;\n",
      "COMPND    13 ENGINEERED: YES;\n",
      "COMPND    14 MOL_ID: 3;\n",
      "COMPND    15 MOLECULE: GUANINE NUCLEOTIDE-BINDING PROTEIN G(I)/G(S)/G(O) GAMMA-2\n",
      "COMPND    16 SUBUNIT;\n",
      "COMPND    17 CHAIN: G;\n",
      "COMPND    18 SYNONYM: G GAMMA-I;\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OPK.pdb A Results/activation_segments/unaligned/1OPK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OPK.pdb with 3933 atoms.\n",
      "File Results/activation_segments/unaligned/1OPK_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-MAR-03   1OPK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: SH3-SH2-KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: P150, C-ABL;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   O.HANTSCHEL,B.NAGAR,S.GUETTLER,J.KRETZSCHMAR,K.DOREY,\n",
      "REMARK     1  AUTH 2 J.KURIYAN,G.SUPERTI-FURGA\n",
      "REMARK     1  TITL   A MYRISTOYL/PHOSPHOTYROSINE SWITCH REGULATES C-ABL\n",
      "REMARK     1  REF    CELL(CAMBRIDGE,MASS.)         V. 112   845 2003\n",
      "REMARK     1  REFN                   ISSN 0092-8674\n",
      "REMARK     1  DOI    10.1016/S0092-8674(03)00191-0\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.80 ANGSTROMS.\n",
      "\n",
      "File Results/activation_segments/unaligned/1OPK_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-MAR-03   1OPK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTO-ONCOGENE TYROSINE-PROTEIN KINASE ABL1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: SH3-SH2-KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: P150, C-ABL;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   O.HANTSCHEL,B.NAGAR,S.GUETTLER,J.KRETZSCHMAR,K.DOREY,\n",
      "REMARK     1  AUTH 2 J.KURIYAN,G.SUPERTI-FURGA\n",
      "REMARK     1  TITL   A MYRISTOYL/PHOSPHOTYROSINE SWITCH REGULATES C-ABL\n",
      "REMARK     1  REF    CELL(CAMBRIDGE,MASS.)         V. 112   845 2003\n",
      "REMARK     1  REFN                   ISSN 0092-8674\n",
      "REMARK     1  DOI    10.1016/S0092-8674(03)00191-0\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.80 ANGSTROMS.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OPL.pdb A Results/activation_segments/unaligned/1OPL_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OPL.pdb with 6655 atoms.\n",
      "File Results/activation_segments/unaligned/1OPL_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-MAR-03   1OPL\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTO-ONCOGENE TYROSINE-PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: N-TERMINAL 531 RESIDUES (MYR-SH3-SH2-KINASE DOMAIN);\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   O.HANTSCHEL,B.NAGAR,S.GUETTLER,J.KRETZSCHMAR,K.DOREY,\n",
      "REMARK     1  AUTH 2 J.KURIYAN,G.SUPERTI-FURGA\n",
      "REMARK     1  TITL   A MYRISTOYL/PHOSPHOTYROSINE SWITCH REGULATES C-ABL\n",
      "REMARK     1  REF    CELL(CAMBRIDGE,MASS.)         V. 112   845 2003\n",
      "REMARK     1  REFN                   ISSN 0092-8674\n",
      "REMARK     1  DOI    10.1016/S0092-8674(03)00191-0\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    3.42 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1OPL_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-MAR-03   1OPL\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PROTO-ONCOGENE TYROSINE-PROTEIN KINASE;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: N-TERMINAL 531 RESIDUES (MYR-SH3-SH2-KINASE DOMAIN);\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   O.HANTSCHEL,B.NAGAR,S.GUETTLER,J.KRETZSCHMAR,K.DOREY,\n",
      "REMARK     1  AUTH 2 J.KURIYAN,G.SUPERTI-FURGA\n",
      "REMARK     1  TITL   A MYRISTOYL/PHOSPHOTYROSINE SWITCH REGULATES C-ABL\n",
      "REMARK     1  REF    CELL(CAMBRIDGE,MASS.)         V. 112   845 2003\n",
      "REMARK     1  REFN                   ISSN 0092-8674\n",
      "REMARK     1  DOI    10.1016/S0092-8674(03)00191-0\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    3.42 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OVE.pdb A Results/activation_segments/unaligned/1OVE_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OVE.pdb with 3170 atoms.\n",
      "File Results/activation_segments/unaligned/1OVE_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             26-MAR-03   1OVE\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 14;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN-ACTIVATED PROTEIN KINASE P38, MAP KINASE P38,\n",
      "COMPND    5 CYTOKINE SUPPRESSIVE ANTI-INFLAMMATORY DRUG BINDING PROTEIN, CSAID\n",
      "COMPND    6 BINDING PROTEIN, CSBP, MAX-INTERACTING PROTEIN 2, MAP KINASE MXI2;\n",
      "COMPND    7 EC: 2.7.1.37;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     J.E.STELMACH,L.LIU,S.B.PATEL,J.V.PIVNICHNY,G.SCAPIN,S.SINGH,\n",
      "REMARK     1  AUTH 2 C.E.HOP,Z.WANG,P.M.CAMERON,E.A.NICHOLS,S.J.O'KEEFE,\n",
      "REMARK     1  AUTH 3 E.A.O'NEILL,D.M.SCHMATZ,C.D.SCHWARTZ,C.M.THOMPSON,\n",
      "REMARK     1  AUTH 4 D.M.ZALLER,J.B.DOHERTY\n",
      "REMARK     1  TITL   DESIGN AND SYNTHESIS OF POTENT, ORALLY BIOAVAILABLE\n",
      "REMARK     1  TITL 2 DIHYDROQUINAZOLINONE INHIBITORS OF P38 MAP KINASE\n",
      "\n",
      "File Results/activation_segments/unaligned/1OVE_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             26-MAR-03   1OVE\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 14;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN-ACTIVATED PROTEIN KINASE P38, MAP KINASE P38,\n",
      "COMPND    5 CYTOKINE SUPPRESSIVE ANTI-INFLAMMATORY DRUG BINDING PROTEIN, CSAID\n",
      "COMPND    6 BINDING PROTEIN, CSBP, MAX-INTERACTING PROTEIN 2, MAP KINASE MXI2;\n",
      "COMPND    7 EC: 2.7.1.37;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     J.E.STELMACH,L.LIU,S.B.PATEL,J.V.PIVNICHNY,G.SCAPIN,S.SINGH,\n",
      "REMARK     1  AUTH 2 C.E.HOP,Z.WANG,P.M.CAMERON,E.A.NICHOLS,S.J.O'KEEFE,\n",
      "REMARK     1  AUTH 3 E.A.O'NEILL,D.M.SCHMATZ,C.D.SCHWARTZ,C.M.THOMPSON,\n",
      "REMARK     1  AUTH 4 D.M.ZALLER,J.B.DOHERTY\n",
      "REMARK     1  TITL   DESIGN AND SYNTHESIS OF POTENT, ORALLY BIOAVAILABLE\n",
      "REMARK     1  TITL 2 DIHYDROQUINAZOLINONE INHIBITORS OF P38 MAP KINASE\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OZ1.pdb A Results/activation_segments/unaligned/1OZ1_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1OZ1.pdb with 3003 atoms.\n",
      "File Results/activation_segments/unaligned/1OZ1_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             07-APR-03   1OZ1\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 14;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN-ACTIVATED PROTEIN KINASE P38ALPHA, MAP KINASE\n",
      "COMPND    5 P38ALPHA, CYTOKINE SUPPRESSIVE ANTI-INFLAMMATORY DRUG BINDING\n",
      "COMPND    6 PROTEIN, CSAID BINDING PROTEIN, CSBP, MAX-INTERACTING PROTEIN 2, MAP\n",
      "COMPND    7 KINASE MXI2, SAPK2A;\n",
      "COMPND    8 EC: 2.7.1.37;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "\n",
      "File Results/activation_segments/unaligned/1OZ1_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             07-APR-03   1OZ1\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 14;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MITOGEN-ACTIVATED PROTEIN KINASE P38ALPHA, MAP KINASE\n",
      "COMPND    5 P38ALPHA, CYTOKINE SUPPRESSIVE ANTI-INFLAMMATORY DRUG BINDING\n",
      "COMPND    6 PROTEIN, CSAID BINDING PROTEIN, CSBP, MAX-INTERACTING PROTEIN 2, MAP\n",
      "COMPND    7 KINASE MXI2, SAPK2A;\n",
      "COMPND    8 EC: 2.7.1.37;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1P14.pdb A Results/activation_segments/unaligned/1P14_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1P14.pdb with 2421 atoms.\n",
      "File Results/activation_segments/unaligned/1P14_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             11-APR-03   1P14\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN, RESIDUES 978-1283;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1P14_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             11-APR-03   1P14\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN, RESIDUES 978-1283;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : NULL\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1P4O.pdb A Results/activation_segments/unaligned/1P4O_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1P4O.pdb with 5593 atoms.\n",
      "File Results/activation_segments/unaligned/1P4O_A.pdb before post_process, first few lines:\n",
      "HEADER    HORMONE/GROWTH FACTOR                   23-APR-03   1P4O\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN-LIKE GROWTH FACTOR I RECEPTOR PROTEIN;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: CD221 ANTIGEN, IGF1R;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "File Results/activation_segments/unaligned/1P4O_A.pdb after post_process, first few lines:\n",
      "HEADER    HORMONE/GROWTH FACTOR                   23-APR-03   1P4O\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN-LIKE GROWTH FACTOR I RECEPTOR PROTEIN;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: CD221 ANTIGEN, IGF1R;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.50 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PHK.pdb A Results/activation_segments/unaligned/1PHK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PHK.pdb with 2417 atoms.\n",
      "File Results/activation_segments/unaligned/1PHK_A.pdb before post_process, first few lines:\n",
      "HEADER    KINASE                                  15-MAR-96   1PHK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PHOSPHORYLASE KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: GAMMA SUBUNIT, TRUNCATED TO RESIDUES 1 - 298;\n",
      "COMPND    5 SYNONYM: RABBIT MUSCLE PHOSPHORYLASE KINASE;\n",
      "COMPND    6 EC: 2.7.1.38;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 OTHER_DETAILS: LIGANDS ATP AND MN(II)\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     D.J.OWEN,A.C.PAPAGEORGIOU,E.F.GARMAN,M.E.NOBLE,L.N.JOHNSON\n",
      "REMARK     1  TITL   EXPRESSION, PURIFICATION AND CRYSTALLISATION OF\n",
      "REMARK     1  TITL 2 PHOSPHORYLASE KINASE CATALYTIC DOMAIN\n",
      "REMARK     1  REF    J.MOL.BIOL.                   V. 246   374 1995\n",
      "REMARK     1  REFN                   ISSN 0022-2836\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "\n",
      "File Results/activation_segments/unaligned/1PHK_A.pdb after post_process, first few lines:\n",
      "HEADER    KINASE                                  15-MAR-96   1PHK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PHOSPHORYLASE KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: GAMMA SUBUNIT, TRUNCATED TO RESIDUES 1 - 298;\n",
      "COMPND    5 SYNONYM: RABBIT MUSCLE PHOSPHORYLASE KINASE;\n",
      "COMPND    6 EC: 2.7.1.38;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 OTHER_DETAILS: LIGANDS ATP AND MN(II)\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     D.J.OWEN,A.C.PAPAGEORGIOU,E.F.GARMAN,M.E.NOBLE,L.N.JOHNSON\n",
      "REMARK     1  TITL   EXPRESSION, PURIFICATION AND CRYSTALLISATION OF\n",
      "REMARK     1  TITL 2 PHOSPHORYLASE KINASE CATALYTIC DOMAIN\n",
      "REMARK     1  REF    J.MOL.BIOL.                   V. 246   374 1995\n",
      "REMARK     1  REFN                   ISSN 0022-2836\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PJK.pdb A Results/activation_segments/unaligned/1PJK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PJK.pdb with 2928 atoms.\n",
      "File Results/activation_segments/unaligned/1PJK_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             03-JUN-03   1PJK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE II, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: RESIDUE 2-335;\n",
      "COMPND    5 SYNONYM: CK II;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,I.ERMAKOVA,O.-G.ISSINGER\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF HUMAN PROTEIN KINASE CK2: INSIGHTS\n",
      "REMARK     INTO\n",
      "REMARK     1  TITL 2 BASIC PROPERTIES OF THE CK2 HOLOENZYME\n",
      "REMARK     1  REF    EMBO J.                       V.  20  5320 2001\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "REMARK     1  DOI    10.1093/EMBOJ/20.19.5320\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.50 ANGSTROMS.\n",
      "\n",
      "File Results/activation_segments/unaligned/1PJK_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             03-JUN-03   1PJK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CASEIN KINASE II, ALPHA CHAIN;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: RESIDUE 2-335;\n",
      "COMPND    5 SYNONYM: CK II;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   K.NIEFIND,B.GUERRA,I.ERMAKOVA,O.-G.ISSINGER\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF HUMAN PROTEIN KINASE CK2: INSIGHTS\n",
      "REMARK     INTO\n",
      "REMARK     1  TITL 2 BASIC PROPERTIES OF THE CK2 HOLOENZYME\n",
      "REMARK     1  REF    EMBO J.                       V.  20  5320 2001\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "REMARK     1  DOI    10.1093/EMBOJ/20.19.5320\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.50 ANGSTROMS.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PKG.pdb A Results/activation_segments/unaligned/1PKG_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PKG.pdb with 4619 atoms.\n",
      "File Results/activation_segments/unaligned/1PKG_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE ACTIVATOR                   05-JUN-03   1PKG\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: C-KIT PROTEIN;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.90\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 10.00\n",
      "\n",
      "File Results/activation_segments/unaligned/1PKG_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE ACTIVATOR                   05-JUN-03   1PKG\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: C-KIT PROTEIN;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.90\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 10.00\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PME.pdb A Results/activation_segments/unaligned/1PME_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PME.pdb with 2973 atoms.\n",
      "File Results/activation_segments/unaligned/1PME_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-JUN-98   1PME\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: ERK2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 EC: 2.7.1.-;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.8\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.00\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 6.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 2.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : NULL\n",
      "\n",
      "File Results/activation_segments/unaligned/1PME_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-JUN-98   1PME\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: ERK2;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 EC: 2.7.1.-;\n",
      "COMPND    5 ENGINEERED: YES;\n",
      "COMPND    6 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 3.8\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.00\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 6.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 2.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : NULL\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PMN.pdb A Results/activation_segments/unaligned/1PMN_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PMN.pdb with 3004 atoms.\n",
      "File Results/activation_segments/unaligned/1PMN_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             11-JUN-03   1PMN\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 10;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: STRESS-ACTIVATED PROTEIN KINASE JNK3, C-JUN N-TERMINAL\n",
      "COMPND    5 KINASE 3, MAP KINASE P49 3F12;\n",
      "COMPND    6 EC: 2.7.1.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1PMN_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             11-JUN-03   1PMN\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: MITOGEN-ACTIVATED PROTEIN KINASE 10;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: STRESS-ACTIVATED PROTEIN KINASE JNK3, C-JUN N-TERMINAL\n",
      "COMPND    5 KINASE 3, MAP KINASE P49 3F12;\n",
      "COMPND    6 EC: 2.7.1.-;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PY5.pdb A Results/activation_segments/unaligned/1PY5_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1PY5.pdb with 2590 atoms.\n",
      "File Results/activation_segments/unaligned/1PY5_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-JUL-03   1PY5\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TGF-BETA RECEPTOR TYPE I;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TRUNCATED KINASE DOMAIN, RESIDUES 175-500;\n",
      "COMPND    5 SYNONYM: TGFR-1, TGF-BETA TYPE I RECEPTOR, SERINE/THREONINE-PROTEIN\n",
      "COMPND    6 KINASE RECEPTOR R4, SKR4, ACTIVIN RECEPTOR-LIKE KINASE 5, ALK-5;\n",
      "COMPND    7 EC: 2.7.1.37;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "File Results/activation_segments/unaligned/1PY5_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-JUL-03   1PY5\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TGF-BETA RECEPTOR TYPE I;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TRUNCATED KINASE DOMAIN, RESIDUES 175-500;\n",
      "COMPND    5 SYNONYM: TGFR-1, TGF-BETA TYPE I RECEPTOR, SERINE/THREONINE-PROTEIN\n",
      "COMPND    6 KINASE RECEPTOR R4, SKR4, ACTIVIN RECEPTOR-LIKE KINASE 5, ALK-5;\n",
      "COMPND    7 EC: 2.7.1.37;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1Q24.pdb A Results/activation_segments/unaligned/1Q24_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1Q24.pdb with 3105 atoms.\n",
      "File Results/activation_segments/unaligned/1Q24_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       23-JUL-03   1Q24\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: I;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.BOSSEMEYER,R.A.ENGH,V.KINZEL,H.PONSTINGL,R.HUBER\n",
      "REMARK     1  TITL   PHOSPHOTRANSFERASE AND SUBSTRATE BINDING MECHANISM OF\n",
      "REMARK     THE\n",
      "\n",
      "File Results/activation_segments/unaligned/1Q24_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       23-JUL-03   1Q24\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: I;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.BOSSEMEYER,R.A.ENGH,V.KINZEL,H.PONSTINGL,R.HUBER\n",
      "REMARK     1  TITL   PHOSPHOTRANSFERASE AND SUBSTRATE BINDING MECHANISM OF\n",
      "REMARK     THE\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1Q61.pdb A Results/activation_segments/unaligned/1Q61_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1Q61.pdb with 3103 atoms.\n",
      "File Results/activation_segments/unaligned/1Q61_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       12-AUG-03   1Q61\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: I;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    13 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    14 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.BOSSEMEYER,R.A.ENGH,V.KINZEL,H.PONSTINGL,R.HUBER\n",
      "REMARK     1  TITL   PHOSPHOTRANSFERASE AND SUBSTRATE BINDING MECHANISM OF\n",
      "\n",
      "File Results/activation_segments/unaligned/1Q61_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       12-AUG-03   1Q61\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: I;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    13 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    14 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   D.BOSSEMEYER,R.A.ENGH,V.KINZEL,H.PONSTINGL,R.HUBER\n",
      "REMARK     1  TITL   PHOSPHOTRANSFERASE AND SUBSTRATE BINDING MECHANISM OF\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1Q8T.pdb A Results/activation_segments/unaligned/1Q8T_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1Q8T.pdb with 3244 atoms.\n",
      "File Results/activation_segments/unaligned/1Q8T_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       22-AUG-03   1Q8T\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: PKA C-ALPHA, PROTEIN KINASE A;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: B;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    13 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    14 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1Q8T_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       22-AUG-03   1Q8T\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: PKA C-ALPHA, PROTEIN KINASE A;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: B;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    13 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    14 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1Q8W.pdb A Results/activation_segments/unaligned/1Q8W_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1Q8W.pdb with 3053 atoms.\n",
      "File Results/activation_segments/unaligned/1Q8W_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       22-AUG-03   1Q8W\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: PKA C-ALPHA, PROTEIN KINASE A;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: B;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    13 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    14 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1Q8W_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       22-AUG-03   1Q8W\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: PKA C-ALPHA, PROTEIN KINASE A;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: B;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    13 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    14 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1QCF.pdb A Results/activation_segments/unaligned/1QCF_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1QCF.pdb with 3959 atoms.\n",
      "File Results/activation_segments/unaligned/1QCF_A.pdb before post_process, first few lines:\n",
      "HEADER    TYROSINE KINASE                         04-MAY-99   1QCF\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE HCK;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: SH3-SH2-KINASE-HIGH AFFINITY TAIL;\n",
      "COMPND    5 SYNONYM: HEMATOPOIETIC CELL KINASE,HEMOPOIETIC CELL KINASE,P59-\n",
      "COMPND    6 HCK/P60-HCK,P59HCK,P61HCK;\n",
      "COMPND    7 EC: 2.7.10.2;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1QCF_A.pdb after post_process, first few lines:\n",
      "HEADER    TYROSINE KINASE                         04-MAY-99   1QCF\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE HCK;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: SH3-SH2-KINASE-HIGH AFFINITY TAIL;\n",
      "COMPND    5 SYNONYM: HEMATOPOIETIC CELL KINASE,HEMOPOIETIC CELL KINASE,P59-\n",
      "COMPND    6 HCK/P60-HCK,P59HCK,P61HCK;\n",
      "COMPND    7 EC: 2.7.10.2;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1QL6.pdb A Results/activation_segments/unaligned/1QL6_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1QL6.pdb with 2412 atoms.\n",
      "File Results/activation_segments/unaligned/1QL6_A.pdb before post_process, first few lines:\n",
      "HEADER    KINASE (GLYCOGEN METABOLISM)            24-AUG-99   1QL6\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PHOSPHORYLASE KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: GAMMA SUBUNIT, TRUNCATED TO RESIDUES 1 - 298;\n",
      "COMPND    5 SYNONYM: RABBIT MUSCLE PHOSPHORYLASE KINASE;\n",
      "COMPND    6 EC: 2.7.1.38;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES;\n",
      "COMPND    9 OTHER_DETAILS: LIGANDS ATP AND MN(II)\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     E.D.LOWE,M.E.M.NOBLE,V.T.SKAMNAKI,N.G.OIKONOMAKOS,D.J.OWEN,\n",
      "REMARK     1  AUTH 2 L.N.JOHNSON\n",
      "REMARK     1  TITL   THE CRYSTAL STRUCTURE OF A PHOSPHORYLASE KINASE PEPTIDE\n",
      "REMARK     1  TITL 2 SUBSTRATE COMPLEX: KINASE SUBSTRATE RECOGNITION\n",
      "REMARK     1  REF    EMBO J.                       V.  16  6646 1997\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "\n",
      "File Results/activation_segments/unaligned/1QL6_A.pdb after post_process, first few lines:\n",
      "HEADER    KINASE (GLYCOGEN METABOLISM)            24-AUG-99   1QL6\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: PHOSPHORYLASE KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: GAMMA SUBUNIT, TRUNCATED TO RESIDUES 1 - 298;\n",
      "COMPND    5 SYNONYM: RABBIT MUSCLE PHOSPHORYLASE KINASE;\n",
      "COMPND    6 EC: 2.7.1.38;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES;\n",
      "COMPND    9 OTHER_DETAILS: LIGANDS ATP AND MN(II)\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH\n",
      "REMARK     E.D.LOWE,M.E.M.NOBLE,V.T.SKAMNAKI,N.G.OIKONOMAKOS,D.J.OWEN,\n",
      "REMARK     1  AUTH 2 L.N.JOHNSON\n",
      "REMARK     1  TITL   THE CRYSTAL STRUCTURE OF A PHOSPHORYLASE KINASE PEPTIDE\n",
      "REMARK     1  TITL 2 SUBSTRATE COMPLEX: KINASE SUBSTRATE RECOGNITION\n",
      "REMARK     1  REF    EMBO J.                       V.  16  6646 1997\n",
      "REMARK     1  REFN                   ISSN 0261-4189\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1QPC.pdb A Results/activation_segments/unaligned/1QPC_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1QPC.pdb with 2492 atoms.\n",
      "File Results/activation_segments/unaligned/1QPC_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             21-MAY-99   1QPC\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE LCK;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 SYNONYM: LEUKOCYTE C-TERMINAL SRC KINASE, LSK, LYMPHOCYTE CELL-\n",
      "COMPND    6 SPECIFIC PROTEIN-TYROSINE KINASE, PROTEIN YT16, PROTO-ONCOGENE LCK, T\n",
      "COMPND    7 CELL-SPECIFIC PROTEIN-TYROSINE KINASE, P56-LCK;\n",
      "COMPND    8 EC: 2.7.10.2;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.60\n",
      "\n",
      "File Results/activation_segments/unaligned/1QPC_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             21-MAY-99   1QPC\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE LCK;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 SYNONYM: LEUKOCYTE C-TERMINAL SRC KINASE, LSK, LYMPHOCYTE CELL-\n",
      "COMPND    6 SPECIFIC PROTEIN-TYROSINE KINASE, PROTEIN YT16, PROTO-ONCOGENE LCK, T\n",
      "COMPND    7 CELL-SPECIFIC PROTEIN-TYROSINE KINASE, P56-LCK;\n",
      "COMPND    8 EC: 2.7.10.2;\n",
      "COMPND    9 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.60\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1QPD.pdb A Results/activation_segments/unaligned/1QPD_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1QPD.pdb with 2378 atoms.\n",
      "File Results/activation_segments/unaligned/1QPD_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             24-MAY-99   1QPD\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: LCK KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 98.0\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.00\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 5.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 1.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : NULL\n",
      "\n",
      "File Results/activation_segments/unaligned/1QPD_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             24-MAY-99   1QPD\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: LCK KINASE;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 EC: 2.7.1.112;\n",
      "COMPND    6 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.00 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : X-PLOR 98.0\n",
      "REMARK     3   AUTHORS     : BRUNGER\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "REMARK     3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.00\n",
      "REMARK     3   RESOLUTION RANGE LOW  (ANGSTROMS) : 5.00\n",
      "REMARK     3   DATA CUTOFF            (SIGMA(F)) : 1.000\n",
      "REMARK     3   DATA CUTOFF HIGH         (ABS(F)) : NULL\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1R0P.pdb A Results/activation_segments/unaligned/1R0P_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1R0P.pdb with 2632 atoms.\n",
      "File Results/activation_segments/unaligned/1R0P_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-SEP-03   1R0P\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: HEPATOCYTE GROWTH FACTOR RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: MET PROTO-ONCOGENE TYROSINE KINASE, C-MET, HGF RECEPTOR,\n",
      "COMPND    6 HGF-SF RECEPTOR;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "\n",
      "File Results/activation_segments/unaligned/1R0P_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             22-SEP-03   1R0P\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: HEPATOCYTE GROWTH FACTOR RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TYROSINE KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: MET PROTO-ONCOGENE TYROSINE KINASE, C-MET, HGF RECEPTOR,\n",
      "COMPND    6 HGF-SF RECEPTOR;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES;\n",
      "COMPND    9 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1RDQ.pdb E Results/activation_segments/unaligned/1RDQ_E.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1RDQ.pdb with 3450 atoms.\n",
      "File Results/activation_segments/unaligned/1RDQ_E.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       05-NOV-03   1RDQ\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: I;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    13 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    14 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   MADHUSUDAN,P.AKAMINE,N.H.XUONG,S.S.TAYLOR\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF A TRANSITION STATE MIMIC OF THE\n",
      "\n",
      "File Results/activation_segments/unaligned/1RDQ_E.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/TRANSFERASE INHIBITOR       05-NOV-03   1RDQ\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: E;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES;\n",
      "COMPND    8 MOL_ID: 2;\n",
      "COMPND    9 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    10 CHAIN: I;\n",
      "COMPND    11 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    12 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    13 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    14 ENGINEERED: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   MADHUSUDAN,P.AKAMINE,N.H.XUONG,S.S.TAYLOR\n",
      "REMARK     1  TITL   CRYSTAL STRUCTURE OF A TRANSITION STATE MIMIC OF THE\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1RJB.pdb A Results/activation_segments/unaligned/1RJB_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1RJB.pdb with 2716 atoms.\n",
      "File Results/activation_segments/unaligned/1RJB_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             19-NOV-03   1RJB\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FL CYTOKINE RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 SYNONYM: TYROSINE-PROTEIN KINASE RECEPTOR FLT3, STEM CELL TYROSINE\n",
      "COMPND    6 KINASE 1, STK-1, CD135 ANTIGEN;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX 2000\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "\n",
      "File Results/activation_segments/unaligned/1RJB_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             19-NOV-03   1RJB\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: FL CYTOKINE RECEPTOR;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC DOMAIN;\n",
      "COMPND    5 SYNONYM: TYROSINE-PROTEIN KINASE RECEPTOR FLT3, STEM CELL TYROSINE\n",
      "COMPND    6 KINASE 1, STK-1, CD135 ANTIGEN;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.10 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX 2000\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1RQQ.pdb A Results/activation_segments/unaligned/1RQQ_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1RQQ.pdb with 6300 atoms.\n",
      "File Results/activation_segments/unaligned/1RQQ_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE/SIGNALING PROTEIN           06-DEC-03   1RQQ\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: IR;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES;\n",
      "COMPND    9 MOL_ID: 2;\n",
      "COMPND    10 MOLECULE: ADAPTOR PROTEIN APS;\n",
      "COMPND    11 CHAIN: C, D;\n",
      "COMPND    12 FRAGMENT: SH2 DOMAIN;\n",
      "COMPND    13 ENGINEERED: YES;\n",
      "COMPND    14 MOL_ID: 3;\n",
      "COMPND    15 MOLECULE: BISUBSTRATE INHIBITOR;\n",
      "COMPND    16 CHAIN: E, F;\n",
      "COMPND    17 ENGINEERED: YES\n",
      "REMARK     2\n",
      "\n",
      "File Results/activation_segments/unaligned/1RQQ_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE/SIGNALING PROTEIN           06-DEC-03   1RQQ\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: INSULIN RECEPTOR;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: IR;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES;\n",
      "COMPND    9 MOL_ID: 2;\n",
      "COMPND    10 MOLECULE: ADAPTOR PROTEIN APS;\n",
      "COMPND    11 CHAIN: C, D;\n",
      "COMPND    12 FRAGMENT: SH2 DOMAIN;\n",
      "COMPND    13 ENGINEERED: YES;\n",
      "COMPND    14 MOL_ID: 3;\n",
      "COMPND    15 MOLECULE: BISUBSTRATE INHIBITOR;\n",
      "COMPND    16 CHAIN: E, F;\n",
      "COMPND    17 ENGINEERED: YES\n",
      "REMARK     2\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1RW8.pdb A Results/activation_segments/unaligned/1RW8_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1RW8.pdb with 2545 atoms.\n",
      "File Results/activation_segments/unaligned/1RW8_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             16-DEC-03   1RW8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TGF-BETA RECEPTOR TYPE I;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TRUNCATED KINASE DOMAIN (RESIDUES 200-500);\n",
      "COMPND    5 SYNONYM: TGFR-1, TGF-BETA TYPE I RECEPTOR, SERINE/THREONINE-PROTEIN\n",
      "COMPND    6 KINASE RECEPTOR R4, SKR4, ACTIVIN RECEPTOR-LIKE KINASE 5, ALK-5;\n",
      "COMPND    7 EC: 2.7.1.37;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "File Results/activation_segments/unaligned/1RW8_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             16-DEC-03   1RW8\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TGF-BETA RECEPTOR TYPE I;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: TRUNCATED KINASE DOMAIN (RESIDUES 200-500);\n",
      "COMPND    5 SYNONYM: TGFR-1, TGF-BETA TYPE I RECEPTOR, SERINE/THREONINE-PROTEIN\n",
      "COMPND    6 KINASE RECEPTOR R4, SKR4, ACTIVIN RECEPTOR-LIKE KINASE 5, ALK-5;\n",
      "COMPND    7 EC: 2.7.1.37;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1S9I.pdb A Results/activation_segments/unaligned/1S9I_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1S9I.pdb with 4812 atoms.\n",
      "File Results/activation_segments/unaligned/1S9I_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             04-FEB-04   1S9I\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: DUAL SPECIFICITY MITOGEN-ACTIVATED PROTEIN KINASE KINASE 2;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: MAP KINASE KINASE 2, MAPKK 2, ERK ACTIVATOR KINASE 2,\n",
      "COMPND    5 MAPK/ERK KINASE 2, MEK2;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    3.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1S9I_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             04-FEB-04   1S9I\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: DUAL SPECIFICITY MITOGEN-ACTIVATED PROTEIN KINASE KINASE 2;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: MAP KINASE KINASE 2, MAPKK 2, ERK ACTIVATOR KINASE 2,\n",
      "COMPND    5 MAPK/ERK KINASE 2, MEK2;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    3.20 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1S9J.pdb A Results/activation_segments/unaligned/1S9J_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1S9J.pdb with 2429 atoms.\n",
      "File Results/activation_segments/unaligned/1S9J_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             04-FEB-04   1S9J\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: DUAL SPECIFICITY MITOGEN-ACTIVATED PROTEIN KINASE KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MAP KINASE KINASE 1, MAPKK 1, ERK ACTIVATOR KINASE 1,\n",
      "COMPND    5 MAPK/ERK KINASE 1, MEK1;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1S9J_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             04-FEB-04   1S9J\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: DUAL SPECIFICITY MITOGEN-ACTIVATED PROTEIN KINASE KINASE 1;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: MAP KINASE KINASE 1, MAPKK 1, ERK ACTIVATOR KINASE 1,\n",
      "COMPND    5 MAPK/ERK KINASE 1, MEK1;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.40 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SM2.pdb A Results/activation_segments/unaligned/1SM2_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SM2.pdb with 4212 atoms.\n",
      "File Results/activation_segments/unaligned/1SM2_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-MAR-04   1SM2\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE ITK/TSK;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: CATALYTIC KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: T-CELL-SPECIFIC KINASE, TYROSINE-PROTEIN KINASE LYK, KINASE\n",
      "COMPND    6 EMT;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "\n",
      "File Results/activation_segments/unaligned/1SM2_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             08-MAR-04   1SM2\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: TYROSINE-PROTEIN KINASE ITK/TSK;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 FRAGMENT: CATALYTIC KINASE DOMAIN;\n",
      "COMPND    5 SYNONYM: T-CELL-SPECIFIC KINASE, TYROSINE-PROTEIN KINASE LYK, KINASE\n",
      "COMPND    6 EMT;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNX\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN,ACCELRYS\n",
      "REMARK     3               : SOFTWARE INC.(BADGER,BERARD,KUMAR,SZALMA,\n",
      "REMARK     3               : YIP,DZAKULA)\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SMH.pdb A Results/activation_segments/unaligned/1SMH_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SMH.pdb with 3230 atoms.\n",
      "File Results/activation_segments/unaligned/1SMH_A.pdb before post_process, first few lines:\n",
      "HEADER    SIGNALING PROTEIN,TRANSFERASE/INHIBITOR 09-MAR-04   1SMH\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES;\n",
      "COMPND    9 MOL_ID: 2;\n",
      "COMPND    10 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    11 CHAIN: B;\n",
      "COMPND    12 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    13 SYNONYM: PKI-ALPHA; CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    14 MUSCLE/BRAIN ISOFORM; PROTEIN KINASE INHIBITOR ALPHA;\n",
      "COMPND    15 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.04 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1SMH_A.pdb after post_process, first few lines:\n",
      "HEADER    SIGNALING PROTEIN,TRANSFERASE/INHIBITOR 09-MAR-04   1SMH\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 FRAGMENT: CATALYTIC SUBUNIT;\n",
      "COMPND    5 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    6 EC: 2.7.1.37;\n",
      "COMPND    7 ENGINEERED: YES;\n",
      "COMPND    8 MUTATION: YES;\n",
      "COMPND    9 MOL_ID: 2;\n",
      "COMPND    10 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    11 CHAIN: B;\n",
      "COMPND    12 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    13 SYNONYM: PKI-ALPHA; CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    14 MUSCLE/BRAIN ISOFORM; PROTEIN KINASE INHIBITOR ALPHA;\n",
      "COMPND    15 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.04 ANGSTROMS.\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SVD.pdb A Results/activation_segments/unaligned/1SVD_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SVD.pdb with 4778 atoms.\n",
      "File Results/activation_segments/unaligned/1SVD_A.pdb before post_process, first few lines:\n",
      "HEADER    LYASE                                   29-MAR-04   1SVD\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RIBULOSE-1,5-BISPHOSPHATE CARBOXYLASE/OXYGENASE LARGE\n",
      "COMPND    3 SUBUNIT;\n",
      "COMPND    4 CHAIN: A;\n",
      "COMPND    5 EC: 4.1.1.39;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: RIBULOSE BISPHOSPHATE CARBOXYLASE SMALL CHAIN;\n",
      "COMPND    8 CHAIN: M;\n",
      "COMPND    9 SYNONYM: RUBISCO SMALL SUBUNIT;\n",
      "COMPND    10 EC: 4.1.1.39\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1SVD_A.pdb after post_process, first few lines:\n",
      "HEADER    LYASE                                   29-MAR-04   1SVD\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: RIBULOSE-1,5-BISPHOSPHATE CARBOXYLASE/OXYGENASE LARGE\n",
      "COMPND    3 SUBUNIT;\n",
      "COMPND    4 CHAIN: A;\n",
      "COMPND    5 EC: 4.1.1.39;\n",
      "COMPND    6 MOL_ID: 2;\n",
      "COMPND    7 MOLECULE: RIBULOSE BISPHOSPHATE CARBOXYLASE SMALL CHAIN;\n",
      "COMPND    8 CHAIN: M;\n",
      "COMPND    9 SYNONYM: RUBISCO SMALL SUBUNIT;\n",
      "COMPND    10 EC: 4.1.1.39\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SVH.pdb A Results/activation_segments/unaligned/1SVH_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SVH.pdb with 3048 atoms.\n",
      "File Results/activation_segments/unaligned/1SVH_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             29-MAR-04   1SVH\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA, PROTEIN KINASE A;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    9 CHAIN: B;\n",
      "COMPND    10 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    11 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    12 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "\n",
      "File Results/activation_segments/unaligned/1SVH_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             29-MAR-04   1SVH\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA, PROTEIN KINASE A;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MOL_ID: 2;\n",
      "COMPND    8 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE INHIBITOR, ALPHA FORM;\n",
      "COMPND    9 CHAIN: B;\n",
      "COMPND    10 FRAGMENT: RESIDUES 5-24;\n",
      "COMPND    11 SYNONYM: PKI-ALPHA, CAMP-DEPENDENT PROTEIN KINASE INHIBITOR,\n",
      "COMPND    12 MUSCLE/BRAIN ISOFORM;\n",
      "COMPND    13 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.30 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.24\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SYK.pdb A Results/activation_segments/unaligned/1SYK_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SYK.pdb with 5584 atoms.\n",
      "File Results/activation_segments/unaligned/1SYK_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             01-APR-04   1SYK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1SYK_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             01-APR-04   1SYK\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    2.80 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : CNS 1.0\n",
      "REMARK     3   AUTHORS     : BRUNGER,ADAMS,CLORE,DELANO,GROS,GROSSE-\n",
      "REMARK     3               : KUNSTLEVE,JIANG,KUSZEWSKI,NILGES,PANNU,\n",
      "REMARK     3               : READ,RICE,SIMONSON,WARREN\n",
      "REMARK     3\n",
      "REMARK     3  REFINEMENT TARGET : ENGH & HUBER\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SZM.pdb A Results/activation_segments/unaligned/1SZM_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1SZM.pdb with 5326 atoms.\n",
      "File Results/activation_segments/unaligned/1SZM_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-APR-04   1SZM\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA, PROTEIN KINASE A;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   L.PRADE,R.A.ENGH,A.GIROD,V.KINZEL,R.HUBER,D.BOSSEMEYER\n",
      "REMARK     1  TITL   STAUROSPORINE-INDUCED CONFORMATIONAL CHANGES OF\n",
      "REMARK     1  TITL 2 CAMP-DEPENDENT PROTEIN KINASE CATALYTIC SUBUNIT EXPLAIN\n",
      "REMARK     1  TITL 3 INHIBITORY POTENTIAL.\n",
      "REMARK     1  REF    STRUCTURE                     V.   5  1627 1997\n",
      "REMARK     1  REFN                   ISSN 0969-2126\n",
      "REMARK     1  PMID   9438863\n",
      "REMARK     1  DOI    10.1016/S0969-2126(97)00310-9\n",
      "REMARK     2\n",
      "\n",
      "File Results/activation_segments/unaligned/1SZM_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE                             06-APR-04   1SZM\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: CAMP-DEPENDENT PROTEIN KINASE, ALPHA-CATALYTIC SUBUNIT;\n",
      "COMPND    3 CHAIN: A, B;\n",
      "COMPND    4 SYNONYM: PKA C-ALPHA, PROTEIN KINASE A;\n",
      "COMPND    5 EC: 2.7.1.37;\n",
      "COMPND    6 ENGINEERED: YES;\n",
      "COMPND    7 MUTATION: YES\n",
      "REMARK     1\n",
      "REMARK     1 REFERENCE 1\n",
      "REMARK     1  AUTH   L.PRADE,R.A.ENGH,A.GIROD,V.KINZEL,R.HUBER,D.BOSSEMEYER\n",
      "REMARK     1  TITL   STAUROSPORINE-INDUCED CONFORMATIONAL CHANGES OF\n",
      "REMARK     1  TITL 2 CAMP-DEPENDENT PROTEIN KINASE CATALYTIC SUBUNIT EXPLAIN\n",
      "REMARK     1  TITL 3 INHIBITORY POTENTIAL.\n",
      "REMARK     1  REF    STRUCTURE                     V.   5  1627 1997\n",
      "REMARK     1  REFN                   ISSN 0969-2126\n",
      "REMARK     1  PMID   9438863\n",
      "REMARK     1  DOI    10.1016/S0969-2126(97)00310-9\n",
      "REMARK     2\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1T45.pdb A Results/activation_segments/unaligned/1T45_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1T45.pdb with 2817 atoms.\n",
      "File Results/activation_segments/unaligned/1T45_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE ACTIVATOR                   28-APR-04   1T45\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: HOMO SAPIENS V-KIT HARDY-ZUCKERMAN 4 FELINE SARCOMA VIRAL\n",
      "COMPND    3 ONCOGENE HOMOLOG;\n",
      "COMPND    4 CHAIN: A;\n",
      "COMPND    5 FRAGMENT: KIT TYROSINE KINASE;\n",
      "COMPND    6 SYNONYM: KIT;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "\n",
      "File Results/activation_segments/unaligned/1T45_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE ACTIVATOR                   28-APR-04   1T45\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: HOMO SAPIENS V-KIT HARDY-ZUCKERMAN 4 FELINE SARCOMA VIRAL\n",
      "COMPND    3 ONCOGENE HOMOLOG;\n",
      "COMPND    4 CHAIN: A;\n",
      "COMPND    5 FRAGMENT: KIT TYROSINE KINASE;\n",
      "COMPND    6 SYNONYM: KIT;\n",
      "COMPND    7 EC: 2.7.1.112;\n",
      "COMPND    8 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.90 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1T46.pdb A Results/activation_segments/unaligned/1T46_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1T46.pdb with 2696 atoms.\n",
      "File Results/activation_segments/unaligned/1T46_A.pdb before post_process, first few lines:\n",
      "HEADER    TRANSFERASE ACTIVATOR                   28-APR-04   1T46\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: HOMO SAPIENS V-KIT HARDY-ZUCKERMAN 4 FELINE SARCOMA VIRAL\n",
      "COMPND    3 ONCOGENE HOMOLOG;\n",
      "COMPND    4 CHAIN: A;\n",
      "COMPND    5 FRAGMENT: TYROSINE KINASE;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "File Results/activation_segments/unaligned/1T46_A.pdb after post_process, first few lines:\n",
      "HEADER    TRANSFERASE ACTIVATOR                   28-APR-04   1T46\n",
      "TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter\n",
      "COMPND    MOL_ID: 1;\n",
      "COMPND    2 MOLECULE: HOMO SAPIENS V-KIT HARDY-ZUCKERMAN 4 FELINE SARCOMA VIRAL\n",
      "COMPND    3 ONCOGENE HOMOLOG;\n",
      "COMPND    4 CHAIN: A;\n",
      "COMPND    5 FRAGMENT: TYROSINE KINASE;\n",
      "COMPND    6 EC: 2.7.1.112;\n",
      "COMPND    7 ENGINEERED: YES\n",
      "REMARK     2\n",
      "REMARK     2 RESOLUTION.    1.60 ANGSTROMS.\n",
      "REMARK     3\n",
      "REMARK     3 REFINEMENT.\n",
      "REMARK     3   PROGRAM     : REFMAC 5.1.19\n",
      "REMARK     3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,\n",
      "REMARK     3               : NICHOLLS,WINN,LONG,VAGIN\n",
      "REMARK     3\n",
      "REMARK     3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD\n",
      "REMARK     3\n",
      "REMARK     3  DATA USED IN REFINEMENT.\n",
      "\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1TVO.pdb A Results/activation_segments/unaligned/1TVO_A.pdb\n",
      "Loaded trajectory from /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/PDBs/1TVO.pdb with 3020 atoms.\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[8], line 111\u001b[0m\n\u001b[1;32m    109\u001b[0m chain \u001b[38;5;241m=\u001b[39m strip_to_chain(fp, chain_ID)\n\u001b[1;32m    110\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chain \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 111\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m mda\u001b[38;5;241m.\u001b[39mWriter(tp) \u001b[38;5;28;01mas\u001b[39;00m w:\n\u001b[1;32m    112\u001b[0m         w\u001b[38;5;241m.\u001b[39mwrite(chain)\n\u001b[1;32m    113\u001b[0m     post_process(tp)\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/coordinates/base.py:612\u001b[0m, in \u001b[0;36mIOBase.__exit__\u001b[0;34m(self, exc_type, exc_val, exc_tb)\u001b[0m\n\u001b[1;32m    610\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__exit__\u001b[39m(\u001b[38;5;28mself\u001b[39m, exc_type, exc_val, exc_tb):\n\u001b[1;32m    611\u001b[0m     \u001b[38;5;66;03m# see http://docs.python.org/2/library/stdtypes.html#typecontextmanager\u001b[39;00m\n\u001b[0;32m--> 612\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mclose\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    613\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;01mFalse\u001b[39;00m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/coordinates/PDB.py:704\u001b[0m, in \u001b[0;36mPDBWriter.close\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    702\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpdbfile\u001b[39m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpdbfile \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    703\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhas_END:\n\u001b[0;32m--> 704\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_write_pdb_bonds\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    705\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mEND()\n\u001b[1;32m    706\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/coordinates/PDB.py:843\u001b[0m, in \u001b[0;36mPDBWriter._write_pdb_bonds\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    838\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (\u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mobj\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;129;01mor\u001b[39;00m\n\u001b[1;32m    839\u001b[0m         \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj \u001b[38;5;129;01mor\u001b[39;00m\n\u001b[1;32m    840\u001b[0m         \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj\u001b[38;5;241m.\u001b[39muniverse, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbonds\u001b[39m\u001b[38;5;124m'\u001b[39m)):\n\u001b[1;32m    841\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m\n\u001b[0;32m--> 843\u001b[0m bondset \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mset\u001b[39m(\u001b[43mitertools\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mchain\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43ma\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbonds\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43ma\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43matoms\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m)\n\u001b[1;32m    844\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_reindex:\n\u001b[1;32m    845\u001b[0m     index_attribute \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mindex\u001b[39m\u001b[38;5;124m'\u001b[39m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/coordinates/PDB.py:843\u001b[0m, in \u001b[0;36m<genexpr>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    838\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (\u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mobj\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;129;01mor\u001b[39;00m\n\u001b[1;32m    839\u001b[0m         \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj \u001b[38;5;129;01mor\u001b[39;00m\n\u001b[1;32m    840\u001b[0m         \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj\u001b[38;5;241m.\u001b[39muniverse, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbonds\u001b[39m\u001b[38;5;124m'\u001b[39m)):\n\u001b[1;32m    841\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m\n\u001b[0;32m--> 843\u001b[0m bondset \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mset\u001b[39m(itertools\u001b[38;5;241m.\u001b[39mchain(\u001b[38;5;241m*\u001b[39m(\u001b[43ma\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbonds\u001b[49m \u001b[38;5;28;01mfor\u001b[39;00m a \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj\u001b[38;5;241m.\u001b[39matoms)))\n\u001b[1;32m    844\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_reindex:\n\u001b[1;32m    845\u001b[0m     index_attribute \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mindex\u001b[39m\u001b[38;5;124m'\u001b[39m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/core/groups.py:254\u001b[0m, in \u001b[0;36m_TopologyAttrContainer._add_prop.<locals>.getter\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    253\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mgetter\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m--> 254\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mattr\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__getitem__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/core/topologyattrs.py:469\u001b[0m, in \u001b[0;36mTopologyAttr.__getitem__\u001b[0;34m(self, group)\u001b[0m\n\u001b[1;32m    467\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Accepts an AtomGroup, ResidueGroup or SegmentGroup\"\"\"\u001b[39;00m\n\u001b[1;32m    468\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(group, (Atom, AtomGroup)):\n\u001b[0;32m--> 469\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_atoms\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgroup\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    470\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(group, (Residue, ResidueGroup)):\n\u001b[1;32m    471\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mget_residues(group)\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/MDAnalysis/core/topologyattrs.py:3061\u001b[0m, in \u001b[0;36m_Connection.get_atoms\u001b[0;34m(self, ag)\u001b[0m\n\u001b[1;32m   3059\u001b[0m unique_bonds \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39marray(\u001b[38;5;28msorted\u001b[39m(unique_bonds), dtype\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mobject\u001b[39m)\n\u001b[1;32m   3060\u001b[0m bond_idx, types, guessed, order \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mhsplit(unique_bonds, \u001b[38;5;241m4\u001b[39m)\n\u001b[0;32m-> 3061\u001b[0m bond_idx \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39marray(\u001b[43mbond_idx\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mravel\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mtolist(), dtype\u001b[38;5;241m=\u001b[39mnp\u001b[38;5;241m.\u001b[39mint32)\n\u001b[1;32m   3062\u001b[0m types \u001b[38;5;241m=\u001b[39m types\u001b[38;5;241m.\u001b[39mravel()\n\u001b[1;32m   3063\u001b[0m guessed \u001b[38;5;241m=\u001b[39m guessed\u001b[38;5;241m.\u001b[39mravel()\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import os\n",
    "from glob import glob\n",
    "import re\n",
    "import MDAnalysis as mda\n",
    "from time import time as t\n",
    "!ls\n",
    "def sglob(fp, absolute=True):\n",
    "    fps = sorted(glob(fp))\n",
    "    if absolute:\n",
    "        fps = [os.path.abspath(f) for f in fps]\n",
    "    return fps\n",
    "\n",
    "def strip_to_chain(pdb_file, chain_ID):\n",
    "    u = mda.Universe(pdb_file)\n",
    "    print(f\"Loaded trajectory from {pdb_file} with {len(u.atoms)} atoms.\")\n",
    "\n",
    "    chain = u.select_atoms(f\"protein and chainID {chain_ID}\")\n",
    "    if len(chain) == 0:\n",
    "        print(f\"Chain {chain_ID} not found in {pdb_file}.\")\n",
    "        return None\n",
    "    return chain\n",
    "\n",
    "def post_process(fname):\n",
    "    with open(fname, \"r\") as f_o:\n",
    "        initial_lines = f_o.readlines()\n",
    "\n",
    "    print(f\"File {fname} before post_process, first few lines:\")\n",
    "    print(\"\".join(initial_lines[:20]))\n",
    "\n",
    "    final_lines = initial_lines[-2:].copy()\n",
    "    no_ter = [line for line in initial_lines if line[:3] != \"TER\" or line in final_lines]\n",
    "\n",
    "    if len(no_ter) != len(initial_lines):\n",
    "        with open(fname, \"w\") as f_o:\n",
    "            print(f\"Rewriting {fname}, lines reduced from {len(initial_lines)} to {len(no_ter)}\")\n",
    "            f_o.write(\"\".join(no_ter))\n",
    "\n",
    "    print(f\"File {fname} after post_process, first few lines:\")\n",
    "    with open(fname, \"r\") as f_r:\n",
    "        print(\"\".join(f_r.readlines()[:20]))\n",
    "\n",
    "def parse_xml(xml_file):\n",
    "    hit_id = re.compile(r\"<Hit_id>(.*?)<.Hit_id>\")\n",
    "    with open(xml_file, \"r\") as f:\n",
    "        text = f.read()\n",
    "        results = [h.split(\"|\")[1:] for h in hit_id.findall(text)]\n",
    "        pdb_chain_dict = {}\n",
    "        for r in results:\n",
    "            pdb_chain_dict[r[0]+f\"_{r[1]}\"] = r[1]\n",
    "    return pdb_chain_dict\n",
    "\n",
    "def get_pdb_id(fp):\n",
    "    fp = fp.rsplit(\".\", 1)[0]\n",
    "    if \"/\" in fp:\n",
    "        fp = fp.rsplit(\"/\", 1)[1]\n",
    "    return fp\n",
    "\n",
    "def target_name(fp, target_dir, chain):\n",
    "    orig_path, file_name = fp.rsplit(\"/\", 1)\n",
    "    fp = fp.replace(orig_path, target_dir)\n",
    "    fp = fp.split(\".\")[0] + f\"_{chain}.pdb\"\n",
    "    return fp\n",
    "\n",
    "def find_pdb_file(PDB_chain_id, files):\n",
    "    print(files)\n",
    "    if \"_\" in PDB_chain_id:\n",
    "        PDB_id = PDB_chain_id.split(\"_\")[0]\n",
    "    assert len(PDB_id) == 4\n",
    "    for f in files:\n",
    "        filename = os.path.basename(f).split('.')[0]\n",
    "        if filename.startswith(PDB_id):\n",
    "            print(f\"Found file: {f} for PDB ID: {PDB_id}\")\n",
    "            return f\n",
    "\n",
    "print(\"BEGIN\")\n",
    "t1 = t()\n",
    "xml = \"braf-blast.xml\"\n",
    "pdb_dir = \"PDBs\"\n",
    "target_dir = \"Results/activation_segments/unaligned\"\n",
    "\n",
    "# Ensure the target directory exists\n",
    "os.makedirs(target_dir, exist_ok=True)\n",
    "\n",
    "xml_chain_dict = parse_xml(xml)\n",
    "print(xml_chain_dict)\n",
    "\n",
    "pdb_files = sorted(sglob(\"PDBs/*.pdb\"))\n",
    "print(pdb_files)\n",
    "keys = sorted([*xml_chain_dict.keys()], key=get_pdb_id)\n",
    "\n",
    "print(keys)\n",
    "files = [find_pdb_file(k, pdb_files) for k in keys]\n",
    "print(files)\n",
    "chain_IDs = [xml_chain_dict[k] for k in keys]\n",
    "print(len(chain_IDs))\n",
    "\n",
    "valid_file_chain_pairs = [(f, c) for f, c in zip(files, chain_IDs) if f is not None]\n",
    "print(len(valid_file_chain_pairs))\n",
    "new_file_paths = [target_name(f, target_dir, c) for f, c in valid_file_chain_pairs]\n",
    "print(new_file_paths)\n",
    "file_paths = [f for f, c in valid_file_chain_pairs]\n",
    "print(file_paths)\n",
    "print(len(file_paths))\n",
    "\n",
    "for (fp, chain_ID, tp) in zip(file_paths, chain_IDs, new_file_paths):\n",
    "    if fp is not None:\n",
    "        try:\n",
    "            print(fp, chain_ID, tp)\n",
    "            chain = strip_to_chain(fp, chain_ID)\n",
    "            if chain is not None:\n",
    "                with mda.Writer(tp) as w:\n",
    "                    w.write(chain)\n",
    "                post_process(tp)\n",
    "        except Exception as e:\n",
    "            print(f\"An error occurred while processing {fp} with chain {chain_ID}: {e}\")\n",
    "        \n",
    "t2 = t()\n",
    "t_t = round(t2 - t1, 3) // 60\n",
    "t_t = str((t_t // 60)) + \":\" + str(t_t % 60)\n",
    "print(f\"Time taken {t_t} for sequential processing\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "616153e7",
   "metadata": {},
   "source": [
    "## Counting again how many pdb files are in the directory after stripping the chains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "3b69d1ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 1060 PDB files in the directory 'Results/activation_segments/unaligned'.\n"
     ]
    }
   ],
   "source": [
    "import glob\n",
    "\n",
    "def count_pdb_files(directory):\n",
    "    # Ensure the directory path ends with a slash\n",
    "    directory = os.path.join(directory, '')\n",
    "\n",
    "    # Use glob to find all .pdb files in the directory\n",
    "    pdb_files = glob.glob(os.path.join(directory, '*.pdb'))\n",
    "\n",
    "    # Return the count of .pdb files\n",
    "    return len(pdb_files)\n",
    "\n",
    "# Specify the directory\n",
    "pdb_directory = 'Results/activation_segments/unaligned'\n",
    "\n",
    "# Get the count of PDB files\n",
    "pdb_count = count_pdb_files(pdb_directory)\n",
    "\n",
    "print(f\"There are {pdb_count} PDB files in the directory '{pdb_directory}'.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0e014c4b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   0%|                             | 2/1060 [00:00<01:04, 16.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1XH7_A.\n",
      "Aligned Sequences for 1XH7_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGFAKRVKGRTWGLCGTPEYLAPE\n",
      "ATOM Segment:   DFGFAKRVKGRTW-LCGTPEYLAPE\n",
      "\n",
      "\n",
      "Motifs not found in 3QTI_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   1%|                            | 7/1060 [00:00<00:34, 30.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 7N9G_A due to gap length: 13 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 1YWN_A.\n",
      "\n",
      "Aligned Sequences for 2F7E_E: (Max gap length: 0)\n",
      "SEQRES Segment: DFGFAKRVKGRTWTLCGTPEYLAPE\n",
      "ATOM Segment:   DFGFAKRVKGRTWTLCGTPEYLAPE\n",
      "\n",
      "\n",
      "Excluding 3OMV_A due to gap length: 12 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 2A19_B.\n",
      "\n",
      "Aligned Sequences for 5HHW_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGMTRDIYETDYYRKGGKGLLPVRWMAPE\n",
      "ATOM Segment:   DFGMTRDIYETDYYRKGGKGLLPVRWMAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   1%|                           | 11/1060 [00:00<00:49, 21.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 5FXQ_A.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1UA2_A.\n",
      "Aligned Sequences for 1UA2_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLAKSFGSPNRAYGHQVVTRWYRAPE\n",
      "ATOM Segment:   DFGLAKSFGSPNRAY-HQVVTRWYRAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 7M0L_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGISAQIGAELARRLEFIGTPYWMAPE\n",
      "ATOM Segment:   DFGISAQIGAELARRLEFIGTPYWMAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   1%|                           | 14/1060 [00:00<01:00, 17.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 8UDV_A due to gap length: 6 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 1OMW_A.\n",
      "\n",
      "Motifs not found in 1MUO_A.\n",
      "\n",
      "Motifs not found in 8BW9_C.\n",
      "\n",
      "Motifs not found in 4OT5_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   2%|                           | 21/1060 [00:01<00:49, 20.86it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 6ES0_A.\n",
      "\n",
      "Excluding 6EGD_A due to gap length: 10 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 6SDC_A.\n",
      "\n",
      "Aligned Sequences for 3TT0_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIHHIDYYKKTTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIHHIDYYKKTTNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Excluding 3OEZ_A due to gap length: 16 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 7RUN_A.\n",
      "\n",
      "Motifs not found in 6IN0_A.\n",
      "\n",
      "Aligned Sequences for 3LCO_A: (Max gap length: 4)\n",
      "SEQRES Segment: DFGLARDIMNDSNYIVKGNARLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIMNDSNYIV----RLPVKWMAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   3%|                           | 29/1060 [00:01<00:46, 22.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 4J99_A: (Max gap length: 2)\n",
      "SEQRES Segment: DFGLARDINNIDYYKTTTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIN--DYYKT--NGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 4FZA_B.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1XJD_A.\n",
      "Aligned Sequences for 1XJD_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGMCKENMLGDAKTNGFCGTPDYIAPE\n",
      "ATOM Segment:   DFGMCKENMLGDAKTN-FCGTPDYIAPE\n",
      "\n",
      "\n",
      "Excluding 5HVJ_A due to gap length: 18 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 2OGV_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIMNDSNYIVKGNARLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIMNDSNYIVKGNARLPVKWMAPE\n",
      "\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 4TRL_A.\n",
      "Aligned Sequences for 4TRL_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLSRVLEDDPEATGTTSGGKIPIRWTAPE\n",
      "ATOM Segment:   DFGLSRVLED-PEAT-TTSGGKIPIRWTAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   3%|                           | 32/1060 [00:01<00:48, 21.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 8FE9_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLMRALPQNDDHYVMQEHRKVPFAWCAPE\n",
      "ATOM Segment:   DFGLMRALPQNDDHYVMQEHRKVPFAWCAPE\n",
      "\n",
      "\n",
      "Motifs not found in 2GS7_A.\n",
      "\n",
      "Excluding 6JOI_A due to gap length: 15 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 1PJK_A.\n",
      "\n",
      "Motifs not found in 2CSN_A.\n",
      "\n",
      "Excluding 4TYE_A due to gap length: 7 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 5XYZ_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   4%|                           | 38/1060 [00:01<00:52, 19.63it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 8GAE_D due to gap length: 9 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 8UOH_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGFSNEFTVGGKLDTFCGSPPYAAPE\n",
      "ATOM Segment:   DFGFSNEFTVGGKLDTFCGSPPYAAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 2E9N_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLATVFRYNNRERLLNKMCGTLPYVAPE\n",
      "ATOM Segment:   DFGLATVFRYNNRERLLNKMCGTLPYVAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   4%|                          | 44/1060 [00:02<00:52, 19.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 6UAN_B.\n",
      "\n",
      "Motifs not found in 3QUP_A.\n",
      "\n",
      "Motifs not found in 4RIW_B.\n",
      "\n",
      "Motifs not found in 7FCZ_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   4%|                          | 47/1060 [00:02<00:56, 17.92it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 8PQ9_A due to gap length: 15 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 1ZMW_A.\n",
      "\n",
      "Excluding 3SVV_A due to gap length: 15 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 6JMF_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGMSREEADGVYAASGGLRQVPVKWTAPE\n",
      "ATOM Segment:   DFGMSREEADGVYAASGGLRQVPVKWTAPE\n",
      "\n",
      "\n",
      "Excluding 6EG9_A due to gap length: 11 or non-natural amino acid difference.\n",
      "\n",
      "Excluding 6P3D_A due to gap length: 9 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 1PKG_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIKNDSNYVVKGNARLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIKNDSNYVVKGNARLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 9D3F_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   5%|                          | 57/1060 [00:02<00:42, 23.51it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 6VQM_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLMRALPQNDDHYVMQEHRKVPFAWCAPE\n",
      "ATOM Segment:   DFGLMRALPQNDDHYVMQEHRKVPFAWCAPE\n",
      "\n",
      "\n",
      "Excluding 5KZ7_A due to gap length: 15 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 1AGW_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIHHIDYYKKTTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIHHIDYYKKTTNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 6CZ3_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARLIKEDVYLSHDHNIPYKWTAPE\n",
      "ATOM Segment:   DFGLARLIKEDVYLSHDHNIPYKWTAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 6CQD_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGISAQIGAELARRLEFIGTPYWMAPE\n",
      "ATOM Segment:   DFGISAQIGAELARRLEFIGTPYWMAPE\n",
      "\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1XH9_A.\n",
      "Aligned Sequences for 1XH9_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLAKRVKGRTWGLCGTPEYLAPE\n",
      "ATOM Segment:   DFGLAKRVKGRTW-LCGTPEYLAPE\n",
      "\n",
      "\n",
      "Motifs not found in 7UOS_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   6%|                          | 64/1060 [00:03<00:40, 24.62it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 2BDW_A.\n",
      "\n",
      "Aligned Sequences for 3G0E_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIKNDSNYVVKGNARLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIKNDSNYVVKGNARLPVKWMAPE\n",
      "\n",
      "\n",
      "Excluding 2J0M_B due to gap length: 16 or non-natural amino acid difference.\n",
      "\n",
      "Excluding 2J0K_A due to gap length: 25 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 6PP9_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLATVKSRWSGSHQFEQLSGSILWMAPE\n",
      "ATOM Segment:   DFGLATVKSRWSGSHQFEQLSGSILWMAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 1BMK_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARHTDDEMTGYVATRWYRAPE\n",
      "ATOM Segment:   DFGLARHTDDEMTGYVATRWYRAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   7%|                          | 72/1060 [00:03<00:35, 27.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 7UYR_A due to gap length: 6 or non-natural amino acid difference.\n",
      "\n",
      "Excluding 4MXY_A due to gap length: 11 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 2F4J_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLSRLMTGDTYTAPAGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLSRLMTGDTYTAPAGAKFPIKWTAPE\n",
      "\n",
      "\n",
      "Motifs not found in 5TO8_A.\n",
      "\n",
      "Motifs not found in 2JIT_A.\n",
      "\n",
      "Excluding 2BDF_A due to gap length: 11 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 1H4L_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   7%|                          | 75/1060 [00:03<00:41, 23.92it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 2BFX_A.\n",
      "\n",
      "Motifs not found in 1SVD_A.\n",
      "\n",
      "Motifs not found in 6CN9_A.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 6NFZ_A.\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 6NFZ_A.\n",
      "Aligned Sequences for 6NFZ_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGISAQIGAGLARRLGFIGTPYWMAPE\n",
      "ATOM Segment:   DFGISAQIGA-LARRL-FIGTPYWMAPE\n",
      "\n",
      "\n",
      "Excluding 5DI1_A due to gap length: 12 or non-natural amino acid difference.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   7%|                          | 78/1060 [00:03<00:48, 20.07it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 4AQC_A: (Max gap length: 2)\n",
      "SEQRES Segment: DFGLTKVLPQDKEXXKVKEPGESPIFWYAPE\n",
      "ATOM Segment:   DFGLTKVLPQDKE--KVKEPGESPIFWYAPE\n",
      "\n",
      "\n",
      "Motifs not found in 6TFP_A.\n",
      "\n",
      "Excluding 3PXK_A due to gap length: 17 or non-natural amino acid difference.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   8%|                         | 84/1060 [00:04<00:55, 17.72it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 7KP6_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLMRALPQNDDHYVMQEHRKVPFAWCAPE\n",
      "ATOM Segment:   DFGLMRALPQNDDHYVMQEHRKVPFAWCAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 1GJO_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDINNIDYYKKTTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDINNIDYYKKTTNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Excluding 6F3G_A due to gap length: 7 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 5HEZ_A: (Max gap length: 2)\n",
      "SEQRES Segment: DFGLTKVLPQDKEXXKVKEPGESPIFWYAPE\n",
      "ATOM Segment:   DFGLTKVLPQDKE--KVKEPGESPIFWYAPE\n",
      "\n",
      "\n",
      "Motifs not found in 1R0P_A.\n",
      "\n",
      "Excluding 1AQ1_A due to gap length: 13 or non-natural amino acid difference.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   8%|                         | 90/1060 [00:04<00:53, 18.00it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 8V1O_A: (Max gap length: 4)\n",
      "SEQRES Segment: DFGLARASEKFAQXVMXXRIVGTTAYMAPE\n",
      "ATOM Segment:   DFGLARASEK----VM--RIVGTTAYMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 4I24_A.\n",
      "\n",
      "Aligned Sequences for 5TQ3_A: (Max gap length: 2)\n",
      "SEQRES Segment: DFGLTKVLPQDKEXXKVKEPGESPIFWYAPE\n",
      "ATOM Segment:   DFGLTKVLPQDKE--KV--PGESPIFWYAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 4J97_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDINNIDYYKETTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDINNIDYYKETTNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 7AAX_A.\n",
      "\n",
      "Excluding 4UYA_A due to gap length: 14 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 4RJ4_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:   9%|                         | 95/1060 [00:04<00:47, 20.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 3C1X_A.\n",
      "\n",
      "Motifs not found in 4Q2A_A.\n",
      "\n",
      "Motifs not found in 7MN5_B.\n",
      "\n",
      "Excluding 6GTT_A due to gap length: 10 or non-natural amino acid difference.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  10%|                        | 101/1060 [00:05<00:53, 17.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 2WZJ_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGFSNEFTFGNKLDEFCGSPPYAAPE\n",
      "ATOM Segment:   DFGFSNEFTFGNKLDEFCGSPPYAAPE\n",
      "\n",
      "\n",
      "Motifs not found in 7OAM_A.\n",
      "\n",
      "Excluding 5SWH_A due to gap length: 18 or non-natural amino acid difference.\n",
      "\n",
      "Excluding 4H58_A due to gap length: 9 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 1VYW_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARAFGVPVRTYTHEVVTLWYRAPE\n",
      "ATOM Segment:   DFGLARAFGVPVRTYTHEVVTLWYRAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  10%|                        | 106/1060 [00:05<01:01, 15.50it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 4Y95_A.\n",
      "\n",
      "Aligned Sequences for 6MOB_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIKNDSNYVVKGNARLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIKNDSNYVVKGNARLPVKWMAPE\n",
      "\n",
      "\n",
      "Excluding 5HIE_A due to gap length: 9 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 2ESM_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  11%|                        | 113/1060 [00:05<00:44, 21.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 4WA9_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLSRLMTGDTYTAHAGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLSRLMTGDTYTAHAGAKFPIKWTAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 5TOZ_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLAKLLPLDKDYYVVREPGQSPIFWYAPE\n",
      "ATOM Segment:   DFGLAKLLPLDKDYYVVREPGQSPIFWYAPE\n",
      "\n",
      "\n",
      "Motifs not found in 2IVT_A.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 2GCD_A.\n",
      "Aligned Sequences for 2GCD_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGSASIMAPANGFVGTPYWMAPE\n",
      "ATOM Segment:   DFGSASIMAPAN-FVGTPYWMAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 2PY3_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDINNIDYYKKTTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDINNIDYYKKTTNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 6FEK_A.\n",
      "\n",
      "Aligned Sequences for 2I0V_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLARDIMNDSNYIVKGNARLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIMNDSNYIVKG-ARLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 6E4F_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  11%|                        | 116/1060 [00:05<00:46, 20.47it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 8G6Z_A.\n",
      "Aligned Sequences for 8G6Z_A: (Max gap length: 2)\n",
      "SEQRES Segment: DFGLTKVLPQDKEGYKVKEPGESPIFWYAPE\n",
      "ATOM Segment:   DFGLTKVLPQDKE-YKV--PGESPIFWYAPE\n",
      "\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 4XLV_A.\n",
      "Aligned Sequences for 4XLV_A: (Max gap length: 2)\n",
      "SEQRES Segment: DFGMTRDIGETDXXRKGGKGLLPVRWMAPE\n",
      "ATOM Segment:   DFGMTRDI-ETD--RKGGKGLLPVRWMAPE\n",
      "\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1GY3_A.\n",
      "Aligned Sequences for 1GY3_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLARAFGVPVRTYGHEVVTLWYRAPE\n",
      "ATOM Segment:   DFGLARAFGVPVRTY-HEVVTLWYRAPE\n",
      "\n",
      "\n",
      "Motifs not found in 4ZSE_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  12%|                        | 122/1060 [00:06<00:47, 19.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 5NUD_A due to gap length: 15 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 8W38_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARVINNIDYYKKTTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARVINNIDYYKKTTNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 2G2F_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLSRLMTGDTYTAPAGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLSRLMTGDTYTAPAGAKFPIKWTAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 7F3M_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARGVHHIDYYKKTSNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARGVHHIDYYKKTSNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 3OY3_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLSRLMTGDTYTAHAGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLSRLMTGDTYTAHAGAKFPIKWTAPE\n",
      "\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1ATP_E.\n",
      "Aligned Sequences for 1ATP_E: (Max gap length: 1)\n",
      "SEQRES Segment: DFGFAKRVKGRTWGLCGTPEYLAPE\n",
      "ATOM Segment:   DFGFAKRVKGRTW-LCGTPEYLAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  12%|                       | 130/1060 [00:06<00:38, 24.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 5HU9_A due to gap length: 5 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 3FE3_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGFSNEFTVGGKLDAFCGAPPYAAPE\n",
      "ATOM Segment:   DFGFSNEFTVGGKLDAFCGAPPYAAPE\n",
      "\n",
      "\n",
      "Motifs not found in 5DRB_A.\n",
      "\n",
      "Aligned Sequences for 8CGC_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIMNDSNYIVKGNARLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIMNDSNYIVKGNARLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 2WGJ_A.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1H1P_A.\n",
      "Aligned Sequences for 1H1P_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLARAFGVPVRTYGHEVVTLWYRAPE\n",
      "ATOM Segment:   DFGLARAFGVPVRTY-HEVVTLWYRAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  13%|                       | 136/1060 [00:06<00:37, 24.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 5I9U_A due to gap length: 13 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 8PQH_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIMHDSNYVSKGSTFLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIMHDSNYVSKGSTFLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 7M5Z_A.\n",
      "\n",
      "Aligned Sequences for 6F3F_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARLIEDNEYTARQGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLARLIEDNEYTARQGAKFPIKWTAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 4V0G_A: (Max gap length: 4)\n",
      "SEQRES Segment: DFGLAKLLPLDKDXXVVREPGQSPIFWYAPE\n",
      "ATOM Segment:   DFGLAKLLPLDKD--VVREP----IFWYAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 6N7A_A: (Max gap length: 2)\n",
      "SEQRES Segment: DFGLTKAIETDKEXXTVKDDRDSPVFWYAPE\n",
      "ATOM Segment:   DFGLTKAIETDKE--TVKDDRDSPVFWYAPE\n",
      "\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1QPD_A.\n",
      "Aligned Sequences for 1QPD_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLARLIEDAEGTAREGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLARLIEDAE-TAREGAKFPIKWTAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  13%|                       | 139/1060 [00:07<00:45, 20.47it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 8S9P_C.\n",
      "\n",
      "Aligned Sequences for 8X2A_A: (Max gap length: 3)\n",
      "SEQRES Segment: DFGMTRYVLDDQYVSSVGTKFPVKWSAPE\n",
      "ATOM Segment:   DFGMTRYVLDDQYV---GTKFPVKWSAPE\n",
      "\n",
      "\n",
      "Motifs not found in 5O1V_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  13%|                       | 142/1060 [00:07<00:58, 15.65it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 8VJB_A.\n",
      "\n",
      "Excluding 6E6E_A due to gap length: 6 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 4GT5_A.\n",
      "\n",
      "Motifs not found in 5CNN_A.\n",
      "\n",
      "Aligned Sequences for 1ZYS_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLATVFRYNNRERLLNKMCGTLPYVAPE\n",
      "ATOM Segment:   DFGLATVFRYNNRERLLNKMCGTLPYVAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  14%|                       | 151/1060 [00:07<00:43, 21.03it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 4JVG_A due to gap length: 11 or non-natural amino acid difference.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 6FD3_A.\n",
      "Aligned Sequences for 6FD3_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGFCAQITPEQSKRSGMVGTPYWMAPE\n",
      "ATOM Segment:   DFGFCAQITPEQSKRS-MVGTPYWMAPE\n",
      "\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1FQ1_B.\n",
      "Aligned Sequences for 1FQ1_B: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLARAFGVPVRTYGHEVVTLWYRAPE\n",
      "ATOM Segment:   DFGLARAFGVPVRTY-HEVVTLWYRAPE\n",
      "\n",
      "\n",
      "Excluding 1LEW_A due to gap length: 9 or non-natural amino acid difference.\n",
      "\n",
      "Excluding 4CQE_A due to gap length: 9 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 6M0U_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  15%|                       | 154/1060 [00:07<00:53, 16.89it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 3ZFX_A due to gap length: 12 or non-natural amino acid difference.\n",
      "\n",
      "Excluding 3IDP_A due to gap length: 9 or non-natural amino acid difference.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1BKX_A.\n",
      "Aligned Sequences for 1BKX_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGFAKRVKGRTWGLCGTPEYLAPE\n",
      "ATOM Segment:   DFGFAKRVKGRTW-LCGTPEYLAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 3WZD_A: (Max gap length: 0)\n",
      "SEQRES Segment: APEDLYKDFLTLEHLICYSFQVAKGMEFLASRKCIHRDLAARNILLSEKNVVKICDFG\n",
      "ATOM Segment:   APEDLYKDFLTLEHLICYSFQVAKGMEFLASRKCIHRDLAARNILLSEKNVVKICDFG\n",
      "\n",
      "\n",
      "Aligned Sequences for 4HZR_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLMRALPQNDDHYVMQEHRKVPFAWCAPE\n",
      "ATOM Segment:   DFGLMRALPQNDDHYVMQEHRKVPFAWCAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  15%|                      | 162/1060 [00:08<00:39, 22.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 4LGH_A.\n",
      "\n",
      "Motifs not found in 2QO7_A.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 5USY_A.\n",
      "Aligned Sequences for 5USY_A: (Max gap length: 3)\n",
      "SEQRES Segment: DFGLTKVLPQDKEGYKVKEPGESPIFWYAPE\n",
      "ATOM Segment:   DFGLTKVLPQDKE-YKVKE---SPIFWYAPE\n",
      "\n",
      "\n",
      "Excluding 1BYG_A due to gap length: 12 or non-natural amino acid difference.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 1CTP_E.\n",
      "Aligned Sequences for 1CTP_E: (Max gap length: 1)\n",
      "SEQRES Segment: DFGFAKRVKGRTWGLCGTPEYLAPE\n",
      "ATOM Segment:   DFGFAKRVKGRTW-LCGTPEYLAPE\n",
      "\n",
      "\n",
      "Excluding 6FNI_A due to gap length: 17 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 4QQT_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARGVHHIDYYKKTSNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARGVHHIDYYKKTSNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Excluding 6YI8_A due to gap length: 15 or non-natural amino acid difference.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  16%|                      | 169/1060 [00:08<00:35, 25.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 1MQB_A due to gap length: 17 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 4PMM_A.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 3CD3_A.\n",
      "Aligned Sequences for 3CD3_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGMSREEADGVGAASGGLRQVPVKWTAPE\n",
      "ATOM Segment:   DFGMSREEADGV-AASGGLRQVPVKWTAPE\n",
      "\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 2CJM_A.\n",
      "Aligned Sequences for 2CJM_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLARAFGVPVRTYGHEVVTLWYRAPE\n",
      "ATOM Segment:   DFGLARAFGVPVRTY-HEVVTLWYRAPE\n",
      "\n",
      "\n",
      "Motifs not found in 2GS2_A.\n",
      "\n",
      "Aligned Sequences for 3QRI_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLSRLMTGDTYTAHAGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLSRLMTGDTYTAHAGAKFPIKWTAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  17%|                      | 176/1060 [00:08<00:35, 25.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 5NG3_B.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 3D5W_A.\n",
      "Aligned Sequences for 3D5W_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLATKIEFDGERKKGLCGTPNYIAPE\n",
      "ATOM Segment:   DFGLATKIEFDGERKK-LCGTPNYIAPE\n",
      "\n",
      "\n",
      "Motifs not found in 4LI5_A.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 3PPZ_A.\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 3PPZ_A.\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 3PPZ_A.\n",
      "Aligned Sequences for 3PPZ_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLSRLKASGFLGSKGAAGTPEWMAPE\n",
      "ATOM Segment:   DFGLSRLKAS-FL-SK-AAGTPEWMAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 2PWL_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDINNIDYYKKTTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDINNIDYYKKTTNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 6XE4_A.\n",
      "\n",
      "Motifs not found in 1X8B_A.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  17%|                      | 182/1060 [00:09<00:37, 23.17it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 6THX_A due to gap length: 6 or non-natural amino acid difference.\n",
      "\n",
      "Excluding 4QQC_A due to gap length: 17 or non-natural amino acid difference.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 2OFU_A.\n",
      "Aligned Sequences for 2OFU_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLARLIEDNEGTAREGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLARLIEDNE-TAREGAKFPIKWTAPE\n",
      "\n",
      "\n",
      "Motifs not found in 8EYR_A.\n",
      "\n",
      "Aligned Sequences for 2XA4_A: (Max gap length: 3)\n",
      "SEQRES Segment: DFGLTKVLPQDKEXXKVKEPGESPIFWYAPE\n",
      "ATOM Segment:   DFGLTKVLPQDKE--KV-E---SPIFWYAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  17%|                      | 185/1060 [00:09<00:41, 21.13it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Motifs not found in 1FVR_A.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 3DK7_A.\n",
      "Aligned Sequences for 3DK7_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGLSRLMTGDTGTAHAGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLSRLMTGDT-TAHAGAKFPIKWTAPE\n",
      "\n",
      "\n",
      "Motifs not found in 1CKI_A.\n",
      "\n",
      "Excluding 3II5_A due to gap length: 8 or non-natural amino acid difference.\n",
      "\n",
      "Excluding 1I44_A due to gap length: 15 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 3DK3_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLSRLMTGDTFTAHAGAKFPIKWTAPE\n",
      "ATOM Segment:   DFGLSRLMTGDTFTAHAGAKFPIKWTAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  18%|                      | 192/1060 [00:09<00:36, 23.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 2YAC_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLATKVEYDGERKKTLCGTPNYIAPE\n",
      "ATOM Segment:   DFGLATKVEYDGERKKTLCGTPNYIAPE\n",
      "\n",
      "\n",
      "Motifs not found in 7MN5_A.\n",
      "\n",
      "Aligned Sequences for 6LVK_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDINNIDYYKKTTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDINNIDYYKKTTNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 3ET7_A.\n",
      "\n",
      "Aligned Sequences for 3Q32_A: (Max gap length: 2)\n",
      "SEQRES Segment: DFGLTKVLPQDKEXXKVKEPGESPIFWYAPE\n",
      "ATOM Segment:   DFGLTKVLPQDKE--KVKEPGESPIFWYAPE\n",
      "\n",
      "\n",
      "Excluding 1OIT_A due to gap length: 7 or non-natural amino acid difference.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  19%|                      | 200/1060 [00:09<00:32, 26.82it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 3OG7_A due to gap length: 9 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 5WR7_A.\n",
      "\n",
      "Substituting non-natural amino acid 'X' with 'G' in SEQRES for 2XIK_A.\n",
      "Aligned Sequences for 2XIK_A: (Max gap length: 1)\n",
      "SEQRES Segment: DFGVAGQLTDTQIKRNGFVGTPFWMAPE\n",
      "ATOM Segment:   DFGVAGQLTDTQIKRN-FVGTPFWMAPE\n",
      "\n",
      "\n",
      "Aligned Sequences for 2P2I_A: (Max gap length: 0)\n",
      "SEQRES Segment: APEDLYKDFLTLEHLICYSFQVAKGMEFLASRKCIHRDLAARNILLSEKNVVKICDFG\n",
      "ATOM Segment:   APEDLYKDFLTLEHLICYSFQVAKGMEFLASRKCIHRDLAARNILLSEKNVVKICDFG\n",
      "\n",
      "\n",
      "Motifs not found in 1ZMV_A.\n",
      "\n",
      "Motifs not found in 5CAV_A.\n",
      "\n",
      "Aligned Sequences for 3BEA_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIMNDSNYIVKGNARLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIMNDSNYIVKGNARLPVKWMAPE\n",
      "\n",
      "\n",
      "Excluding 7VJL_A due to gap length: 19 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 3GOP_A.\n",
      "\n",
      "Aligned Sequences for 4RIO_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLAKLLPLDKDYYVVREPGQSPIFWYAPE\n",
      "ATOM Segment:   DFGLAKLLPLDKDYYVVREPGQSPIFWYAPE\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  19%|                     | 205/1060 [00:10<00:35, 24.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned Sequences for 5FLF_A: (Max gap length: 0)\n",
      "SEQRES Segment: DFGLARDIHHIDYYKKTTNGRLPVKWMAPE\n",
      "ATOM Segment:   DFGLARDIHHIDYYKKTTNGRLPVKWMAPE\n",
      "\n",
      "\n",
      "Motifs not found in 6AUB_A.\n",
      "\n",
      "Excluding 4ZP5_A due to gap length: 12 or non-natural amino acid difference.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  20%|                     | 212/1060 [00:10<00:40, 21.17it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding 6MOM_A due to gap length: 5 or non-natural amino acid difference.\n",
      "\n",
      "Motifs not found in 5BPY_A.\n",
      "\n",
      "Excluding 4XV1_A due to gap length: 9 or non-natural amino acid difference.\n",
      "\n",
      "Aligned Sequences for 3ZMM_A: (Max gap length: 3)\n",
      "SEQRES Segment: DFGLTKVLPQDKEXXKVKEPGESPIFWYAPE\n",
      "ATOM Segment:   DFGLTKVLPQDKE--KV-E---SPIFWYAPE\n",
      "\n",
      "\n",
      "Motifs not found in 8CHF_E.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing PDB files:  20%|                     | 212/1060 [00:10<00:42, 19.99it/s]\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[10], line 180\u001b[0m\n\u001b[1;32m    177\u001b[0m         \u001b[38;5;28mprint\u001b[39m(count_msg)\n\u001b[1;32m    179\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;18m__name__\u001b[39m \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m__main__\u001b[39m\u001b[38;5;124m'\u001b[39m:\n\u001b[0;32m--> 180\u001b[0m     \u001b[43mmain\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn[10], line 115\u001b[0m, in \u001b[0;36mmain\u001b[0;34m()\u001b[0m\n\u001b[1;32m    112\u001b[0m     \u001b[38;5;28;01mcontinue\u001b[39;00m\n\u001b[1;32m    114\u001b[0m seqres_seqs \u001b[38;5;241m=\u001b[39m extract_seqres_sequence(full_pdb_path)\n\u001b[0;32m--> 115\u001b[0m atom_seq \u001b[38;5;241m=\u001b[39m \u001b[43mextract_atom_sequence\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfull_pdb_path\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mchain_id\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    117\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chain_id \u001b[38;5;129;01min\u001b[39;00m seqres_seqs \u001b[38;5;129;01mand\u001b[39;00m atom_seq:\n\u001b[1;32m    118\u001b[0m     seqres_sequence \u001b[38;5;241m=\u001b[39m seqres_seqs[chain_id]\n",
      "Cell \u001b[0;32mIn[10], line 54\u001b[0m, in \u001b[0;36mextract_atom_sequence\u001b[0;34m(pdb_file, chain_id)\u001b[0m\n\u001b[1;32m     52\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Extract sequence from atomic coordinates for a specific chain.\"\"\"\u001b[39;00m\n\u001b[1;32m     53\u001b[0m parser \u001b[38;5;241m=\u001b[39m PDBParser(QUIET\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m---> 54\u001b[0m structure \u001b[38;5;241m=\u001b[39m \u001b[43mparser\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_structure\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mPDB\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpdb_file\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     56\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m model \u001b[38;5;129;01min\u001b[39;00m structure:\n\u001b[1;32m     57\u001b[0m     chain \u001b[38;5;241m=\u001b[39m model[chain_id]\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/PDB/PDBParser.py:100\u001b[0m, in \u001b[0;36mPDBParser.get_structure\u001b[0;34m(self, id, file)\u001b[0m\n\u001b[1;32m     98\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m lines:\n\u001b[1;32m     99\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mEmpty file.\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m--> 100\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_parse\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlines\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    102\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstructure_builder\u001b[38;5;241m.\u001b[39mset_header(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mheader)\n\u001b[1;32m    103\u001b[0m \u001b[38;5;66;03m# Return the Structure instance\u001b[39;00m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/PDB/PDBParser.py:123\u001b[0m, in \u001b[0;36mPDBParser._parse\u001b[0;34m(self, header_coords_trailer)\u001b[0m\n\u001b[1;32m    121\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mheader, coords_trailer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_header(header_coords_trailer)\n\u001b[1;32m    122\u001b[0m \u001b[38;5;66;03m# Parse the atomic data; return the PDB file trailer\u001b[39;00m\n\u001b[0;32m--> 123\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrailer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_parse_coordinates\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcoords_trailer\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/PDB/PDBParser.py:299\u001b[0m, in \u001b[0;36mPDBParser._parse_coordinates\u001b[0;34m(self, coords_trailer)\u001b[0m\n\u001b[1;32m    296\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mis_pqr:\n\u001b[1;32m    297\u001b[0m     \u001b[38;5;66;03m# init atom with pdb fields\u001b[39;00m\n\u001b[1;32m    298\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 299\u001b[0m         \u001b[43mstructure_builder\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43minit_atom\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    300\u001b[0m \u001b[43m            \u001b[49m\u001b[43mname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    301\u001b[0m \u001b[43m            \u001b[49m\u001b[43mcoord\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    302\u001b[0m \u001b[43m            \u001b[49m\u001b[43mbfactor\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    303\u001b[0m \u001b[43m            \u001b[49m\u001b[43moccupancy\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    304\u001b[0m \u001b[43m            \u001b[49m\u001b[43maltloc\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    305\u001b[0m \u001b[43m            \u001b[49m\u001b[43mfullname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    306\u001b[0m \u001b[43m            \u001b[49m\u001b[43mserial_number\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    307\u001b[0m \u001b[43m            \u001b[49m\u001b[43melement\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    308\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    309\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m PDBConstructionException \u001b[38;5;28;01mas\u001b[39;00m message:\n\u001b[1;32m    310\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_handle_PDB_exception(message, global_line_counter)\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/PDB/StructureBuilder.py:289\u001b[0m, in \u001b[0;36mStructureBuilder.init_atom\u001b[0;34m(self, name, coord, b_factor, occupancy, altloc, fullname, serial_number, element, pqr_charge, radius, is_pqr)\u001b[0m\n\u001b[1;32m    286\u001b[0m         residue\u001b[38;5;241m.\u001b[39mflag_disordered()\n\u001b[1;32m    287\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    288\u001b[0m     \u001b[38;5;66;03m# The atom is not disordered\u001b[39;00m\n\u001b[0;32m--> 289\u001b[0m     \u001b[43mresidue\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43madd\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43matom\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/PDB/Residue.py:51\u001b[0m, in \u001b[0;36mResidue.add\u001b[0;34m(self, atom)\u001b[0m\n\u001b[1;32m     47\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhas_id(atom_id):\n\u001b[1;32m     48\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m PDBConstructionException(\n\u001b[1;32m     49\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mAtom \u001b[39m\u001b[38;5;132;01m{\u001b[39;00matom_id\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m defined twice in residue \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mself\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     50\u001b[0m     )\n\u001b[0;32m---> 51\u001b[0m \u001b[43mEntity\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43madd\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43matom\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/PDB/Entity.py:218\u001b[0m, in \u001b[0;36mEntity.add\u001b[0;34m(self, entity)\u001b[0m\n\u001b[1;32m    216\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhas_id(entity_id):\n\u001b[1;32m    217\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m PDBConstructionException(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mentity_id\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m defined twice\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m--> 218\u001b[0m \u001b[43mentity\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mset_parent\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m    219\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mchild_list\u001b[38;5;241m.\u001b[39mappend(entity)\n\u001b[1;32m    220\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mchild_dict[entity_id] \u001b[38;5;241m=\u001b[39m entity\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/PDB/Atom.py:344\u001b[0m, in \u001b[0;36mAtom.set_parent\u001b[0;34m(self, parent)\u001b[0m\n\u001b[1;32m    337\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Set the parent residue.\u001b[39;00m\n\u001b[1;32m    338\u001b[0m \n\u001b[1;32m    339\u001b[0m \u001b[38;5;124;03mArguments:\u001b[39;00m\n\u001b[1;32m    340\u001b[0m \u001b[38;5;124;03m - parent - Residue object\u001b[39;00m\n\u001b[1;32m    341\u001b[0m \n\u001b[1;32m    342\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    343\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mparent \u001b[38;5;241m=\u001b[39m parent\n\u001b[0;32m--> 344\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfull_id \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_full_id\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "#Extracting the full sequence from the pdb files, checking if there are any non-natural amino acids and substituting them and selecting only sequences with a maximum gap length of 4 amino acids to be reconstructed\n",
    "from Bio.PDB import PDBParser, PPBuilder\n",
    "from Bio.SeqUtils import seq1\n",
    "from glob import glob\n",
    "from tqdm import tqdm\n",
    "import os\n",
    "\n",
    "def substitute_non_natural_amino_acid(residue, aligned_atom, index):\n",
    "    \"\"\"Substitute non-natural amino acids with their natural counterparts.\"\"\"\n",
    "    substitutions = {\n",
    "        'X': 'G',  # Glycine\n",
    "        'B': 'N',  # Asparagine\n",
    "        'Z': 'Q',  # Glutamine\n",
    "        'J': 'L'   # Leucine\n",
    "    }\n",
    "    \n",
    "    if residue == 'X':\n",
    "        # Check if 'X' is surrounded by missing amino acids in aligned_atom\n",
    "        if index > 0 and aligned_atom[index - 1] == '-':\n",
    "            return residue\n",
    "        if index < len(aligned_atom) - 1 and aligned_atom[index + 1] == '-':\n",
    "            return residue\n",
    "\n",
    "    return substitutions.get(residue, residue)\n",
    "\n",
    "def extract_seqres_sequence(pdb_file):\n",
    "    \"\"\"Extract SEQRES sequences for each chain from a PDB file.\"\"\"\n",
    "    seq_dict = {}\n",
    "    with open(pdb_file, \"r\") as file:\n",
    "        lines = file.readlines()\n",
    "\n",
    "    current_chain = None\n",
    "    current_seq = []\n",
    "\n",
    "    for line in lines:\n",
    "        if line.startswith(\"SEQRES\"):\n",
    "            parts = line.split()\n",
    "            chain_id = parts[2]\n",
    "            if chain_id != current_chain:\n",
    "                if current_chain is not None:\n",
    "                    seq_dict[current_chain] = ''.join(seq1(residue) for residue in current_seq)\n",
    "                current_chain = chain_id\n",
    "                current_seq = []\n",
    "            current_seq.extend(parts[4:])\n",
    "\n",
    "    if current_chain is not None:\n",
    "        seq_dict[current_chain] = ''.join(seq1(residue) for residue in current_seq)\n",
    "\n",
    "    return seq_dict\n",
    "\n",
    "def extract_atom_sequence(pdb_file, chain_id):\n",
    "    \"\"\"Extract sequence from atomic coordinates for a specific chain.\"\"\"\n",
    "    parser = PDBParser(QUIET=True)\n",
    "    structure = parser.get_structure('PDB', pdb_file)\n",
    "    \n",
    "    for model in structure:\n",
    "        chain = model[chain_id]\n",
    "        \n",
    "        ppb = PPBuilder()\n",
    "        sequence = ''\n",
    "        for pp in ppb.build_peptides(chain):\n",
    "            sequence += pp.get_sequence()\n",
    "        return sequence\n",
    "    return None\n",
    "\n",
    "def find_motif_indices(sequence, motif):\n",
    "    \"\"\"Find the start index of a motif in a sequence.\"\"\"\n",
    "    index = sequence.find(motif)\n",
    "    return index if index != -1 else None\n",
    "\n",
    "def align_and_highlight_gaps(seqres_segment, atom_segment):\n",
    "    \"\"\"Align SEQRES and ATOM segments and highlight gaps in ATOM.\"\"\"\n",
    "    aligned_seqres = ''\n",
    "    aligned_atom = ''\n",
    "    atom_index = 0\n",
    "    max_gap_length = 0\n",
    "    current_gap_length = 0\n",
    "\n",
    "    for res_seqres in seqres_segment:\n",
    "        if atom_index < len(atom_segment) and res_seqres == atom_segment[atom_index]:\n",
    "            aligned_seqres += res_seqres\n",
    "            aligned_atom += atom_segment[atom_index]\n",
    "            atom_index += 1\n",
    "            current_gap_length = 0\n",
    "        else:\n",
    "            aligned_seqres += res_seqres\n",
    "            aligned_atom += '-'\n",
    "            current_gap_length += 1\n",
    "            max_gap_length = max(max_gap_length, current_gap_length)\n",
    "\n",
    "    return aligned_seqres, aligned_atom, max_gap_length\n",
    "\n",
    "def main():\n",
    "    target_dir = \"Results/activation_segments/unaligned\"\n",
    "    pdb_dir = \"PDBs\"\n",
    "    fasta_output_file = \"seqres_sequences.fasta\"  # File to store full sequences\n",
    "    text_output_file = \"seqres_info.txt\"\n",
    "    aligned_sequences = {}\n",
    "    satisfying_structures_count = 0\n",
    "\n",
    "    pdb_files = glob(os.path.join(target_dir, \"*.pdb\"))\n",
    "\n",
    "    with open(text_output_file, \"w\") as text_output, open(fasta_output_file, \"w\") as fasta_output:\n",
    "        for pdb_file in tqdm(pdb_files, desc=\"Processing PDB files\"):\n",
    "            pdb_name = os.path.basename(pdb_file)\n",
    "            pdb_id, chain_id_file = os.path.splitext(pdb_name)[0].split('_')\n",
    "            chain_id = chain_id_file\n",
    "\n",
    "            full_pdb_path = os.path.join(pdb_dir, pdb_id + '.pdb')\n",
    "            if not os.path.isfile(full_pdb_path):\n",
    "                print(f\"Corresponding full PDB for {pdb_id} not found.\")\n",
    "                continue\n",
    "\n",
    "            seqres_seqs = extract_seqres_sequence(full_pdb_path)\n",
    "            atom_seq = extract_atom_sequence(full_pdb_path, chain_id)\n",
    "\n",
    "            if chain_id in seqres_seqs and atom_seq:\n",
    "                seqres_sequence = seqres_seqs[chain_id]\n",
    "                seqres_dfg_index = find_motif_indices(seqres_sequence, 'DFG')\n",
    "                seqres_ape_index = find_motif_indices(seqres_sequence, 'APE')\n",
    "                atom_dfg_index = find_motif_indices(atom_seq, 'DFG')\n",
    "                atom_ape_index = find_motif_indices(atom_seq, 'APE')\n",
    "\n",
    "                # Determine the start and end indices for the segments\n",
    "                if None not in [seqres_dfg_index, seqres_ape_index, atom_dfg_index, atom_ape_index]:\n",
    "                    seqres_start = min(seqres_dfg_index, seqres_ape_index)\n",
    "                    seqres_end = max(seqres_dfg_index + 3, seqres_ape_index + 3)\n",
    "                    atom_start = min(atom_dfg_index, atom_ape_index)\n",
    "                    atom_end = max(atom_dfg_index + 3, atom_ape_index + 3)\n",
    "\n",
    "                    seqres_segment = seqres_sequence[seqres_start:seqres_end]\n",
    "                    atom_segment = atom_seq[atom_start:atom_end]\n",
    "\n",
    "                    aligned_seqres, aligned_atom, max_gap_length = align_and_highlight_gaps(seqres_segment, atom_segment)\n",
    "                    \n",
    "                    # Check for differences and substitute non-natural amino acids\n",
    "                    exclude_due_to_non_natural_diff = False\n",
    "                    corrected_seqres = ''\n",
    "                    for index, (res_seqres, res_atom) in enumerate(zip(aligned_seqres, aligned_atom)):\n",
    "                        if res_seqres != res_atom:\n",
    "                            corrected_residue = substitute_non_natural_amino_acid(res_seqres, aligned_atom, index)\n",
    "                            corrected_seqres += corrected_residue\n",
    "                            if corrected_residue != res_seqres:\n",
    "                                print(f\"Substituting non-natural amino acid '{res_seqres}' with '{corrected_residue}' in SEQRES for {pdb_id}_{chain_id}.\")\n",
    "                        else:\n",
    "                            corrected_seqres += res_seqres\n",
    "\n",
    "                    if not exclude_due_to_non_natural_diff and max_gap_length <= 4:\n",
    "                        satisfying_structures_count += 1\n",
    "                        info = (f\"Aligned Sequences for {pdb_id}_{chain_id}: (Max gap length: {max_gap_length})\\n\"\n",
    "                                f\"SEQRES Segment: {corrected_seqres}\\n\"\n",
    "                                f\"ATOM Segment:   {aligned_atom}\\n\\n\")\n",
    "                        print(info)\n",
    "                        text_output.write(info)\n",
    "                        \n",
    "                        aligned_sequences[f\"{pdb_id}_{chain_id}_SEQRES\"] = corrected_seqres\n",
    "                        aligned_sequences[f\"{pdb_id}_{chain_id}_ATOM\"] = aligned_atom\n",
    "\n",
    "                        # Write full SEQRES and ATOM sequences to the FASTA file\n",
    "                        fasta_output.write(f\">{pdb_id}_{chain_id}_SEQRES\\n{seqres_sequence}\\n\")\n",
    "                        fasta_output.write(f\">{pdb_id}_{chain_id}_ATOM\\n{atom_seq}\\n\")\n",
    "                    else:\n",
    "                        exclusion_msg = f\"Excluding {pdb_id}_{chain_id} due to gap length: {max_gap_length} or non-natural amino acid difference.\\n\"\n",
    "                        print(exclusion_msg)\n",
    "                        text_output.write(exclusion_msg)\n",
    "                else:\n",
    "                    motif_msg = f\"Motifs not found in {pdb_id}_{chain_id}.\\n\"\n",
    "                    print(motif_msg)\n",
    "                    text_output.write(motif_msg)\n",
    "            else:\n",
    "                chain_msg = f\"Chain {chain_id} not found in SEQRES of {pdb_id} or no atomic sequence available.\\n\"\n",
    "                print(chain_msg)\n",
    "                text_output.write(chain_msg)\n",
    "                \n",
    "        count_msg = f\"Total structures satisfying the condition: {satisfying_structures_count}\"\n",
    "        text_output.write(count_msg)\n",
    "        print(count_msg)\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f3c5b45f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of PDB IDs: 86\n"
     ]
    }
   ],
   "source": [
    "#COunting how many seqences are in the fasta file\n",
    "def count_total_pdb_ids(file_path):\n",
    "    total_pdb_ids = 0\n",
    "\n",
    "    with open(file_path, 'r') as file:\n",
    "        for line in file:\n",
    "            if line.startswith('>'):\n",
    "                total_pdb_ids += 1\n",
    "\n",
    "    print(f\"Total number of PDB IDs: {int(total_pdb_ids/2)}\") #here we divide by 2 because we have two lines per PDB ID\n",
    "\n",
    "# Provide the path to your seqres_sequence.fasta file\n",
    "file_path = \"seqres_sequences.fasta\"\n",
    "count_total_pdb_ids(file_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "844a44c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing 1XH7_A\n",
      "Reconstructing full sequence for 1XH7_A using MODELLER\n",
      "\n",
      "                         MODELLER 10.4, 2022/10/28, r12463\n",
      "\n",
      "     PROTEIN STRUCTURE MODELLING BY SATISFACTION OF SPATIAL RESTRAINTS\n",
      "\n",
      "\n",
      "                     Copyright(c) 1989-2022 Andrej Sali\n",
      "                            All Rights Reserved\n",
      "\n",
      "                             Written by A. Sali\n",
      "                               with help from\n",
      "              B. Webb, M.S. Madhusudhan, M-Y. Shen, G.Q. Dong,\n",
      "          M.A. Marti-Renom, N. Eswar, F. Alber, M. Topf, B. Oliva,\n",
      "             A. Fiser, R. Sanchez, B. Yerkovich, A. Badretdinov,\n",
      "                     F. Melo, J.P. Overington, E. Feyfant\n",
      "                 University of California, San Francisco, USA\n",
      "                    Rockefeller University, New York, USA\n",
      "                      Harvard University, Cambridge, USA\n",
      "                   Imperial Cancer Research Fund, London, UK\n",
      "              Birkbeck College, University of London, London, UK\n",
      "\n",
      "\n",
      "Kind, OS, HostName, Kernel, Processor: 4, Linux euporie 6.8.0-51-generic x86_64\n",
      "Date and time of compilation         : 2022/10/28 20:54:52\n",
      "MODELLER executable type             : x86_64-intel8\n",
      "Job starting time (YY/MM/DD HH:MM:SS): 2025/02/26 08:44:03\n",
      "\n",
      "environ____W> The class 'environ' is deprecated; use 'Environ' instead\n",
      "alignment__W> The class 'alignment' is deprecated; use 'Alignment' instead\n",
      "model______W> The class 'model' is deprecated; use 'Model' instead\n",
      "readlinef__W> File: Results/activation_segments/unaligned/1XH7_A.pdb, Line: 13\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "mkapsa__637W> No residue topology library is in memory.\n",
      "              Better radii would be used if topology.read() is called first.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  N -->  N\n",
      "              This message is written only for the first such atom.\n",
      "\n",
      "Pairwise dynamic programming alignment (ALIGN2D):\n",
      "  Residue-residue metric   : $(LIB)/as1.sim.mat\n",
      "  Diagonal                 :          100\n",
      "  Overhang                 :            0\n",
      "  Maximal gap length       :           50\n",
      "  Local alignment          :            F\n",
      "  MATRIX_OFFSET (local aln):       0.0000\n",
      "  FIX_OFFSETS              :      0.0    -1.0    -2.0    -3.0    -4.0\n",
      "  N_SUBOPT                 :            0\n",
      "  SUBOPT_OFFSET            :       0.0000\n",
      "  Alignment block          :            1\n",
      "  Gap introduction penalty :    -100.0000\n",
      "  Gap extension penalty    :       0.0000\n",
      "  Gap diagonal penalty     :       0.0000\n",
      "  Structure gap penalties  :    3.500   3.500   3.500   0.200   4.000   6.500   2.000   0.000\n",
      "  Break-break bonus        :   10000.0000\n",
      "  Length of alignment      :          350\n",
      "  Score                    :  296822.7812\n",
      "automodel__W> The class 'automodel' is deprecated; use 'AutoModel' instead\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    37099    34569\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      350\n",
      "Number of all, selected real atoms                :     2856    2856\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    34569   34569\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     5767\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1726.4313\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    2929       0      0   0.005   0.005      21.280       1.000\n",
      " 2 Bond angle potential               :    3947       0      8   1.902   1.902      279.26       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1908       0     52  45.912  45.912      623.26       1.000\n",
      " 4 Stereochemical improper torsion pot:    1240       0      2   1.203   1.203      38.031       1.000\n",
      " 5 Soft-sphere overlap restraints     :    5767       0      0   0.001   0.001     0.64369       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    6975       0      0   0.092   0.092      65.094       1.000\n",
      "10 Distance restraints 2 (N-O)        :    6681       0      0   0.148   0.148      133.14       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     345       0      1   3.790   3.790      58.437       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     301       0      1  50.791  50.791      16.252       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     253       0      2  60.425  60.425      63.946       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     111       0      0  60.796  60.796      70.977       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      49       0      0  92.847  92.847      31.822       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    5646       0      0   0.366   0.366      119.62       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     344      10     22  21.331  35.159     -8.6939       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    3840       0      0   0.646   0.646      213.37       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: target.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   28676.2129\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  10025   1G   2N C   N       3    5  -77.38  -71.20   27.38    1.59  -63.20  150.10   19.56\n",
      "    1          2N   2N N   CA      5    6  169.47  142.80                  -41.10\n",
      "    2  10026   2N   3A C   N      11   13  -60.18  -68.20    9.60    0.94  -62.50  168.55   27.51\n",
      "    2          3A   3A N   CA     13   14  150.57  145.30                  -40.90\n",
      "    3  10028   4A   5A C   N      21   23 -136.26 -134.00    2.43    0.06  -62.50 -173.56   34.51\n",
      "    3          5A   5A N   CA     23   24  147.88  147.00                  -40.90\n",
      "    4  10029   5A   6A C   N      26   28  -72.70  -68.20    6.80    0.43  -62.50  169.00   28.21\n",
      "    4          6A   6A N   CA     28   29  150.41  145.30                  -40.90\n",
      "    5  10030   6A   7K C   N      31   33  -74.86  -70.20   11.51    0.73  -62.90  168.70   22.58\n",
      "    5          7K   7K N   CA     33   34  150.92  140.40                  -40.80\n",
      "    6  10031   7K   8K C   N      40   42  -68.95  -70.20    1.57    0.13  -62.90  177.96   23.45\n",
      "    6          8K   8K N   CA     42   43  141.34  140.40                  -40.80\n",
      "    7  10032   8K   9G C   N      49   51 -152.02 -167.20   15.49    0.24   82.20 -154.09   15.31\n",
      "    7          9G   9G N   CA     51   52  171.53  174.60                    8.50\n",
      "    8  10033  10X  11E C   N      57   59  -74.55  -69.30   18.95    1.61  -63.60  164.96   21.41\n",
      "    8         11E  11E N   CA     59   60  124.30  142.50                  -40.30\n",
      "    9  10187 165R 166D C   N    1346 1348 -106.30  -63.30   64.60    7.15  -63.30   64.60    7.15\n",
      "    9        166D 166D N   CA   1348 1349    8.21  -40.00                  -40.00\n",
      "   10  10303 282G 283N C   N    2291 2293 -110.34  -63.20   86.75    9.33  -63.20   86.75    9.33\n",
      "   10        283N 283N N   CA   2293 2294   31.73  -41.10                  -41.10\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0   10   29   82  158  201  225  312  396  471  491\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "target.B99990001.pdb          1726.43127\n",
      "\n",
      "Reconstruction completed for 1XH7_A. File saved at Results/activation_segments/reconstructedModeller/1XH7_A_filled.pdb\n",
      "Removed REMARK lines from Results/activation_segments/reconstructedModeller/1XH7_A_filled.pdb\n",
      "Processing 2F7E_E\n",
      "No differences found for 2F7E_E. Original PDB copied to target directory.\n",
      "Processing 5HHW_A\n",
      "No differences found for 5HHW_A. Original PDB copied to target directory.\n",
      "Processing 1UA2_A\n",
      "Reconstructing full sequence for 1UA2_A using MODELLER\n",
      "environ____W> The class 'environ' is deprecated; use 'Environ' instead\n",
      "alignment__W> The class 'alignment' is deprecated; use 'Alignment' instead\n",
      "model______W> The class 'model' is deprecated; use 'Model' instead\n",
      "readlinef__W> File: Results/activation_segments/unaligned/1UA2_A.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "mkapsa__637W> No residue topology library is in memory.\n",
      "              Better radii would be used if topology.read() is called first.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  N -->  N\n",
      "              This message is written only for the first such atom.\n",
      "\n",
      "Pairwise dynamic programming alignment (ALIGN2D):\n",
      "  Residue-residue metric   : $(LIB)/as1.sim.mat\n",
      "  Diagonal                 :          100\n",
      "  Overhang                 :            0\n",
      "  Maximal gap length       :           50\n",
      "  Local alignment          :            F\n",
      "  MATRIX_OFFSET (local aln):       0.0000\n",
      "  FIX_OFFSETS              :      0.0    -1.0    -2.0    -3.0    -4.0\n",
      "  N_SUBOPT                 :            0\n",
      "  SUBOPT_OFFSET            :       0.0000\n",
      "  Alignment block          :            1\n",
      "  Gap introduction penalty :    -100.0000\n",
      "  Gap extension penalty    :       0.0000\n",
      "  Gap diagonal penalty     :       0.0000\n",
      "  Structure gap penalties  :    3.500   3.500   3.500   0.200   4.000   6.500   2.000   0.000\n",
      "  Break-break bonus        :   10000.0000\n",
      "  Length of alignment      :          346\n",
      "  Score                    :  252694.7500\n",
      "automodel__W> The class 'automodel' is deprecated; use 'AutoModel' instead\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "     55     1   43    56      I     N   12.852\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    32949    30415\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      346\n",
      "Number of all, selected real atoms                :     2748    2748\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    30415   30415\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     5723\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        2651.6462\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    2812       0      4   0.006   0.006      33.878       1.000\n",
      " 2 Bond angle potential               :    3807       0     11   2.124   2.124      331.23       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1787       0     63  47.762  47.762      634.40       1.000\n",
      " 4 Stereochemical improper torsion pot:    1142       0      1   1.381   1.381      51.353       1.000\n",
      " 5 Soft-sphere overlap restraints     :    5723       0      0   0.003   0.003      5.4254       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    5841       7     32   0.290   0.290      318.09       1.000\n",
      "10 Distance restraints 2 (N-O)        :    5881       4     23   0.302   0.302      373.48       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     344       0      4   4.566   4.566      84.569       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     298       0      0  57.854  57.854      37.999       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     246       0      1  67.284  67.284      80.998       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     102       0      3  74.563  74.563      74.349       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      42       0      0  90.051  90.051      28.290       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    4840       0      0   0.444   0.444      149.72       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     343      55     43  27.640  71.210      170.84       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    2930       0      1   0.828   0.828      277.02       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: target.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   40783.1016\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  9                           : Distance restraints 1 (CA-CA)           \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  11281  25T  55I CA  CA    196  433    9.98    7.00    2.98    5.34    7.00    2.98    5.34\n",
      "    2  11308  26V  55I CA  CA    203  433   11.90    9.34    2.56    4.53    9.34    2.56    4.53\n",
      "    3  11343  27Y  55I CA  CA    210  433   14.28   11.68    2.60    4.52   11.68    2.60    4.52\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 10                           : Distance restraints 2 (N-O)             \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  16981  25T  55I N   O     195  439   10.51    7.70    2.81    4.84    7.70    2.81    4.84\n",
      "    2  17342  55I  25T N   O     432  201    8.54    5.57    2.97    5.33    5.57    2.97    5.33\n",
      "    3  17343  55I  26V N   O     432  208   13.02   10.29    2.73    4.50   10.29    2.73    4.50\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   9549   1M   2A C   N       7    9 -112.19 -134.00   21.92    0.64  -62.50 -178.65   32.36\n",
      "    1          2A   2A N   CA      9   10  144.69  147.00                  -40.90\n",
      "    2   9550   2A   3L C   N      12   14  -89.85 -108.50   18.94    0.94  -63.50  178.97   23.47\n",
      "    2          3L   3L N   CA     14   15  135.82  132.50                  -41.20\n",
      "    3   9551   3L   4D C   N      20   22  -83.82  -70.90   23.20    0.80  -63.30  151.83   20.03\n",
      "    3          4D   4D N   CA     22   23  169.57  150.30                  -40.00\n",
      "    4   9553   5V   6K C   N      35   37  -81.54  -70.20   46.52    3.62  -62.90  137.35   16.90\n",
      "    4          6K   6K N   CA     37   38   95.28  140.40                  -40.80\n",
      "    5   9554   6K   7S C   N      44   46 -148.44 -136.60   52.36    2.86  -64.10  159.35   10.73\n",
      "    5          7S   7S N   CA     46   47  100.20  151.20                  -35.00\n",
      "    6   9555   7S   8R C   N      50   52 -143.94 -125.20   25.07    0.75  -63.00 -179.22   29.92\n",
      "    6          8R   8R N   CA     52   53  157.25  140.60                  -41.10\n",
      "    7   9556   8R   9A C   N      61   63  -72.22  -68.20    7.97    0.51  -62.50  167.21   27.89\n",
      "    7          9A   9A N   CA     63   64  152.18  145.30                  -40.90\n",
      "    8   9557   9A  10K C   N      66   68  -93.63 -118.00   55.24    2.25  -62.90  133.90   15.97\n",
      "    8         10K  10K N   CA     68   69   89.53  139.10                  -40.80\n",
      "    9   9558  10K  11R C   N      75   77 -168.21 -125.20   53.21    1.57  -63.00 -179.25   31.32\n",
      "    9         11R  11R N   CA     77   78  171.93  140.60                  -41.10\n",
      "   10   9559  11R  12Y C   N      86   88 -126.73 -124.30    4.17    0.17  -63.50 -171.28   33.13\n",
      "   10         12Y  12Y N   CA     88   89  138.79  135.40                  -43.40\n",
      "   11   9560  12Y  13E C   N      98  100  -97.90 -117.80   27.75    0.88  -63.60  161.44   19.84\n",
      "   11         13E  13E N   CA    100  101  117.46  136.80                  -40.30\n",
      "   12   9591  43I  44K C   N     347  349  156.90 -118.00   85.13    3.26  -62.90 -133.91   26.57\n",
      "   12         44K  44K N   CA    349  350  136.57  139.10                  -40.80\n",
      "   13   9592  44K  45L C   N     356  358 -164.01 -108.50   64.83    2.86  -63.50 -177.12   31.53\n",
      "   13         45L  45L N   CA    358  359  166.01  132.50                  -41.20\n",
      "   14   9593  45L  46G C   N     364  366   90.04   78.70   23.89    0.36   82.20  164.55    8.38\n",
      "   14         46G  46G N   CA    366  367  172.87 -166.10                    8.50\n",
      "   15   9595  47H  48R C   N     378  380 -134.57 -125.20   26.47    1.04  -63.00  169.40   27.77\n",
      "   15         48R  48R N   CA    380  381  165.36  140.60                  -41.10\n",
      "   16   9596  48R  49S C   N     389  391 -120.81 -136.60   16.66    0.80  -64.10  177.78   16.85\n",
      "   16         49S  49S N   CA    391  392  156.50  151.20                  -35.00\n",
      "   17   9597  49S  50E C   N     395  397  -72.02  -69.30   14.71    1.03  -63.60  162.96   22.19\n",
      "   17         50E  50E N   CA    397  398  156.96  142.50                  -40.30\n",
      "   18   9599  51A  52K C   N     409  411  -95.20 -118.00   48.87    1.97  -62.90  140.44   16.75\n",
      "   18         52K  52K N   CA    411  412   95.87  139.10                  -40.80\n",
      "   19   9600  52K  53D C   N     418  420 -160.94  -96.50   65.36    2.68  -63.30 -168.15   20.38\n",
      "   19         53D  53D N   CA    420  421  125.15  114.20                  -40.00\n",
      "   20   9601  53D  54G C   N     426  428  160.93 -167.20   31.87    0.62   82.20 -176.75   12.50\n",
      "   20         54G  54G N   CA    428  429  173.98  174.60                    8.50\n",
      "   21   9602  54G  55I C   N     430  432  -67.49  -63.40   44.20    7.39  -63.40   44.20    7.39\n",
      "   21         55I  55I N   CA    432  433  -87.61  -43.60                  -43.60\n",
      "   22   9603  55I  56N C   N     438  440  164.76 -119.90   86.36    2.62   55.90  177.11   20.08\n",
      "   22         56N  56N N   CA    440  441  179.20  137.00                   39.50\n",
      "   23   9604  56N  57R C   N     446  448  -68.84  -63.00   23.37    3.56 -125.20  165.56    8.94\n",
      "   23         57R  57R N   CA    448  449  -63.72  -41.10                  140.60\n",
      "   24   9620  72P  73N C   N     579  581 -105.77  -63.20   86.87    9.34  -63.20   86.87    9.34\n",
      "   24         73N  73N N   CA    581  582   34.63  -41.10                  -41.10\n",
      "   25   9631  83H  84K C   N     661  663  -52.44  -62.90   13.66    2.35 -118.00 -176.75    7.79\n",
      "   25         84K  84K N   CA    663  664  -32.02  -40.80                  139.10\n",
      "   26   9659 111P 112S C   N     883  885  -52.46  -64.10   37.63    3.54   56.90  115.02   14.58\n",
      "   26        112S 112S N   CA    885  886    0.79  -35.00                   36.40\n",
      "   27   9684 136R 137D C   N    1102 1104  -94.67  -63.30   63.70    6.76  -63.30   63.70    6.76\n",
      "   27        137D 137D N   CA   1104 1105   15.45  -40.00                  -40.00\n",
      "   28   9707 159A 160K C   N    1276 1278 -109.68  -62.90   52.37    9.15  -62.90   52.37    9.15\n",
      "   28        160K 160K N   CA   1278 1279  -64.32  -40.80                  -40.80\n",
      "   29   9857 310P 311N C   N    2465 2467 -127.28 -119.90   79.31    3.43  -63.20  121.25   18.71\n",
      "   29        311N 311N N   CA   2467 2468 -144.03  137.00                  -41.10\n",
      "   30   9861 314V 315E C   N    2493 2495  -81.21  -69.30   19.09    1.08  -63.60  163.23   22.75\n",
      "   30        315E 315E N   CA   2495 2496  157.43  142.50                  -40.30\n",
      "   31   9862 315E 316T C   N    2502 2504  -77.20  -78.10   13.37    0.60  -63.20  155.39   20.86\n",
      "   31        316T 316T N   CA   2504 2505  163.14  149.80                  -42.10\n",
      "   32   9863 316T 317L C   N    2509 2511  -78.20  -70.70    7.51    0.70  -63.50  178.21   25.61\n",
      "   32        317L 317L N   CA   2511 2512  141.19  141.60                  -41.20\n",
      "   33   9864 317L 318K C   N    2517 2519  -80.72  -70.20   15.53    0.93  -62.90  168.32   22.87\n",
      "   33        318K 318K N   CA   2519 2520  151.82  140.40                  -40.80\n",
      "   34   9865 318K 319E C   N    2526 2528  -68.60  -69.30    0.74    0.05  -63.60  177.52   23.94\n",
      "   34        319E 319E N   CA   2528 2529  142.25  142.50                  -40.30\n",
      "   35   9866 319E 320Q C   N    2535 2537  -76.29  -73.00   10.75    0.64  -63.80  169.22   25.54\n",
      "   35        320Q 320Q N   CA   2537 2538  150.94  140.70                  -40.30\n",
      "   36   9867 320Q 321S C   N    2544 2546 -119.45 -136.60   17.17    0.71  -64.10 -178.48   17.01\n",
      "   36        321S 321S N   CA   2546 2547  152.12  151.20                  -35.00\n",
      "   37   9868 321S 322N C   N    2550 2552 -141.15 -119.90   25.73    0.76  -63.20 -175.34   27.49\n",
      "   37        322N 322N N   CA   2552 2553  151.51  137.00                  -41.10\n",
      "   38   9871 324A 325L C   N    2570 2572  -89.50 -108.50   19.08    0.89  -63.50  173.89   22.78\n",
      "   38        325L 325L N   CA   2572 2573  130.74  132.50                  -41.20\n",
      "   39   9872 325L 326A C   N    2578 2580  -74.00  -68.20    6.35    0.61  -62.50  176.78   29.55\n",
      "   39        326A 326A N   CA   2580 2581  142.70  145.30                  -40.90\n",
      "   40   9874 327I 328K C   N    2591 2593  -64.63  -70.20    5.79    0.38  -62.90  179.63   23.23\n",
      "   40        328K 328K N   CA   2593 2594  138.82  140.40                  -40.80\n",
      "   41   9875 328K 329R C   N    2600 2602  -88.29  -72.10   18.13    1.22  -63.00  170.74   24.81\n",
      "   41        329R 329R N   CA   2602 2603  150.04  141.90                  -41.10\n",
      "   42   9876 329R 330K C   N    2611 2613 -132.01 -118.00   24.95    0.95  -62.90  173.79   26.70\n",
      "   42        330K 330K N   CA   2613 2614  159.74  139.10                  -40.80\n",
      "   43   9877 330K 331R C   N    2620 2622 -136.28 -125.20   23.98    0.87  -63.00  173.29   28.41\n",
      "   43        331R 331R N   CA   2622 2623  161.87  140.60                  -41.10\n",
      "   44   9878 331R 332T C   N    2631 2633 -131.42 -124.80   10.77    0.33  -63.20  179.39   27.49\n",
      "   44        332T 332T N   CA   2633 2634  151.99  143.50                  -42.10\n",
      "   45   9879 332T 333E C   N    2638 2640 -136.00 -117.80   20.72    0.61  -63.60 -172.46   29.08\n",
      "   45        333E 333E N   CA   2640 2641  146.70  136.80                  -40.30\n",
      "   46   9880 333E 334A C   N    2647 2649  -69.45  -68.20    2.78    0.18  -62.50  171.45   28.44\n",
      "   46        334A 334A N   CA   2649 2650  147.79  145.30                  -40.90\n",
      "   47   9881 334A 335L C   N    2652 2654 -126.05 -108.50   26.93    1.52  -63.50  165.54   20.42\n",
      "   47        335L 335L N   CA   2654 2655  112.07  132.50                  -41.20\n",
      "   48   9882 335L 336E C   N    2660 2662 -117.50 -117.80   19.02    0.95  -63.60  167.02   19.85\n",
      "   48        336E 336E N   CA   2662 2663  117.78  136.80                  -40.30\n",
      "   49   9883 336E 337Q C   N    2669 2671 -153.56 -121.10   33.50    1.11  -63.80 -166.24   33.14\n",
      "   49        337Q 337Q N   CA   2671 2672  147.98  139.70                  -40.30\n",
      "   50   9884 337Q 338G C   N    2678 2680 -142.63 -167.20   57.62    1.80   82.20  176.81   13.73\n",
      "   50        338G 338G N   CA   2680 2681  122.48  174.60                    8.50\n",
      "   51   9885 338G 339G C   N    2682 2684   59.65   78.70   20.58    0.79   82.20  168.36    9.29\n",
      "   51        339G 339G N   CA   2684 2685 -158.34 -166.10                    8.50\n",
      "   52   9886 339G 340L C   N    2686 2688 -106.67 -108.50   20.21    1.07  -63.50  159.53   20.12\n",
      "   52        340L 340L N   CA   2688 2689  112.37  132.50                  -41.20\n",
      "   53   9888 341P 342K C   N    2701 2703  179.21 -118.00   63.54    2.57  -62.90 -152.97   24.03\n",
      "   53        342K 342K N   CA   2703 2704  129.39  139.10                  -40.80\n",
      "   54   9889 342K 343K C   N    2710 2712 -178.85 -118.00   67.22    2.23  -62.90 -169.20   31.58\n",
      "   54        343K 343K N   CA   2712 2713  167.67  139.10                  -40.80\n",
      "   55   9890 343K 344L C   N    2719 2721  -72.49  -70.70    3.70    0.23  -63.50  174.19   24.71\n",
      "   55        344L 344L N   CA   2721 2722  144.84  141.60                  -41.20\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0   19   51  152  157  225  296  318  354  408  453\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "target.B99990001.pdb          2651.64624\n",
      "\n",
      "Reconstruction completed for 1UA2_A. File saved at Results/activation_segments/reconstructedModeller/1UA2_A_filled.pdb\n",
      "Removed REMARK lines from Results/activation_segments/reconstructedModeller/1UA2_A_filled.pdb\n",
      "Processing 7M0L_A\n",
      "No differences found for 7M0L_A. Original PDB copied to target directory.\n",
      "Processing 3TT0_A\n",
      "No differences found for 3TT0_A. Original PDB copied to target directory.\n",
      "Processing 3LCO_A\n",
      "Reconstructing full sequence for 3LCO_A using MODELLER\n",
      "environ____W> The class 'environ' is deprecated; use 'Environ' instead\n",
      "alignment__W> The class 'alignment' is deprecated; use 'Alignment' instead\n",
      "model______W> The class 'model' is deprecated; use 'Model' instead\n",
      "mkapsa__637W> No residue topology library is in memory.\n",
      "              Better radii would be used if topology.read() is called first.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  N -->  N\n",
      "              This message is written only for the first such atom.\n",
      "\n",
      "Pairwise dynamic programming alignment (ALIGN2D):\n",
      "  Residue-residue metric   : $(LIB)/as1.sim.mat\n",
      "  Diagonal                 :          100\n",
      "  Overhang                 :            0\n",
      "  Maximal gap length       :           50\n",
      "  Local alignment          :            F\n",
      "  MATRIX_OFFSET (local aln):       0.0000\n",
      "  FIX_OFFSETS              :      0.0    -1.0    -2.0    -3.0    -4.0\n",
      "  N_SUBOPT                 :            0\n",
      "  SUBOPT_OFFSET            :       0.0000\n",
      "  Alignment block          :            1\n",
      "  Gap introduction penalty :    -100.0000\n",
      "  Gap extension penalty    :       0.0000\n",
      "  Gap diagonal penalty     :       0.0000\n",
      "  Structure gap penalties  :    3.500   3.500   3.500   0.200   4.000   6.500   2.000   0.000\n",
      "  Break-break bonus        :   10000.0000\n",
      "  Length of alignment      :          324\n",
      "  Score                    :  258120.8281\n",
      "automodel__W> The class 'automodel' is deprecated; use 'AutoModel' instead\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "    151     1  686   747      D     G   15.136\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    31442    29116\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      324\n",
      "Number of all, selected real atoms                :     2551    2551\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    29116   29116\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     5047\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1474.2644\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    2613       0      0   0.005   0.005      18.155       1.000\n",
      " 2 Bond angle potential               :    3540       0      2   1.790   1.790      221.89       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1679       0     50  46.983  46.983      564.04       1.000\n",
      " 4 Stereochemical improper torsion pot:    1091       0      0   1.101   1.101      28.432       1.000\n",
      " 5 Soft-sphere overlap restraints     :    5047       0      0   0.001   0.001     0.82545       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    5917       0      3   0.195   0.195      72.346       1.000\n",
      "10 Distance restraints 2 (N-O)        :    5787       0      1   0.222   0.222      113.71       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     323       1      0  10.322  10.322      38.616       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     271       0      1  52.053  52.053      12.349       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     217       0      2  69.390  69.390      62.429       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      92       0      0  74.734  74.734      55.371       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      32       0      0  86.335  86.335      20.166       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    4516       0      0   0.361   0.361      89.891       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     322      25     25  25.243  51.666      10.039       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    2716       0      2   0.701   0.701      166.00       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: target.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   27487.1191\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   8924   1W   2K C   N      13   15  -73.39  -70.20    5.41    0.33  -62.90  174.75   23.28\n",
      "    1          2K   2K N   CA     15   16  144.77  140.40                  -40.80\n",
      "    2   8928   5E   6S C   N      47   49   85.53 -136.60  158.00    4.83  -64.10  178.12   17.12\n",
      "    2          6S   6S N   CA     49   50 -131.63  151.20                  -35.00\n",
      "    3   8929   6S   7Y C   N      53   55  -54.45  -98.40   46.40    2.18  -63.50  173.55   27.08\n",
      "    3          7Y   7Y N   CA     55   56  143.28  128.40                  -43.40\n",
      "    4   8930   7Y   8E C   N      65   67  -49.26  -69.30   31.00    1.74  -63.60  159.80   22.10\n",
      "    4          8E   8E N   CA     67   68  118.85  142.50                  -40.30\n",
      "    5   8931   8E   9G C   N      74   76 -104.05  -80.20   24.70    1.79   82.20 -124.37   18.20\n",
      "    5          9G   9G N   CA     76   77  167.67  174.10                    8.50\n",
      "    6   8933  10N  11S C   N      86   88  -59.75  -72.40   29.32    1.37  -64.10  161.01   11.93\n",
      "    6         11S  11S N   CA     88   89  125.95  152.40                  -35.00\n",
      "    7   8935  12Y  13T C   N     104  106 -128.73 -124.80   18.76    0.92  -63.20  179.63   19.53\n",
      "    7         13T  13T N   CA    106  107  125.15  143.50                  -42.10\n",
      "    8   8954  31N  32N C   N     274  276  -86.25  -63.20   79.57    8.88  -63.20   79.57    8.88\n",
      "    8         32N  32N N   CA    276  277   35.06  -41.10                  -41.10\n",
      "    9   9018  95E  96N C   N     748  750 -107.46 -119.90   93.19    3.94  -63.20   96.49   10.40\n",
      "    9         96N  96N N   CA    750  751   44.64  137.00                  -41.10\n",
      "   10   9057 134L 135G C   N    1045 1047   51.88   82.20   77.52    2.35  -62.40  166.47   30.71\n",
      "   10        135G 135G N   CA   1047 1048   79.85    8.50                  -41.20\n",
      "   11   9058 135G 136P C   N    1049 1051  -61.20  -64.50   10.01    0.60  -58.70  168.27   13.70\n",
      "   11        136P 136P N   CA   1051 1052  137.75  147.20                  -30.50\n",
      "   12   9059 136P 137S C   N    1056 1058 -137.04 -136.60   12.84    0.64  -64.10  176.72   18.01\n",
      "   12        137S 137S N   CA   1058 1059  164.03  151.20                  -35.00\n",
      "   13   9061 138L 139A C   N    1070 1072  -66.49  -68.20    9.45    0.82  -62.50  164.56   27.16\n",
      "   13        139A 139A N   CA   1072 1073  154.59  145.30                  -40.90\n",
      "   14   9063 140P 141G C   N    1082 1084  111.50   78.70   59.85    0.96   82.20  138.46    8.15\n",
      "   14        141G 141G N   CA   1084 1085  143.83 -166.10                    8.50\n",
      "   15   9065 142Q 143D C   N    1095 1097  -95.47  -70.90   25.88    1.63  -63.30  164.75   22.39\n",
      "   15        143D 143D N   CA   1097 1098  158.42  150.30                  -40.00\n",
      "   16   9067 144P 145E C   N    1110 1112  -63.18  -69.30   20.52    1.35  -63.60  163.22   21.79\n",
      "   16        145E 145E N   CA   1112 1113  122.92  142.50                  -40.30\n",
      "   17   9068 145E 146G C   N    1119 1121  -66.12  -62.40    6.00    0.83   82.20  155.00   11.69\n",
      "   17        146G 146G N   CA   1121 1122  -36.49  -41.20                    8.50\n",
      "   18   9070 147L 148D C   N    1131 1133 -105.62  -96.50   18.40    0.76  -63.30  144.55   15.78\n",
      "   18        148D 148D N   CA   1133 1134   98.22  114.20                  -40.00\n",
      "   19   9071 148D 149K C   N    1139 1141 -117.74 -118.00   31.73    1.54  -62.90  157.99   18.33\n",
      "   19        149K 149K N   CA   1141 1142  107.37  139.10                  -40.80\n",
      "   20   9072 149K 150E C   N    1148 1150 -117.83 -117.80   55.36    2.78  -63.60  133.27   15.55\n",
      "   20        150E 150E N   CA   1150 1151   81.44  136.80                  -40.30\n",
      "   21   9142 219N 220A C   N    1687 1689  -90.47  -68.20   28.04    1.81  -62.50  159.24   27.55\n",
      "   21        220A 220A N   CA   1689 1690  162.34  145.30                  -40.90\n",
      "   22   9242 319A 320Q C   N    2501 2503  -72.90  -73.00   18.68    1.22  -63.80  160.58   24.09\n",
      "   22        320Q 320Q N   CA   2503 2504  159.38  140.70                  -40.30\n",
      "   23   9243 320Q 321E C   N    2510 2512  -63.49  -69.30    8.82    0.81  -63.60  170.56   22.74\n",
      "   23        321E 321E N   CA   2512 2513  149.14  142.50                  -40.30\n",
      "   24   9244 321E 322D C   N    2519 2521  -62.29  -70.90   10.22    0.52  -63.30  175.21   21.57\n",
      "   24        322D 322D N   CA   2521 2522  144.79  150.30                  -40.00\n",
      "   25   9245 322D 323R C   N    2527 2529  -77.97  -72.10   10.01    0.66  -63.00  169.56   23.93\n",
      "   25        323R 323R N   CA   2529 2530  150.00  141.90                  -41.10\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0    8   26   76  118  218  225  301  343  398  424\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "target.B99990001.pdb          1474.26440\n",
      "\n",
      "Reconstruction completed for 3LCO_A. File saved at Results/activation_segments/reconstructedModeller/3LCO_A_filled.pdb\n",
      "Removed REMARK lines from Results/activation_segments/reconstructedModeller/3LCO_A_filled.pdb\n",
      "Processing 4J99_A\n",
      "Reconstructing full sequence for 4J99_A using MODELLER\n",
      "environ____W> The class 'environ' is deprecated; use 'Environ' instead\n",
      "alignment__W> The class 'alignment' is deprecated; use 'Alignment' instead\n",
      "model______W> The class 'model' is deprecated; use 'Model' instead\n",
      "mkapsa__637W> No residue topology library is in memory.\n",
      "              Better radii would be used if topology.read() is called first.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  N -->  N\n",
      "              This message is written only for the first such atom.\n",
      "\n",
      "Pairwise dynamic programming alignment (ALIGN2D):\n",
      "  Residue-residue metric   : $(LIB)/as1.sim.mat\n",
      "  Diagonal                 :          100\n",
      "  Overhang                 :            0\n",
      "  Maximal gap length       :           50\n",
      "  Local alignment          :            F\n",
      "  MATRIX_OFFSET (local aln):       0.0000\n",
      "  FIX_OFFSETS              :      0.0    -1.0    -2.0    -3.0    -4.0\n",
      "  N_SUBOPT                 :            0\n",
      "  SUBOPT_OFFSET            :       0.0000\n",
      "  Alignment block          :            1\n",
      "  Gap introduction penalty :    -100.0000\n",
      "  Gap extension penalty    :       0.0000\n",
      "  Gap diagonal penalty     :       0.0000\n",
      "  Structure gap penalties  :    3.500   3.500   3.500   0.200   4.000   6.500   2.000   0.000\n",
      "  Break-break bonus        :   10000.0000\n",
      "  Length of alignment      :          324\n",
      "  Score                    :  245890.9062\n",
      "automodel__W> The class 'automodel' is deprecated; use 'AutoModel' instead\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "    217     1  659   662      T     N    9.039\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    31466    29075\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      324\n",
      "Number of all, selected real atoms                :     2594    2594\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    29075   29075\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     5281\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1893.2256\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    2654       0      0   0.005   0.005      20.202       1.000\n",
      " 2 Bond angle potential               :    3590       0      5   1.999   1.999      278.63       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1689       0     58  48.144  48.144      600.19       1.000\n",
      " 4 Stereochemical improper torsion pot:    1078       0      0   1.328   1.328      38.928       1.000\n",
      " 5 Soft-sphere overlap restraints     :    5281       0      0   0.001   0.001     0.92274       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    5687       0      4   0.192   0.192      139.73       1.000\n",
      "10 Distance restraints 2 (N-O)        :    5624       1      9   0.244   0.244      182.23       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     323       0      1   3.745   3.745      52.508       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     284       0      4  54.774  54.774      40.474       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     226       0      3  62.301  62.301      73.638       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     109       0      1  73.784  73.784      72.589       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      38       0      0  90.941  90.941      26.883       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    4486       0      0   0.401   0.401      99.413       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     322      42     40  27.927  63.574      84.665       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    2965       0      0   0.766   0.766      182.23       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: target.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   32733.4512\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   9012   1M   2G C   N       7    9  -80.90  -80.20   37.72    1.52   82.20 -152.74    7.49\n",
      "    1          2G   2G N   CA      9   10  136.38  174.10                    8.50\n",
      "    2   9013   2G   3S C   N      11   13  -88.21  -72.40   30.43    1.39  -64.10  148.57   12.36\n",
      "    2          3S   3S N   CA     13   14  178.40  152.40                  -35.00\n",
      "    3   9014   3S   4S C   N      17   19 -127.11 -136.60   13.99    0.78  -64.10  175.24   17.15\n",
      "    3          4S   4S N   CA     19   20  161.48  151.20                  -35.00\n",
      "    4   9015   4S   5H C   N      23   25  -77.47  -67.60   36.13    3.04  -63.20  148.24   17.69\n",
      "    4          5H   5H N   CA     25   26  105.25  140.00                  -42.30\n",
      "    5   9016   5H   6H C   N      33   35 -177.77 -125.60   56.00    1.55  -63.20 -164.40   31.70\n",
      "    5          6H   6H N   CA     35   36  159.16  138.80                  -42.30\n",
      "    6   9017   6H   7H C   N      43   45  -74.53  -67.60    9.18    0.82  -63.20  176.65   21.40\n",
      "    6          7H   7H N   CA     45   46  133.98  140.00                  -42.30\n",
      "    7   9018   7H   8H C   N      53   55 -152.13 -125.60   40.15    1.82  -63.20  175.20   18.63\n",
      "    7          8H   8H N   CA     55   56  108.66  138.80                  -42.30\n",
      "    8   9019   8H   9H C   N      63   65  -81.96  -67.60   30.74    2.70  -63.20  156.26   18.44\n",
      "    8          9H   9H N   CA     65   66  112.83  140.00                  -42.30\n",
      "    9   9020   9H  10H C   N      73   75 -104.76 -125.60   40.65    1.11  -63.20  151.99   16.83\n",
      "    9         10H  10H N   CA     75   76  103.90  138.80                  -42.30\n",
      "   10   9021  10H  11S C   N      83   85  -83.54  -72.40   33.11    1.61  -64.10  142.74   11.61\n",
      "   10         11S  11S N   CA     85   86 -176.41  152.40                  -35.00\n",
      "   11   9022  11S  12Q C   N      89   91  -72.74  -73.00    8.64    0.57  -63.80  170.59   25.56\n",
      "   11         12Q  12Q N   CA     91   92  149.34  140.70                  -40.30\n",
      "   12   9023  12Q  13D C   N      98  100  -65.31  -96.50   31.49    1.29  -63.30  149.93   18.39\n",
      "   12         13D  13D N   CA    100  101  109.91  114.20                  -40.00\n",
      "   13   9025  14P  15M C   N     113  115  -78.38  -73.00   12.30    0.93  -63.40  173.09   24.71\n",
      "   13         15M  15M N   CA    115  116  131.93  143.00                  -40.50\n",
      "   14   9026  15M  16L C   N     121  123  -74.61  -70.70   13.65    1.19  -63.50  170.09   23.00\n",
      "   14         16L  16L N   CA    123  124  128.53  141.60                  -41.20\n",
      "   15   9027  16L  17A C   N     129  131 -168.16 -134.00   34.42    1.30  -62.50 -154.44   39.32\n",
      "   15         17A  17A N   CA    131  132  142.77  147.00                  -40.90\n",
      "   16   9028  17A  18G C   N     134  136 -150.30 -167.20   18.40    0.27   82.20 -156.34   15.23\n",
      "   16         18G  18G N   CA    136  137  167.32  174.60                    8.50\n",
      "   17   9030  19V  20S C   N     145  147 -117.09 -136.60   33.13    1.17  -64.10  167.99   10.35\n",
      "   17         20S  20S N   CA    147  148  124.42  151.20                  -35.00\n",
      "   18   9031  20S  21E C   N     151  153   69.43  -69.30  141.22   11.58  -63.60 -158.93   23.64\n",
      "   18         21E  21E N   CA    153  154  168.92  142.50                  -40.30\n",
      "   19   9112 101K 102N C   N     782  784 -102.04  -63.20   85.92    9.26  -63.20   85.92    9.26\n",
      "   19        102N 102N N   CA    784  785   35.54  -41.10                  -41.10\n",
      "   20   9149 138P 139G C   N    1075 1077  -66.42  -80.20   30.08    0.70   82.20 -156.60    6.61\n",
      "   20        139G 139G N   CA   1077 1078  147.37  174.10                    8.50\n",
      "   21   9150 139G 140M C   N    1079 1081 -103.29 -125.60   23.09    1.06  -63.40  177.57   29.70\n",
      "   21        140M 140M N   CA   1081 1082  146.47  140.50                  -40.50\n",
      "   22   9151 140M 141E C   N    1087 1089 -104.78 -117.80   56.69    3.03  -63.60  134.20   20.22\n",
      "   22        141E 141E N   CA   1089 1090 -168.03  136.80                  -40.30\n",
      "   23   9152 141E 142Y C   N    1096 1098  -80.35  -98.40   31.51    2.91  -63.50  146.94   22.47\n",
      "   23        142Y 142Y N   CA   1098 1099  102.57  128.40                  -43.40\n",
      "   24   9153 142Y 143S C   N    1108 1110 -118.45 -136.60   48.78    2.74  -64.10  139.54   13.97\n",
      "   24        143S 143S N   CA   1110 1111 -163.52  151.20                  -35.00\n",
      "   25   9154 143S 144Y C   N    1114 1116 -115.14 -124.30    9.34    0.64  -63.50 -175.62   26.82\n",
      "   25        144Y 144Y N   CA   1116 1117  133.60  135.40                  -43.40\n",
      "   26   9155 144Y 145D C   N    1126 1128 -135.01  -70.90   99.60    3.70  -63.30  117.82   19.04\n",
      "   26        145D 145D N   CA   1128 1129 -133.48  150.30                  -40.00\n",
      "   27   9157 146I 147N C   N    1142 1144  -82.88  -71.20   35.41    2.59  -63.20  151.75   17.96\n",
      "   27        147N 147N N   CA   1144 1145  109.37  142.80                  -41.10\n",
      "   28   9158 147N 148R C   N    1150 1152 -160.25 -125.20   45.20    1.34  -63.00  178.57   30.58\n",
      "   28        148R 148R N   CA   1152 1153  169.14  140.60                  -41.10\n",
      "   29   9162 151E 152E C   N    1184 1186 -132.31 -117.80   57.97    2.60  -63.60  144.19   23.04\n",
      "   29        152E 152E N   CA   1186 1187 -167.07  136.80                  -40.30\n",
      "   30   9192 181R 182D C   N    1429 1431 -105.64  -63.30   64.22    7.09  -63.30   64.22    7.09\n",
      "   30        182D 182D N   CA   1431 1432    8.28  -40.00                  -40.00\n",
      "   31   9197 186R 187N C   N    1466 1468 -102.79  -63.20   81.86    8.80  -63.20   81.86    8.80\n",
      "   31        187N 187N N   CA   1468 1469   30.56  -41.10                  -41.10\n",
      "   32   9218 207I 208N C   N    1628 1630   59.24   55.90   27.58    1.69  -71.20 -175.36    9.93\n",
      "   32        208N 208N N   CA   1630 1631   12.12   39.50                  142.80\n",
      "   33   9219 208N 209N C   N    1636 1638 -123.47 -119.90   38.04    1.81  -63.20  152.63   16.60\n",
      "   33        209N 209N N   CA   1638 1639   99.13  137.00                  -41.10\n",
      "   34   9220 209N 210I C   N    1644 1646  -59.81  -63.40    5.10    0.96 -120.60 -179.19    8.16\n",
      "   34        210I 210I N   CA   1646 1647  -39.98  -43.60                  130.30\n",
      "   35   9225 214K 215T C   N    1693 1695  -70.82  -78.10    9.53    0.62  -63.20  162.13   21.29\n",
      "   35        215T 215T N   CA   1695 1696  155.94  149.80                  -42.10\n",
      "   36   9226 215T 216T C   N    1700 1702  -64.31  -78.10   44.53    1.53  -63.20  149.56   19.11\n",
      "   36        216T 216T N   CA   1702 1703  107.45  149.80                  -42.10\n",
      "   37   9228 217T 218N C   N    1714 1716  -72.52  -63.20   10.30    1.32  -71.20  179.52   11.54\n",
      "   37        218N 218N N   CA   1716 1717  -36.71  -41.10                  142.80\n",
      "   38   9229 218N 219G C   N    1722 1724  112.23 -167.20  174.08    7.08  -80.20 -132.54   16.24\n",
      "   38        219G 219G N   CA   1724 1725   20.28  174.60                  174.10\n",
      "   39   9330 319L 320T C   N    2552 2554  -62.47  -63.20    1.65    0.18 -124.80 -176.19    9.67\n",
      "   39        320T 320T N   CA   2554 2555  -43.58  -42.10                  143.50\n",
      "   40   9331 320T 321T C   N    2559 2561  -91.98 -124.80   75.92    2.43  -63.20  120.63   13.76\n",
      "   40        321T 321T N   CA   2561 2562   75.04  143.50                  -42.10\n",
      "   41   9332 321T 322N C   N    2566 2568 -153.49 -119.90   34.33    1.25  -63.20 -163.20   29.67\n",
      "   41        322N 322N N   CA   2568 2569  144.04  137.00                  -41.10\n",
      "   42   9333 322N 323E C   N    2574 2576 -111.39 -117.80   44.36    2.10  -63.60  141.52   16.75\n",
      "   42        323E 323E N   CA   2576 2577   92.90  136.80                  -40.30\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0    9   24   86  164  210  229  296  382  415  435\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "target.B99990001.pdb          1893.22559\n",
      "\n",
      "Reconstruction completed for 4J99_A. File saved at Results/activation_segments/reconstructedModeller/4J99_A_filled.pdb\n",
      "Removed REMARK lines from Results/activation_segments/reconstructedModeller/4J99_A_filled.pdb\n",
      "Processing 1XJD_A\n",
      "Reconstructing full sequence for 1XJD_A using MODELLER\n",
      "environ____W> The class 'environ' is deprecated; use 'Environ' instead\n",
      "alignment__W> The class 'alignment' is deprecated; use 'Alignment' instead\n",
      "model______W> The class 'model' is deprecated; use 'Model' instead\n",
      "mkapsa__637W> No residue topology library is in memory.\n",
      "              Better radii would be used if topology.read() is called first.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  N -->  N\n",
      "              This message is written only for the first such atom.\n",
      "\n",
      "Pairwise dynamic programming alignment (ALIGN2D):\n",
      "  Residue-residue metric   : $(LIB)/as1.sim.mat\n",
      "  Diagonal                 :          100\n",
      "  Overhang                 :            0\n",
      "  Maximal gap length       :           50\n",
      "  Local alignment          :            F\n",
      "  MATRIX_OFFSET (local aln):       0.0000\n",
      "  FIX_OFFSETS              :      0.0    -1.0    -2.0    -3.0    -4.0\n",
      "  N_SUBOPT                 :            0\n",
      "  SUBOPT_OFFSET            :       0.0000\n",
      "  Alignment block          :            1\n",
      "  Gap introduction penalty :    -100.0000\n",
      "  Gap extension penalty    :       0.0000\n",
      "  Gap diagonal penalty     :       0.0000\n",
      "  Structure gap penalties  :    3.500   3.500   3.500   0.200   4.000   6.500   2.000   0.000\n",
      "  Break-break bonus        :   10000.0000\n",
      "  Length of alignment      :          345\n",
      "  Score                    :  245856.7656\n",
      "automodel__W> The class 'automodel' is deprecated; use 'AutoModel' instead\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "    288     1  649   688      P     P   48.621\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "\n",
      "getf_______W> RTF restraint not found in the atoms list:\n",
      "              residue type, indices:    13     1\n",
      "              atom names           : N     -C    CA    CD\n",
      "              atom indices         :     1     0     2     3\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    33405    30850\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[12], line 148\u001b[0m\n\u001b[1;32m    145\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mProcessing complete!\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    147\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;18m__name__\u001b[39m \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m__main__\u001b[39m\u001b[38;5;124m'\u001b[39m:\n\u001b[0;32m--> 148\u001b[0m     \u001b[43mmain\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn[12], line 143\u001b[0m, in \u001b[0;36mmain\u001b[0;34m()\u001b[0m\n\u001b[1;32m    140\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mexists(target_dir):\n\u001b[1;32m    141\u001b[0m             os\u001b[38;5;241m.\u001b[39mmakedirs(target_dir)\n\u001b[0;32m--> 143\u001b[0m         \u001b[43mreconstruct_with_modeller\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpdb_chain_id\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpdb_file_path\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtarget_dir\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfull_sequence\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43matom_sequence\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    145\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mProcessing complete!\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "Cell \u001b[0;32mIn[12], line 92\u001b[0m, in \u001b[0;36mreconstruct_with_modeller\u001b[0;34m(pdb_chain_id, pdb_path, target_path, full_sequence, atom_sequence)\u001b[0m\n\u001b[1;32m     89\u001b[0m a\u001b[38;5;241m.\u001b[39mending_model \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n\u001b[1;32m     91\u001b[0m \u001b[38;5;66;03m# Build the model\u001b[39;00m\n\u001b[0;32m---> 92\u001b[0m \u001b[43ma\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmake\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     94\u001b[0m \u001b[38;5;66;03m# Save the best model to the target directory\u001b[39;00m\n\u001b[1;32m     95\u001b[0m model_path \u001b[38;5;241m=\u001b[39m os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mjoin(target_path, \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mpdb_chain_id\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m_filled.pdb\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/modeller/automodel/automodel.py:152\u001b[0m, in \u001b[0;36mAutoModel.make\u001b[0;34m(self, exit_stage)\u001b[0m\n\u001b[1;32m    149\u001b[0m \u001b[38;5;66;03m# getting model(s) (topology library must be in memory; ensured\u001b[39;00m\n\u001b[1;32m    150\u001b[0m \u001b[38;5;66;03m# now by one of the three GENERATE_METHOD routines):\u001b[39;00m\n\u001b[1;32m    151\u001b[0m atmsel \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_select_atoms()\n\u001b[0;32m--> 152\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmultiple_models\u001b[49m\u001b[43m(\u001b[49m\u001b[43matmsel\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    154\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mwrite_summary(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moutputs, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmodels\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m    156\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfinal_malign3d:\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/modeller/automodel/automodel.py:259\u001b[0m, in \u001b[0;36mAutoModel.multiple_models\u001b[0;34m(self, atmsel)\u001b[0m\n\u001b[1;32m    257\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    258\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m num \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstarting_model, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mending_model \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m1\u001b[39m):\n\u001b[0;32m--> 259\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moutputs\u001b[38;5;241m.\u001b[39mappend(\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msingle_model\u001b[49m\u001b[43m(\u001b[49m\u001b[43matmsel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnum\u001b[49m\u001b[43m)\u001b[49m)\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/modeller/automodel/automodel.py:347\u001b[0m, in \u001b[0;36mAutoModel.single_model\u001b[0;34m(self, atmsel, num, parallel)\u001b[0m\n\u001b[1;32m    345\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    346\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m irepeat \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;241m0\u001b[39m, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrepeat_optimization):\n\u001b[0;32m--> 347\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msingle_model_pass\u001b[49m\u001b[43m(\u001b[49m\u001b[43matmsel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnum\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msched\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    348\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mto_iupac()\n\u001b[1;32m    349\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ModellerError, \u001b[38;5;167;01mOverflowError\u001b[39;00m):\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/modeller/automodel/automodel.py:549\u001b[0m, in \u001b[0;36mAutoModel.single_model_pass\u001b[0;34m(self, atmsel, num, sched)\u001b[0m\n\u001b[1;32m    545\u001b[0m         log\u001b[38;5;241m.\u001b[39merror(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124msingle_model\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[1;32m    546\u001b[0m                   \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mObj. func. (\u001b[39m\u001b[38;5;132;01m%.3f\u001b[39;00m\u001b[38;5;124m) exceeded max_molpdf (\u001b[39m\u001b[38;5;132;01m%.3f\u001b[39;00m\u001b[38;5;124m) \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    547\u001b[0m                   \u001b[38;5;241m%\u001b[39m (molpdf, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmax_molpdf))\n\u001b[1;32m    548\u001b[0m actions \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mget_refine_actions()\n\u001b[0;32m--> 549\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrefine\u001b[49m\u001b[43m(\u001b[49m\u001b[43matmsel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mactions\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/modeller/automodel/automodel.py:979\u001b[0m, in \u001b[0;36mAutoModel.refine\u001b[0;34m(self, atmsel, actions)\u001b[0m\n\u001b[1;32m    977\u001b[0m \u001b[38;5;66;03m# Do simulated annealing MD:\u001b[39;00m\n\u001b[1;32m    978\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmd_level:\n\u001b[0;32m--> 979\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmd_level\u001b[49m\u001b[43m(\u001b[49m\u001b[43mhot_atmsel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mactions\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    981\u001b[0m \u001b[38;5;66;03m# Possibly skip 'HOT CG' after MD:\u001b[39;00m\n\u001b[1;32m    982\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrefine_hot_only:\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/modeller/automodel/refine.py:9\u001b[0m, in \u001b[0;36mvery_fast\u001b[0;34m(atmsel, actions)\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Very fast MD annealing\"\"\"\u001b[39;00m\n\u001b[1;32m      8\u001b[0m \u001b[38;5;66;03m# at T=1000, max_atom_shift for 4fs is cca 0.15 A.\u001b[39;00m\n\u001b[0;32m----> 9\u001b[0m \u001b[43mrefine\u001b[49m\u001b[43m(\u001b[49m\u001b[43matmsel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mactions\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcap\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m0.39\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimestep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m4.0\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     10\u001b[0m \u001b[43m       \u001b[49m\u001b[43mequil_its\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m50\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mequil_equil\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m10\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     11\u001b[0m \u001b[43m       \u001b[49m\u001b[43mequil_temps\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m150.0\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m400.0\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1000.0\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     12\u001b[0m \u001b[43m       \u001b[49m\u001b[43msampl_its\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m300\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msampl_equil\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m100\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     13\u001b[0m \u001b[43m       \u001b[49m\u001b[43msampl_temps\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m1000.0\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m800.0\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m500.0\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m300.0\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/modeller/automodel/refine.py:66\u001b[0m, in \u001b[0;36mrefine\u001b[0;34m(atmsel, actions, cap, timestep, equil_its, equil_equil, equil_temps, sampl_its, sampl_equil, sampl_temps, **args)\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m (its, equil, temps) \u001b[38;5;129;01min\u001b[39;00m ((equil_its, equil_equil, equil_temps),\n\u001b[1;32m     64\u001b[0m                             (sampl_its, sampl_equil, sampl_temps)):\n\u001b[1;32m     65\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m temp \u001b[38;5;129;01min\u001b[39;00m temps:\n\u001b[0;32m---> 66\u001b[0m         \u001b[43mmd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptimize\u001b[49m\u001b[43m(\u001b[49m\u001b[43matmsel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmax_iterations\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mits\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mequilibrate\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mequil\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     67\u001b[0m \u001b[43m                    \u001b[49m\u001b[43mtemperature\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtemp\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43minit_velocities\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minit_vel\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     68\u001b[0m         init_vel \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/modeller/optimizers/molecular_dynamics.py:52\u001b[0m, in \u001b[0;36mMolecularDynamics.optimize\u001b[0;34m(self, atmsel, **vars)\u001b[0m\n\u001b[1;32m     50\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39matmsel \u001b[38;5;241m=\u001b[39m atmsel\n\u001b[1;32m     51\u001b[0m func \u001b[38;5;241m=\u001b[39m _modeller\u001b[38;5;241m.\u001b[39mmod_md_optimize\n\u001b[0;32m---> 52\u001b[0m ret \u001b[38;5;241m=\u001b[39m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_modpt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmdl\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmodpt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43medat\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmodpt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlibs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmodpt\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43minds\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     53\u001b[0m \u001b[43m           \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;28;43mvars\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m     54\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39matmsel \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m     55\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ret\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/modeller/optimizers/actions.py:50\u001b[0m, in \u001b[0;36mTrace.__call__\u001b[0;34m(self, opt)\u001b[0m\n\u001b[1;32m     47\u001b[0m         output \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(output, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mw\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m     48\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moutput \u001b[38;5;241m=\u001b[39m output\n\u001b[0;32m---> 50\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__call__\u001b[39m(\u001b[38;5;28mself\u001b[39m, opt):\n\u001b[1;32m     51\u001b[0m     opt\u001b[38;5;241m.\u001b[39mtrace(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moutput)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "#Code to use MODELLER to reconstruct the sequences that have a gap length of 4 or less, if there are no differences between the SEQRES and ATOM sequences, the original PDB file is copied to the target directory\n",
    "# FIX was to eliminate remarks from the pdb file\n",
    "import os\n",
    "import shutil\n",
    "from Bio.PDB import PDBParser, PPBuilder\n",
    "from modeller import *\n",
    "from modeller.automodel import *\n",
    "\n",
    "def read_fasta_sequences(fasta_file):\n",
    "    \"\"\"Read sequences from a FASTA file into a dictionary.\"\"\"\n",
    "    sequences = {}\n",
    "    with open(fasta_file, \"r\") as f:\n",
    "        lines = f.readlines()\n",
    "        header = None\n",
    "        sequence = []\n",
    "        for line in lines:\n",
    "            line = line.strip()\n",
    "            if line.startswith(\">\"):\n",
    "                if header:\n",
    "                    sequences[header] = ''.join(sequence)\n",
    "                header = line[1:]\n",
    "                sequence = []\n",
    "            else:\n",
    "                sequence.append(line)\n",
    "        if header:\n",
    "            sequences[header] = ''.join(sequence)\n",
    "    return sequences\n",
    "\n",
    "def extract_atom_sequence(pdb_file):\n",
    "    \"\"\"Extract sequence from atomic coordinates for the first chain found in the PDB file.\"\"\"\n",
    "    parser = PDBParser(QUIET=True)\n",
    "    structure = parser.get_structure('PDB', pdb_file)\n",
    "    \n",
    "    for model in structure:\n",
    "        for chain in model:\n",
    "            ppb = PPBuilder()\n",
    "            sequence = ''\n",
    "            for pp in ppb.build_peptides(chain):\n",
    "                sequence += pp.get_sequence()\n",
    "            return str(sequence)\n",
    "    return None\n",
    "\n",
    "def find_motif_indices(sequence, motif):\n",
    "    \"\"\"Find the start index of a motif in a sequence.\"\"\"\n",
    "    index = sequence.find(motif)\n",
    "    return index if index != -1 else None\n",
    "\n",
    "def reconstruct_with_modeller(pdb_chain_id, pdb_path, target_path, full_sequence, atom_sequence):\n",
    "    print(f\"Processing {pdb_chain_id}\")\n",
    "\n",
    "    # Find indices of the DFG and APE motifs\n",
    "    seqres_dfg_index = find_motif_indices(full_sequence, 'DFG')\n",
    "    seqres_ape_index = find_motif_indices(full_sequence, 'APE')\n",
    "    atom_dfg_index = find_motif_indices(atom_sequence, 'DFG')\n",
    "    atom_ape_index = find_motif_indices(atom_sequence, 'APE')\n",
    "\n",
    "    # Determine if reconstruction is needed\n",
    "    if None not in [seqres_dfg_index, seqres_ape_index, atom_dfg_index, atom_ape_index]:\n",
    "        seqres_start = min(seqres_dfg_index, seqres_ape_index)\n",
    "        seqres_end = max(seqres_dfg_index + 3, seqres_ape_index + 3)\n",
    "        atom_start = min(atom_dfg_index, atom_ape_index)\n",
    "        atom_end = max(atom_dfg_index + 3, atom_ape_index + 3)\n",
    "\n",
    "        seqres_segment = full_sequence[seqres_start:seqres_end]\n",
    "        atom_segment = atom_sequence[atom_start:atom_end]\n",
    "\n",
    "        # Check for differences in the segment\n",
    "        if seqres_segment != atom_segment:\n",
    "            print(f\"Reconstructing full sequence for {pdb_chain_id} using MODELLER\")\n",
    "\n",
    "            # Setting up MODELLER\n",
    "            env = environ()\n",
    "            aln = alignment(env)\n",
    "            \n",
    "            # Read the structure to work on\n",
    "            mdl = model(env, file=pdb_path)\n",
    "            aln.append_model(mdl, align_codes='template', atom_files=pdb_path)\n",
    "\n",
    "            # Append the full target sequence\n",
    "            aln.append_sequence(full_sequence)\n",
    "            aln[-1].code = 'target'\n",
    "            \n",
    "            # Perform the alignment\n",
    "            aln.align2d(max_gap_length=50)\n",
    "\n",
    "            # Create AutoModel object and build models\n",
    "            a = automodel(env, alnfile=aln, knowns='template', sequence='target')\n",
    "            a.starting_model = 1\n",
    "            a.ending_model = 1\n",
    "            \n",
    "            # Build the model\n",
    "            a.make()\n",
    "            \n",
    "            # Save the best model to the target directory\n",
    "            model_path = os.path.join(target_path, f\"{pdb_chain_id}_filled.pdb\")\n",
    "            os.rename(a.outputs[0]['name'], model_path)\n",
    "            print(f\"Reconstruction completed for {pdb_chain_id}. File saved at {model_path}\")\n",
    "\n",
    "            # Post-process to remove REMARK lines\n",
    "            remove_remark_lines(model_path)\n",
    "        else:\n",
    "            # No reconstruction needed, copy original PDB\n",
    "            shutil.copy(pdb_path, os.path.join(target_path, f\"{pdb_chain_id}.pdb\"))\n",
    "            print(f\"No differences found for {pdb_chain_id}. Original PDB copied to target directory.\")\n",
    "    else:\n",
    "        print(f\"Motifs not found in {pdb_chain_id}.\")\n",
    "\n",
    "def remove_remark_lines(pdb_file):\n",
    "    \"\"\"Remove lines starting with 'REMARK' from the PDB file.\"\"\"\n",
    "    with open(pdb_file, 'r') as file:\n",
    "        lines = file.readlines()\n",
    "    \n",
    "    with open(pdb_file, 'w') as file:\n",
    "        for line in lines:\n",
    "            if not line.startswith(\"REMARK\"):\n",
    "                file.write(line)\n",
    "    \n",
    "    print(f\"Removed REMARK lines from {pdb_file}\")\n",
    "\n",
    "def main():\n",
    "    seqres_fasta = \"seqres_sequences.fasta\"\n",
    "    pdb_dir = \"Results/activation_segments/unaligned\"\n",
    "    target_dir = \"Results/activation_segments/reconstructedModeller\"\n",
    "\n",
    "    # Read the sequences from the FASTA file\n",
    "    seqres_sequences = read_fasta_sequences(seqres_fasta)\n",
    "\n",
    "    for header, full_sequence in seqres_sequences.items():\n",
    "        if \"_SEQRES\" in header:  # Only consider SEQRES entries\n",
    "            pdb_chain_id = header.replace(\"_SEQRES\", \"\")\n",
    "            pdb_file_path = os.path.join(pdb_dir, f\"{pdb_chain_id}.pdb\")\n",
    "\n",
    "            # Extract the atomic sequence\n",
    "            atom_sequence = extract_atom_sequence(pdb_file_path)\n",
    "\n",
    "            if atom_sequence is None:\n",
    "                print(f\"Could not extract sequence for {pdb_chain_id}. Skipping...\")\n",
    "                continue\n",
    "\n",
    "            if not os.path.exists(target_dir):\n",
    "                os.makedirs(target_dir)\n",
    "            \n",
    "            reconstruct_with_modeller(pdb_chain_id, pdb_file_path, target_dir, full_sequence, atom_sequence)\n",
    "\n",
    "    print(\"Processing complete!\")\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "fccfa004",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 323 PDB files in the directory 'Results/activation_segments/reconstructedModeller'.\n"
     ]
    }
   ],
   "source": [
    "#Counting the number of pdb files in the directory after reconstruction\n",
    "import glob\n",
    "import os\n",
    "def count_pdb_files(directory):\n",
    "    # Ensure the directory path ends with a slash\n",
    "    directory = os.path.join(directory, '')\n",
    "\n",
    "    # Use glob to find all .pdb files in the directory\n",
    "    pdb_files = glob.glob(os.path.join(directory, '*.pdb'))\n",
    "\n",
    "    # Return the count of .pdb files\n",
    "    return len(pdb_files)\n",
    "\n",
    "# Specify the directory\n",
    "pdb_directory = 'Results/activation_segments/reconstructedModeller'\n",
    "\n",
    "# Get the count of PDB files\n",
    "pdb_count = count_pdb_files(pdb_directory)\n",
    "\n",
    "print(f\"There are {pdb_count} PDB files in the directory '{pdb_directory}'.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "119237e7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "1cd0ead83d3741dab9c395875078d070",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": []
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'Results/activation_segments/unaligned/7OPO_A.pdb'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[14], line 44\u001b[0m\n\u001b[1;32m     41\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m index \u001b[38;5;28;01mif\u001b[39;00m index \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m     43\u001b[0m \u001b[38;5;66;03m# Extract the sequence and mapping for the single chain\u001b[39;00m\n\u001b[0;32m---> 44\u001b[0m atom_sequence, index_to_resid \u001b[38;5;241m=\u001b[39m \u001b[43mextract_sequence_and_mapping\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mResults/activation_segments/unaligned/7OPO_A.pdb\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m     46\u001b[0m \u001b[38;5;66;03m# Find indices of DFG and APE motifs\u001b[39;00m\n\u001b[1;32m     47\u001b[0m dfg_index \u001b[38;5;241m=\u001b[39m find_motif_indices(atom_sequence, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mDFG\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "Cell \u001b[0;32mIn[14], line 19\u001b[0m, in \u001b[0;36mextract_sequence_and_mapping\u001b[0;34m(pdb_file)\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Extract sequence and create a mapping from sequence index to PDB residue ID.\"\"\"\u001b[39;00m\n\u001b[1;32m     18\u001b[0m parser \u001b[38;5;241m=\u001b[39m PDBParser(QUIET\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m---> 19\u001b[0m structure \u001b[38;5;241m=\u001b[39m \u001b[43mparser\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_structure\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mPDB\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpdb_file\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     21\u001b[0m sequence \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m     22\u001b[0m index_to_resid \u001b[38;5;241m=\u001b[39m {}\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/PDB/PDBParser.py:96\u001b[0m, in \u001b[0;36mPDBParser.get_structure\u001b[0;34m(self, id, file)\u001b[0m\n\u001b[1;32m     93\u001b[0m \u001b[38;5;66;03m# Make a StructureBuilder instance (pass id of structure as parameter)\u001b[39;00m\n\u001b[1;32m     94\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstructure_builder\u001b[38;5;241m.\u001b[39minit_structure(\u001b[38;5;28mid\u001b[39m)\n\u001b[0;32m---> 96\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m as_handle(file) \u001b[38;5;28;01mas\u001b[39;00m handle:\n\u001b[1;32m     97\u001b[0m     lines \u001b[38;5;241m=\u001b[39m handle\u001b[38;5;241m.\u001b[39mreadlines()\n\u001b[1;32m     98\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m lines:\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/contextlib.py:135\u001b[0m, in \u001b[0;36m_GeneratorContextManager.__enter__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    133\u001b[0m \u001b[38;5;28;01mdel\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39margs, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mkwds, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfunc\n\u001b[1;32m    134\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 135\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mnext\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mgen\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    136\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mStopIteration\u001b[39;00m:\n\u001b[1;32m    137\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mRuntimeError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgenerator didn\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mt yield\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[0;32m~/miniforge3/envs/molearn/lib/python3.10/site-packages/Bio/File.py:72\u001b[0m, in \u001b[0;36mas_handle\u001b[0;34m(handleish, mode, **kwargs)\u001b[0m\n\u001b[1;32m     30\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Context manager to ensure we are using a handle.\u001b[39;00m\n\u001b[1;32m     31\u001b[0m \n\u001b[1;32m     32\u001b[0m \u001b[38;5;124;03mContext manager for arguments that can be passed to SeqIO and AlignIO read, write,\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     69\u001b[0m \n\u001b[1;32m     70\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m---> 72\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mhandleish\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m \u001b[38;5;28;01mas\u001b[39;00m fp:\n\u001b[1;32m     73\u001b[0m         \u001b[38;5;28;01myield\u001b[39;00m fp\n\u001b[1;32m     74\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'Results/activation_segments/unaligned/7OPO_A.pdb'"
     ]
    }
   ],
   "source": [
    "#Fast checking if reconstruction was successful, need to just change the pdb id and chain id\n",
    "import MDAnalysis as mda\n",
    "import nglview as nv\n",
    "from Bio.PDB import PDBParser\n",
    "from MDAnalysis.analysis import align\n",
    "\n",
    "# Dictionary to convert three-letter amino acid codes to one-letter codes\n",
    "three_to_one = {\n",
    "    'ALA': 'A', 'ARG': 'R', 'ASN': 'N', 'ASP': 'D',\n",
    "    'CYS': 'C', 'GLU': 'E', 'GLN': 'Q', 'GLY': 'G',\n",
    "    'HIS': 'H', 'ILE': 'I', 'LEU': 'L', 'LYS': 'K',\n",
    "    'MET': 'M', 'PHE': 'F', 'PRO': 'P', 'SER': 'S',\n",
    "    'THR': 'T', 'TRP': 'W', 'TYR': 'Y', 'VAL': 'V'\n",
    "}\n",
    "\n",
    "def extract_sequence_and_mapping(pdb_file):\n",
    "    \"\"\"Extract sequence and create a mapping from sequence index to PDB residue ID.\"\"\"\n",
    "    parser = PDBParser(QUIET=True)\n",
    "    structure = parser.get_structure('PDB', pdb_file)\n",
    "    \n",
    "    sequence = []\n",
    "    index_to_resid = {}\n",
    "    \n",
    "    for model in structure:\n",
    "        for chain in model:\n",
    "            for residue in chain:\n",
    "                if 'CA' in residue:  # Check if it's an amino acid\n",
    "                    resname = residue.get_resname()\n",
    "                    if resname in three_to_one:\n",
    "                        sequence.append(three_to_one[resname])\n",
    "                        index_to_resid[len(sequence) - 1] = residue.get_id()[1]  # Map sequence index to PDB resid\n",
    "            break\n",
    "        break\n",
    "\n",
    "    return sequence, index_to_resid\n",
    "\n",
    "def find_motif_indices(sequence, motif):\n",
    "    \"\"\"Find the start index of a motif in a sequence.\"\"\"\n",
    "    sequence_str = ''.join(sequence)\n",
    "    index = sequence_str.find(motif)\n",
    "    return index if index != -1 else None\n",
    "\n",
    "# Extract the sequence and mapping for the single chain\n",
    "atom_sequence, index_to_resid = extract_sequence_and_mapping(\"Results/activation_segments/unaligned/7OPO_A.pdb\")\n",
    "\n",
    "# Find indices of DFG and APE motifs\n",
    "dfg_index = find_motif_indices(atom_sequence, 'DFG')\n",
    "ape_index = find_motif_indices(atom_sequence, 'APE')\n",
    "\n",
    "# Ensure indices are found and select the residues between them\n",
    "if dfg_index is not None and ape_index is not None:\n",
    "    # Use the mapping to get the correct residue IDs\n",
    "    dfg_resid = index_to_resid[dfg_index]\n",
    "    ape_resid = index_to_resid[ape_index + 2]  # +2 to include the entire 'APE' motif\n",
    "\n",
    "    u_missing = mda.Universe(\"Results/activation_segments/unaligned/7OPO_A.pdb\")\n",
    "    selected_atoms = u_missing.select_atoms(f\"resid {dfg_resid}:{ape_resid}\")\n",
    "\n",
    "    print(\"Number of Atoms Selected:\", selected_atoms.n_atoms)\n",
    "\n",
    "    u_reconstructed = mda.Universe(\"Results/activation_segments/reconstructedModeller/7OPO_A_filled.pdb\")\n",
    "    print(\"Number of Atoms Reconstructed:\", u_reconstructed.select_atoms(f\"all\").n_atoms)\n",
    "\n",
    "    # Merge the aligned atoms for visualization\n",
    "    merged = mda.Merge(selected_atoms, u_reconstructed.atoms)\n",
    "    print(merged.residues)\n",
    "\n",
    "    # Create NGLView widget\n",
    "    w = nv.show_mdanalysis(merged)\n",
    "\n",
    "    # Add a representation for each residue name with the corresponding color\n",
    "    w.clear()\n",
    "    w.add_cartoon(color=\"resname\")\n",
    "\n",
    "    \n",
    "\n",
    "else:\n",
    "    print(\"Motifs not found in the sequence.\")\n",
    "\n",
    "w"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "766c8fc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Helper functions and function to run MUSTANG on the reconstructed pdb files\n",
    "\n",
    "import subprocess\n",
    "import os\n",
    "from glob import glob as g\n",
    "import mdtraj as md\n",
    "from mpi4py import MPI\n",
    "from time import time as t\n",
    "from tqdm import tqdm\n",
    "\n",
    "\n",
    "def sg(f_p):\n",
    "    return sorted(g(f_p))\n",
    "\n",
    "\n",
    "def find_pdbs(directory):\n",
    "    \"\"\"\n",
    "    Find topologies in a directory.\n",
    "    Currently excludes cif files.\n",
    "    \"\"\"\n",
    "    return sg(directory+\"/*.pdb\")\n",
    "\n",
    "\n",
    "def fname(file):\n",
    "    return file.rsplit(\".\", 1)[0].rsplit(\"/\", 1)[-1]\n",
    "\n",
    "\n",
    "def ifnotmake(dir_path):\n",
    "    if not os.path.isdir(dir_path):\n",
    "        os.makedirs(dir_path)\n",
    "    return dir_path\n",
    "\n",
    "\n",
    "def run_mustang(f1, f2, name=None):\n",
    "    \"\"\"\n",
    "    Writes a MUSTANG input file which aligns\n",
    "    file1 to file 2.\n",
    "    If no name defaults to the second file.\n",
    "    \"\"\"\n",
    "    if name is None:\n",
    "        name = fname(f2)\n",
    "    if not os.path.isdir(f\"./{name}\"):\n",
    "        os.makedirs(f\"./{name}\")\n",
    "    new_fp = f\"./{name}/{name}\"\n",
    "    structs = f\"{f1} {f2} \"\n",
    "    command = f\"/home/marmatt/Downloads/MUSTANG_v3.2.4/bin/mustang-3.2.4 -i {structs} -o {new_fp} -F fasta -s ON\" #here you should change the path to where you install MUSTANG\n",
    "    command = command.split()\n",
    "    new_fp = f\"{new_fp}.pdb\"\n",
    "    try:\n",
    "        result = subprocess.run(command, capture_output=True, text=True)\n",
    "        print(f\"Running command: {' '.join(command)}\")\n",
    "        print(f\"STDOUT: {result.stdout}\")\n",
    "        print(f\"STDERR: {result.stderr}\")\n",
    "        if result.returncode != 0:\n",
    "            print(\"Error in MUSTANG execution\")\n",
    "            return None\n",
    "    except Exception as e:\n",
    "        print(f\"Exception occurred: {e}\")\n",
    "        return None\n",
    "    return new_fp\n",
    "\n",
    "\n",
    "def postprocess(file_path):\n",
    "    \"\"\"\n",
    "    file_path is the name of a pdb file.\n",
    "    It deletes the first chain which is always the alignment structures\n",
    "    \"\"\"\n",
    "    structure = md.load(file_path)\n",
    "    aligned_chain_idx = [[atom.index for atom in res.atoms] for res in\n",
    "                         structure.top._chains[1]._residues]\n",
    "    aligned_chain_idx = sum(aligned_chain_idx, [])\n",
    "    structure = structure.atom_slice(aligned_chain_idx)\n",
    "    structure.save(file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "993598b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/Results/activation_segments/mustangs/Results/activation_segments/mustangs/Results/activation_segments/reconstructedModeller /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/Results/activation_segments/mustangs/Results/activation_segments/mustangs/Results/activation_segments/mustangs /home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/Results/activation_segments/mustangs/Results/activation_segments/mustangs/6UAN_chainD.pdb\n",
      "/home/marmatt/Documents/projects/BRAF/myWork/gitTrials/BRAF/Results/activation_segments/mustangs/Results/activation_segments/mustangs/Results/activation_segments/mustangs\n"
     ]
    },
    {
     "ename": "ZeroDivisionError",
     "evalue": "division by zero",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mZeroDivisionError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[34], line 28\u001b[0m\n\u001b[1;32m     26\u001b[0m n_files \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(pdbs)\n\u001b[1;32m     27\u001b[0m n_slices \u001b[38;5;241m=\u001b[39m (n_files \u001b[38;5;241m/\u001b[39m\u001b[38;5;241m/\u001b[39m size)\n\u001b[0;32m---> 28\u001b[0m step \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mint\u001b[39m(\u001b[43mn_files\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m/\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mn_slices\u001b[49m)\n\u001b[1;32m     29\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m n_files \u001b[38;5;241m%\u001b[39m n_slices \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m     30\u001b[0m     n_slices \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n",
      "\u001b[0;31mZeroDivisionError\u001b[0m: division by zero"
     ]
    }
   ],
   "source": [
    "#Perform MUSTANG alignment on the sequences that have not been reconstructed, I need to fix this bug\n",
    "\n",
    "#from mustang import *\n",
    "import subprocess\n",
    "from mpi4py import MPI\n",
    "\n",
    "# Constants (most to be made variable)\n",
    "\n",
    "comm = MPI.COMM_WORLD\n",
    "rank = comm.Get_rank()\n",
    "size = comm.Get_size()\n",
    "# Define the paths explicitly\n",
    "pdb_path = \"Results/activation_segments/reconstructedModeller\"\n",
    "target_dir = \"Results/activation_segments/mustangs\"\n",
    "template_pdb = \"6UAN_chainD.pdb\"\n",
    "# Ensure the target directory exists\n",
    "ifnotmake(target_dir)\n",
    "pdb_path = os.path.abspath(pdb_path)\n",
    "target_dir = os.path.abspath(target_dir)\n",
    "template_pdb = os.path.abspath(template_pdb)\n",
    "print(pdb_path, target_dir, template_pdb)\n",
    "os.chdir(target_dir)\n",
    "os.system(\"pwd\")\n",
    "if rank == 0:\n",
    "    pdbs = find_pdbs(pdb_path)\n",
    "    n_files = len(pdbs)\n",
    "    n_slices = (n_files // size)\n",
    "    step = int(n_files / n_slices)\n",
    "    if n_files % n_slices != 0:\n",
    "        n_slices += 1\n",
    "    slices = [slice(i*n_slices, (i+1)*n_slices) for i in range(step)]\n",
    "    pdbs = [pdbs[s] for s in slices]\n",
    "else:\n",
    "    pdbs = None\n",
    "\n",
    "pdbs = comm.scatter(pdbs, root=0)\n",
    "print(\"RANK:\\t\", rank, \"DATA SIZE:\\t\", len(pdbs))\n",
    "t1 = t()\n",
    "failures = []\n",
    "for pdb in tqdm(pdbs):\n",
    "    name = fname(pdb)\n",
    "    new_fp = run_mustang(template_pdb, pdb, name=name)\n",
    "    if new_fp:\n",
    "        if os.path.isfile(new_fp):\n",
    "            postprocess(new_fp)\n",
    "        else:\n",
    "            failures.append(pdb)\n",
    "    else:\n",
    "        failures.append(pdb)\n",
    "t2 = t()\n",
    "print(\"FINISHED RANK:\\t\", rank, \"DATA SIZE:\\t\", len(pdbs),\n",
    "      \"TIME:\\t\", round(t2-t1, 4))\n",
    "failures = comm.gather(failures, root=0)\n",
    "if rank == 0:\n",
    "    failures = sum(failures, [])\n",
    "    with open(\"./failures.txt\", \"w\") as f_o:\n",
    "        f_o.write(\"\\n\".join(f for f in failures))\n",
    "    t2 = t()\n",
    "    print(round(t2-t1, 3))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "14388f28",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 1 directories in the directory 'Results/activation_segments/mustangs'.\n"
     ]
    }
   ],
   "source": [
    "#Counting the number of directories representing the number of pdb files that have been aligned\n",
    "import os\n",
    "def count_directories(directory):\n",
    "    # List all entries in the given directory\n",
    "    entries = os.listdir(directory)\n",
    "\n",
    "    # Use os.path.join to get the full path and os.path.isdir to check if it's a directory\n",
    "    directories = [entry for entry in entries if os.path.isdir(os.path.join(directory, entry))]\n",
    "\n",
    "    # Return the count of directories\n",
    "    return len(directories)\n",
    "\n",
    "# Specify the directory\n",
    "directory_path = 'Results/activation_segments/mustangs'\n",
    "\n",
    "# Get the count of directories\n",
    "directory_count = count_directories(directory_path)\n",
    "\n",
    "print(f\"There are {directory_count} directories in the directory '{directory_path}'.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "a3ce23ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 0 PDB files without '_filled' in the directory 'Results/activation_segments/reconstructedModeller'.\n"
     ]
    }
   ],
   "source": [
    "#Counting the number of files representing the number of pdb files that did not need reconstruction\n",
    "import os\n",
    "from glob import glob as g\n",
    "\n",
    "def count_non_filled_pdbs(directory):\n",
    "    # Find all PDB files in the directory\n",
    "    pdb_files = g(os.path.join(directory, \"*.pdb\"))\n",
    "\n",
    "    # Filter out files with '_filled' in their names\n",
    "    non_filled_pdbs = [pdb for pdb in pdb_files if \"_filled\" not in os.path.basename(pdb)]\n",
    "\n",
    "    # Return the count of non '_filled' PDB files\n",
    "    return len(non_filled_pdbs)\n",
    "\n",
    "# Specify the directory\n",
    "pdb_directory_path = 'Results/activation_segments/reconstructedModeller'\n",
    "\n",
    "# Get the count of non '_filled' PDB files\n",
    "non_filled_pdb_count = count_non_filled_pdbs(pdb_directory_path)\n",
    "\n",
    "print(f\"There are {non_filled_pdb_count} PDB files without '_filled' in the directory '{pdb_directory_path}'.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "c47016e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import os\n",
    "import h5py\n",
    "import numpy as np\n",
    "import mdtraj as md\n",
    "import pickle as p\n",
    "from tqdm import tqdm\n",
    "from glob import glob\n",
    "from pprint import pprint as pp\n",
    "alignment_dir = \"Results/activation_segments/mustangs\"\n",
    "\n",
    "class alignment:\n",
    "    \"\"\"\n",
    "    Class to hold alignments.\n",
    "    Currently only supports braf_monomers!\n",
    "    \"\"\"\n",
    "    def __init__(self,name,seq1,seq2):\n",
    "        self.name = name\n",
    "        self.seq1 = seq1\n",
    "        self.seq2 = seq2\n",
    "        self.aligned = self.find_aligned()\n",
    "\n",
    "    def find_pdb(self):\n",
    "        pdb_files = []\n",
    "        for root, dirs, files in os.walk(alignment_dir):\n",
    "            pdb_files += [os.path.join(root, file) for file in files if file.endswith('.pdb')]\n",
    "        \n",
    "        pdb = [f for f in pdb_files if self.name in f]\n",
    "        print(pdb)\n",
    "        if len(pdb) == 1:\n",
    "            return pdb[0]\n",
    "        \n",
    "\n",
    "    def find_aligned(self):\n",
    "        aligned = []\n",
    "        for char1, char2 in zip(self.seq1,self.seq2):\n",
    "            if char1 != \"-\":\n",
    "                aligned.append((char1,char2))\n",
    "        return aligned\n",
    "\n",
    "    def aligned_res(self):\n",
    "        seq1, seq2 = self.seq1, self.seq2\n",
    "        aligned = [[*item] for item in self.aligned]\n",
    "        seq_length = len(aligned)\n",
    "        pdb2_top = self.load_pdb()\n",
    "        full_seq2 = \"\".join(char for char in seq2 if char != \"-\")\n",
    "        residues = pdb2_top.top._residues\n",
    "        n_res = len(residues)\n",
    "        res_counter = 0\n",
    "        for i in range(seq_length):\n",
    "            if res_counter >= n_res:\n",
    "                break\n",
    "            if aligned[i][1] != \"-\":\n",
    "                aligned[i][1] = residues[res_counter]\n",
    "                res_counter += 1\n",
    "            else:\n",
    "                continue\n",
    "        return [tuple(a) for a in aligned]\n",
    "\n",
    "    def aligned_xyz(self):\n",
    "        \"\"\"\n",
    "        Return xyz of aligned residues\n",
    "        \"\"\"\n",
    "        xyz = self.load_pdb()._xyz[0] # Only one frame\n",
    "        aligned = [[*item] for item in self.aligned]\n",
    "        for k,(_,res) in enumerate(self.residues):\n",
    "            if not isinstance(res,str):\n",
    "                idxs = []\n",
    "                for atom in res._atoms:\n",
    "                    idxs.append(atom.index)\n",
    "                res_xyz = xyz[idxs]\n",
    "                aligned[k][1] = res_xyz\n",
    "            else:\n",
    "                continue\n",
    "        return [tuple(a) for a in aligned]\n",
    "\n",
    "    def aligned_ca_xyz(self):\n",
    "        \"\"\"\n",
    "        Return xyz of aligned residues\n",
    "        \"\"\"\n",
    "        xyz = self.load_pdb()._xyz[0] # Only one frame\n",
    "        aligned = [[*item] for item in self.aligned]\n",
    "        for k,(_,res) in enumerate(self.residues):\n",
    "            if not isinstance(res,str):\n",
    "                for atom in res._atoms:\n",
    "                    if atom.name == \"CA\":\n",
    "                        idxs = atom.index\n",
    "                        break\n",
    "                try:\n",
    "                    res_xyz = xyz[idxs]\n",
    "                except Exception as e:\n",
    "                    print(e)\n",
    "                    print(\"ERROR FOR:\")\n",
    "                    print(self.name)\n",
    "                    return None\n",
    "                aligned[k][1] = res_xyz\n",
    "            else:\n",
    "                continue\n",
    "        return [tuple(a) for a in aligned]\n",
    "\n",
    "    def load_pdb(self):\n",
    "        return md.load(self.pdb_file)\n",
    "\n",
    "    def __getitem__(self,idx):\n",
    "        return (self.seq1[idx],self.seq2[idx])\n",
    "\n",
    "    def __repr__(self):\n",
    "        return self.name\n",
    "\n",
    "    def find_match_id(self):\n",
    "        seq1, seq2 = self.seq1, self.seq2\n",
    "        full_seq2 = \"\".join(char for char in seq2 if char != \"-\")\n",
    "        pp(full_seq2)\n",
    "        aligned = self.aligned\n",
    "        actv_low = 155\n",
    "        actv_hgh = 181\n",
    "        match_residues = aligned[actv_low:actv_hgh] # These are what we need\n",
    "        seq2_Seq = [a[1] for a in match_residues if a[1] != \"-\"]\n",
    "        begin_idx = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "19861f06",
   "metadata": {},
   "outputs": [],
   "source": [
    "def afasta_parse(file):\n",
    "    \"\"\"\n",
    "    Parse mustang afasta format output file.\n",
    "    Returns two lists of equal length\n",
    "    \"\"\"\n",
    "    with open(file,\"r\") as f:\n",
    "        lines = f.readlines()\n",
    "    names = [l.split(\".\")[0][1:] for l in lines if l[0] == \">\"]\n",
    "    for i in range(1,len(lines)):\n",
    "        if lines[i].isspace():\n",
    "            lines[i] = \"BREAK\"\n",
    "            break\n",
    "        elif lines[i][0] == \">\":\n",
    "            lines[i] = \"BREAK\" + lines[i]\n",
    "            break\n",
    "    lines = [l.strip() for l in lines if l[0] != \">\"]\n",
    "    lines = \"\".join(lines)\n",
    "    fastas = lines.split(\"BREAK\")\n",
    "    fastas = [*filter(None,fastas)]\n",
    "    return fastas[0], fastas[1]\n",
    "\n",
    "def load_alignments(kind=\"mustang\"):\n",
    "    if kind==\"mustang\":\n",
    "        #ppath = \"/home/marmatt/Documents/projects/BRAF/myWork/reproduceBRAFWork/Results/activation_segments/mustangs/mustang_alignments.fasta\"\n",
    "        ppath = \"Results/activation_segments/mustangs/mustang_alignments.fasta\"\n",
    "        \n",
    "        #print(\"Loading pickled alignments...\")\n",
    "    elif kind==\"blast\":\n",
    "        ppath = \"blast_alignments.fasta\"\n",
    "    if os.path.isfile(ppath):\n",
    "        with open(ppath,\"rb\") as pickled:\n",
    "            #print(\"Loading pickled alignments...\")\n",
    "            return p.load(pickled)\n",
    "    else:\n",
    "        make_align_pickle()\n",
    "        #print(\"No pickled alignments found. Creating...\")\n",
    "        return load_alignments()\n",
    "    \n",
    "def make_align_pickle(kind=\"mustang\"):\n",
    "    if kind == \"mustang\":\n",
    "\n",
    "        alignments = []\n",
    "\n",
    "        # Iterate over directories in the alignment directory\n",
    "        for directory_name in os.listdir(alignment_dir):\n",
    "            directory_path = os.path.join(alignment_dir, directory_name)\n",
    "\n",
    "            # Ensure we are working with directories\n",
    "            if os.path.isdir(directory_path):\n",
    "                #print(f\"Processing directory: {directory_name}\")\n",
    "                fasta_files = tqdm(glob(os.path.join(directory_path, \"*.afasta\")), desc=f\"Processing {directory_name} .afasta files\")\n",
    "                \n",
    "                for fasta_file in fasta_files:\n",
    "                    name = os.path.splitext(os.path.basename(fasta_file))[0]\n",
    "                    fasta_files.set_description(f\"Working on {name}\")\n",
    "\n",
    "                    # Simulate the alignment logic\n",
    "                    aligned = alignment(name, *afasta_parse(fasta_file))  # Assuming `alignment` and `afasta_parse` are predefined\n",
    "                    alignments.append(aligned)\n",
    "\n",
    "        # Define a path for the output pickle file\n",
    "        ppath = os.path.join(alignment_dir, \"mustang_alignments.fasta\")\n",
    "        with open(ppath, \"wb\") as pickled:\n",
    "            p.dump(alignments, pickled)\n",
    "    \n",
    "    elif kind == \"blast\":\n",
    "        b = BLAST_results()\n",
    "        alignments = []\n",
    "        for k, dicti in tqdm(b.alignments.items(),total=len(b.alignments)):\n",
    "            seq1 = dicti[\"Query\"]\n",
    "            seq2 = dicti[\"Subject\"]\n",
    "            alignments.append(alignment(k,seq1,seq2))\n",
    "        ppath = \"blast_alignments.fasta\"\n",
    "        with open(ppath, \"wb\") as pickled:\n",
    "            p.dump(alignments, pickled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "40a51087",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4bb52fb6640f4db1b60644522676552c",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Processing Results .afasta files: 0it [00:00, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pickle\n",
    "import numpy as np\n",
    "from matplotlib import pyplot as plt\n",
    "from sklearn.decomposition import PCA\n",
    "from glob import glob as g\n",
    "import os\n",
    "import matplotlib as mpl\n",
    "from tqdm.notebook import tqdm\n",
    "mpl.rcParams['figure.dpi'] = 300\n",
    "mpl.rcParams.update({'font.size': 8})\n",
    "kind = \"mustang\"\n",
    "make_align_pickle(kind) #create MSA file --> important, substituted .p with .html in alignment class\n",
    "aligned = load_alignments(kind)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e0a71ccc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Some helper functions to extract the sequence from the pdb files and to extract the sequence from the alignment files\n",
    "def braf_res():\n",
    "    fp = \"./6UAN_chainD.pdb\"\n",
    "    top = md.load(fp).top\n",
    "    return [res_namer(res) for res in top.residues]\n",
    "\n",
    "\n",
    "def res_namer(res):\n",
    "    return f\"{res.name}-{res.resSeq}\"\n",
    "\n",
    "def fname(fp):\n",
    "    return fp.rsplit(\".\",1)[0].rsplit(\"/\",1)[-1]\n",
    "\n",
    "def make_seg(a):\n",
    "    seq = [t for t in a.aligned if t[0] != \"-\"]\n",
    "    return seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "8f3ed920",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[]\n"
     ]
    },
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[31], line 6\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mmatplotlib\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mticker\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m FuncFormatter\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28mprint\u001b[39m(aligned)\n\u001b[0;32m----> 6\u001b[0m seq1mag \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(\u001b[43maligned\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241m.\u001b[39mseq1\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m-\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m))\n\u001b[1;32m      7\u001b[0m counts \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mzeros(seq1mag)\n\u001b[1;32m      8\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m a \u001b[38;5;129;01min\u001b[39;00m aligned:\n",
      "\u001b[0;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Plotting the number of aligned residues over the BRAF BLAST search results to show what are the most conserved residues throughout the alignment\n",
    "\"\"\"\n",
    "from matplotlib.ticker import FuncFormatter\n",
    "print(aligned)\n",
    "seq1mag = len(aligned[0].seq1.replace(\"-\",\"\"))\n",
    "counts = np.zeros(seq1mag)\n",
    "for a in aligned:\n",
    "    segment = make_seg(a)\n",
    "    for i,(b,c) in enumerate(segment):\n",
    "        if c != \"-\":\n",
    "            counts[i] += 1\n",
    "counts =  counts / max(counts)\n",
    "# sns.set_theme(style=\"whitegrid\")\n",
    "fig,ax = plt.subplots(1,figsize=(10,5))\n",
    "x = [*range(len(counts))]\n",
    "ax.set_xticks(x[::5])\n",
    "x_lbl = braf_res()\n",
    "ax.set_xticklabels(x_lbl[::5],rotation=90,fontsize=7)\n",
    "# subtract 11\n",
    "ax.axvspan(36,48, facecolor='g', alpha=0.5)\n",
    "ax.axvspan(92,100, facecolor='c', alpha=0.5)\n",
    "ax.axvspan(144,168, facecolor='r', alpha=0.5)\n",
    "ax.axvspan(177,186, facecolor='y', alpha=0.5)\n",
    "ax.axvspan(204,215, facecolor='pink', alpha=0.8)\n",
    "ax.axvspan(222,240, facecolor='dodgerblue', alpha=0.8)\n",
    "ax.bar(x,counts,linewidth=0.05,width=1)\n",
    "ax.yaxis.set_major_formatter(FuncFormatter(lambda y, _: '{:.0%}'.format(y))) \n",
    "title = ax.set_title(\"Number of MUSTANG aligned residues over the BRAF BLAST search results\")\n",
    "ax1 = plt.xlabel(\"Resiude Name-Number\")\n",
    "ax1 = plt.ylabel(\"Percent matching in structural alignments\")\n",
    "ax.tick_params(length=2,color=\"black\",direction=\"out\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "792d1493",
   "metadata": {},
   "outputs": [],
   "source": [
    "\"\"\"\n",
    "This cell finds PDBs with DFG and APE motifs aligned.\n",
    "Need to look over this to understand if completely necessary.\n",
    "\"\"\"\n",
    "from collections import Counter\n",
    "import csv\n",
    "\n",
    "# Initialize counters and lists\n",
    "count = 0\n",
    "new_aligned = []\n",
    "bad = []\n",
    "\n",
    "# First filtering loop\n",
    "for k, a in enumerate(aligned):\n",
    "    if a.seq1.find(\"DFG\") == -1 or a.seq1.find(\"APE\") == -1:\n",
    "        bad.append(a)\n",
    "        count += 1\n",
    "    else:\n",
    "        new_aligned.append(a)\n",
    "\n",
    "print(count, \"/\", len(aligned), \" don't match.\")\n",
    "print(f\"Continuing with {len(new_aligned)} structures\")\n",
    "\n",
    "# Second filtering and analysis loop\n",
    "counts = {}\n",
    "counter = 0\n",
    "new_aligned2 = []\n",
    "aligning_segs = {}\n",
    "lengths = []\n",
    "\n",
    "for a in new_aligned:\n",
    "    DFG_index = a.seq1.find(\"DFG\")\n",
    "    APE_index = a.seq1.find(\"APE\")\n",
    "    dfg = a.seq2[DFG_index]\n",
    "    ape = a.seq2[APE_index]\n",
    "    \n",
    "    if dfg != \"-\" and ape != \"-\":\n",
    "        counter += 1\n",
    "        length = APE_index - DFG_index\n",
    "        counts.setdefault(length, []).append((a.name, DFG_index, APE_index))\n",
    "        new_aligned2.append(a)\n",
    "        aligning_segs.setdefault(a.name,\n",
    "                                 (a.seq2[DFG_index:DFG_index+3],\n",
    "                                  a.seq2[APE_index:APE_index+3],\n",
    "                                  a.seq2[DFG_index:APE_index+3]))\n",
    "        lengths.append(len(a.seq2[DFG_index:APE_index+3]))\n",
    "\n",
    "print(f\"{counter} structures with an alignment to the D and A\")\n",
    "\n",
    "# Write to CSV\n",
    "with open('alignment_results.csv', mode='w', newline='') as file:\n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow(['Name', 'Seq1 Segment', 'Seq2 Segment', 'Status'])\n",
    "\n",
    "    # Write new_aligned2\n",
    "    for a in new_aligned2:\n",
    "        DFG_index = a.seq1.find(\"DFG\")\n",
    "        APE_index = a.seq1.find(\"APE\")\n",
    "        seq1_segment = a.seq1[DFG_index:APE_index+3]\n",
    "        seq2_segment = a.seq2[DFG_index:APE_index+3]\n",
    "        writer.writerow([a.name, seq1_segment, seq2_segment, 'Aligned'])\n",
    "\n",
    "    # Write bad\n",
    "    for a in bad:\n",
    "        writer.writerow([a.name, '', '', 'Not Aligned'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9b5bd0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pymol\n",
    "from pymol import cmd\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "\n",
    "# Initialize PyMOL\n",
    "print(\"Initializing PyMOL...\")\n",
    "pymol.finish_launching(['pymol', '-qc'])  # Launch PyMOL quietly\n",
    "\n",
    "alignment_dir = \"Results/activation_segments/mustangs\"\n",
    "output_dir = \"Results/activation_segments/mustangs_realigned\"\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "print(f\"Output directory is set to: {output_dir}\")\n",
    "\n",
    "# Extract names of structures to exclude from 'bad'\n",
    "bad_names = [a.name for a in bad]  # Assuming 'bad' is a list of alignment objects\n",
    "print(f\"Excluding structures: {bad_names}\")\n",
    "\n",
    "# Function to calculate RMSD between two selections\n",
    "def calculate_rmsd(selection1, selection2):\n",
    "    print(f\"Calculating RMSD between {selection1} and {selection2}\")\n",
    "    return cmd.rms_cur(selection1, selection2)\n",
    "\n",
    "# Function to print details of a selection\n",
    "def print_selection_details(selection_name):\n",
    "    print(f\"Details for selection: {selection_name}\")\n",
    "    cmd.iterate(selection_name, \"print(resi, resn, name)\")\n",
    "    count = cmd.count_atoms(selection_name)\n",
    "    print(f\"Number of atoms in {selection_name}: {count}\")\n",
    "\n",
    "# Initialize variables to store the first structure's selections\n",
    "first_structure_name = None\n",
    "first_structure_D = None\n",
    "first_structure_E = None\n",
    "\n",
    "# Function to map sequence index to PDB residue number\n",
    "def map_seq_to_pdb(seq, target_index):\n",
    "    pdb_residue_number = 0\n",
    "    for i, char in enumerate(seq):\n",
    "        if char == '-':\n",
    "            pdb_residue_number += 1\n",
    "        if i == target_index:\n",
    "            break\n",
    "    return pdb_residue_number\n",
    "\n",
    "# Iterate over alignments in new_aligned2\n",
    "print(\"Starting alignment process...\")\n",
    "for idx, a in enumerate(tqdm(new_aligned2, desc=\"Aligning Structures\")):\n",
    "    print(f\"Processing structure: {a.name}\")\n",
    "    if a.name in bad_names:\n",
    "        print(f\"Skipping {a.name} as it is in the bad list.\")\n",
    "        continue\n",
    "    \n",
    "    # Load the structure\n",
    "    pdb_file = a.find_pdb()\n",
    "    print(f\"Loading PDB file: {pdb_file}\")\n",
    "    if not pdb_file:\n",
    "        print(f\"No PDB file found for {a.name}. Skipping...\")\n",
    "        continue\n",
    "    cmd.load(pdb_file, a.name)\n",
    "    print(f\"Loaded {a.name}\")\n",
    "    \n",
    "    # Debug: Print sequences to verify alignment\n",
    "    print(f\"Sequence for {a.name}:\")\n",
    "    print(f\"Seq1: {a.seq1}\")\n",
    "    print(f\"Seq2: {a.seq2}\")\n",
    "    \n",
    "    # Find indices for DFG and APE motifs\n",
    "    DFG_index = a.seq1.find(\"DFG\")\n",
    "    APE_index = a.seq1.find(\"APE\")\n",
    "    print(f\"Found DFG at index {DFG_index} and APE at index {APE_index}\")\n",
    "    \n",
    "    # Map sequence indices to PDB residue numbers\n",
    "    resi_D = map_seq_to_pdb(a.seq1, DFG_index+1)\n",
    "    resi_E = map_seq_to_pdb(a.seq1, APE_index + 3)  # +2 to get the \"E\" in \"APE\"\n",
    "    print(f\"Residue indices for {a.name}: resi_D={resi_D}, resi_E={resi_E}\")\n",
    "\n",
    "    # Select side chains of D in DFG and E in APE\n",
    "    if idx == 0:\n",
    "        # Store the selections of the first structure\n",
    "        first_structure_name = a.name\n",
    "        cmd.select(\"first_D\", f\"{a.name} and resi {resi_D} and name CA+CB\")\n",
    "        cmd.select(\"first_E\", f\"{a.name} and resi {resi_E} and name CA+CB\")\n",
    "        first_structure_D = \"first_D\"\n",
    "        first_structure_E = \"first_E\"\n",
    "        print(f\"Stored selections for the first structure: {first_structure_D}, {first_structure_E}\")\n",
    "    else:\n",
    "        # Create selections for the current structure\n",
    "        cmd.select(f\"{a.name}_D\", f\"{a.name} and resi {resi_D} and name CA+CB\")\n",
    "        cmd.select(f\"{a.name}_E\", f\"{a.name} and resi {resi_E} and name CA+CB\")\n",
    "        \n",
    "        # Print details of each selection\n",
    "        print_selection_details(f\"{a.name}_D\")\n",
    "        print_selection_details(f\"{a.name}_E\")\n",
    "        \n",
    "        # Debug: Check if selections are empty\n",
    "        d_count = cmd.count_atoms(f\"{a.name}_D\")\n",
    "        e_count = cmd.count_atoms(f\"{a.name}_E\")\n",
    "        if d_count == 0 or e_count == 0:\n",
    "            print(f\"Empty selection for {a.name}. D count: {d_count}, E count: {e_count}. Skipping...\")\n",
    "            cmd.delete(a.name)\n",
    "            continue\n",
    "        \n",
    "        # Calculate RMSD before alignment using the first structure's selections\n",
    "        if first_structure_D and first_structure_E:\n",
    "            rmsd_D_before = calculate_rmsd(first_structure_D, f\"{a.name}_D\")\n",
    "            print(f\"RMSD before alignment for {a.name}_D against first structure's D: {rmsd_D_before}\")\n",
    "            \n",
    "            rmsd_E_before = calculate_rmsd(first_structure_E, f\"{a.name}_E\")\n",
    "            print(f\"RMSD before alignment for {a.name}_E against first structure's E: {rmsd_E_before}\")\n",
    "            \n",
    "            # Align the side chains using fit\n",
    "            if d_count == e_count:\n",
    "                print(f\"Fitting {a.name}_D to first structure's D\")\n",
    "                cmd.fit(f\"{a.name}_D\", first_structure_D)\n",
    "            \n",
    "                print(f\"Fitting {a.name}_E to first structure's E\")\n",
    "                cmd.fit(f\"{a.name}_E\", first_structure_E)\n",
    "            \n",
    "            # Calculate RMSD after alignment using the first structure's selections\n",
    "            rmsd_D_after = calculate_rmsd(first_structure_D, f\"{a.name}_D\")\n",
    "            print(f\"RMSD after alignment for {a.name}_D against first structure's D: {rmsd_D_after}\")\n",
    "            \n",
    "            rmsd_E_after = calculate_rmsd(first_structure_E, f\"{a.name}_E\")\n",
    "            print(f\"RMSD after alignment for {a.name}_E against first structure's E: {rmsd_E_after}\")\n",
    "    \n",
    "    # Save the aligned structure\n",
    "    output_path = os.path.join(output_dir, f\"{a.name}_aligned.pdb\")\n",
    "    print(f\"Saving aligned structure to: {output_path}\")\n",
    "    cmd.save(output_path, a.name)\n",
    "\n",
    "# Clean up PyMOL session\n",
    "print(\"Closing PyMOL session.\")\n",
    "cmd.quit()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56ef154a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Code to save only first and last residue in activation segment for further visualization (mainly to double check if mustang alignment makes sense or not)\n",
    "\n",
    "import biobox as bb\n",
    "import glob\n",
    "import numpy as np\n",
    "\n",
    "files = f\"Results/activation_segments/CA_segments/{kind}/*.pdb\"\n",
    "\n",
    "start = []\n",
    "end = []\n",
    "for f in glob.glob(files):\n",
    "\n",
    "    M = bb.Molecule(f)\n",
    "    pts = M.atomselect(\"*\", \"*\", \"CA\")\n",
    "\n",
    "    start.append(pts[0])\n",
    "    end.append(pts[-1])\n",
    "\n",
    "#start = np.array(start)\n",
    "#end = np.array(end)\n",
    "\n",
    "M_start = bb.Structure(p=np.array(start))\n",
    "M_end = bb.Structure(p=np.array(end))\n",
    "\n",
    "M_start.write_pdb(\"start_CA.pdb\")\n",
    "M_end.write_pdb(\"end_CA.pdb\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "994a7765",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pymol import cmd\n",
    "from glob import glob\n",
    "import os\n",
    "from Bio.PDB import PDBParser, PPBuilder\n",
    "\n",
    "# Paths\n",
    "pdb_dir = \"Results/activation_segments/mustangs/\"\n",
    "reference_pdb = \"6UAN_chainD.pdb\"\n",
    "output_dir = \"Results/activation_segments/mustangs_realigned/\"\n",
    "image_output_path = \"Results/aligned_loops.png\"\n",
    "\n",
    "# Ensure output directory exists\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "\n",
    "# Load the reference structure\n",
    "cmd.load(reference_pdb, \"6UAN_chainD\")\n",
    "\n",
    "# Create selections for the reference structure\n",
    "cmd.select(\"6UAN_chainD_dfg_selection\", \"6UAN_chainD and resi 594-596 and name CA\")\n",
    "cmd.select(\"6UAN_chainD_ape_selection\", \"6UAN_chainD and resi 621-623 and name CA\")\n",
    "cmd.select(\"6UAN_chainD_ends_selection\", \"6UAN_chainD and (resi 594-596 or resi 621-623) and name CA\")\n",
    "\n",
    "def extract_atom_sequence(pdb_file):\n",
    "    \"\"\"Extract sequence from atomic coordinates for the first chain found in the PDB file.\"\"\"\n",
    "    parser = PDBParser(QUIET=True)\n",
    "    structure = parser.get_structure('PDB', pdb_file)\n",
    "    \n",
    "    for model in structure:\n",
    "        for chain in model:\n",
    "            ppb = PPBuilder()\n",
    "            sequence = ''\n",
    "            for pp in ppb.build_peptides(chain):\n",
    "                sequence += pp.get_sequence()\n",
    "            return str(sequence)\n",
    "    return None\n",
    "\n",
    "def find_motif_indices(seq, motif):\n",
    "    index = seq.find(motif)\n",
    "    if index == -1:\n",
    "        return None\n",
    "    return index, index + len(motif)\n",
    "\n",
    "def print_residues_in_selection(selection_name):\n",
    "    model = cmd.get_model(selection_name)\n",
    "    residues = set((atom.resi, atom.resn) for atom in model.atom)\n",
    "    print(f\"Residues in {selection_name}: {sorted(residues)}\")\n",
    "\n",
    "def process_structure(pdb_file, ref_name=\"6UAN_chainD\"):\n",
    "    pdb_code = os.path.basename(pdb_file).split('.')[0]\n",
    "    cmd.load(pdb_file, pdb_code)\n",
    "\n",
    "    # Extract sequence from atomic coordinates\n",
    "    seq = extract_atom_sequence(pdb_file)\n",
    "    if seq is None:\n",
    "        print(f\"Skipping {pdb_code} due to inability to extract sequence.\")\n",
    "        return\n",
    "    print(f\"Sequence for {pdb_code}: {seq}\")\n",
    "\n",
    "    # Find indices for DFG and APE\n",
    "    dfg_indices = find_motif_indices(seq, \"DFG\")\n",
    "    ape_indices = find_motif_indices(seq, \"APE\")\n",
    "\n",
    "    if not dfg_indices or not ape_indices:\n",
    "        print(f\"Skipping {pdb_code} due to missing motifs.\")\n",
    "        return\n",
    "\n",
    "    # Create selections using sequence indices\n",
    "    dfg_residues = list(range(dfg_indices[0] + 1, dfg_indices[1] + 1))\n",
    "    ape_residues = list(range(ape_indices[0] + 1, ape_indices[1] + 1))\n",
    "\n",
    "    # Create selections\n",
    "    cmd.select(f\"{pdb_code}_dfg_selection\", f\"{pdb_code} and resi {dfg_residues[0]}-{dfg_residues[-1]} and name CA\")\n",
    "    cmd.select(f\"{pdb_code}_ape_selection\", f\"{pdb_code} and resi {ape_residues[0]}-{ape_residues[-1]} and name CA\")\n",
    "    cmd.select(f\"{pdb_code}_ends_selection\", f\"{pdb_code} and (resi {dfg_residues[0]}-{dfg_residues[-1]} or resi {ape_residues[0]}-{ape_residues[-1]}) and name CA\")\n",
    "\n",
    "    # Print residues in selections\n",
    "    print_residues_in_selection(f\"{pdb_code}_dfg_selection\")\n",
    "    print_residues_in_selection(f\"{pdb_code}_ape_selection\")\n",
    "    print_residues_in_selection(f\"{pdb_code}_ends_selection\")\n",
    "\n",
    "    # RMSD before alignment\n",
    "    rms_dfg_before = cmd.rms_cur(f\"{pdb_code}_dfg_selection\", f\"{ref_name}_dfg_selection\", matchmaker=-1)\n",
    "    rms_ape_before = cmd.rms_cur(f\"{pdb_code}_ape_selection\", f\"{ref_name}_ape_selection\", matchmaker=-1)\n",
    "    rms_ends_before = cmd.rms_cur(f\"{pdb_code}_ends_selection\", f\"{ref_name}_ends_selection\", matchmaker=-1)\n",
    "\n",
    "    print(f\"Before alignment RMSD for {pdb_code}: DFG={rms_dfg_before}, APE={rms_ape_before}, ENDS={rms_ends_before}\")\n",
    "\n",
    "    # Aligning\n",
    "    cmd.align(f\"{pdb_code}_ends_selection\", f\"{ref_name}_ends_selection\", cycles=0, transform=1)\n",
    "\n",
    "    # Save aligned structure\n",
    "    aligned_pdb_path = os.path.join(output_dir, f\"{pdb_code}_aligned.pdb\")\n",
    "    cmd.save(aligned_pdb_path, pdb_code)\n",
    "    print(f\"Saved aligned structure to {aligned_pdb_path}\")\n",
    "\n",
    "    # RMSD after alignment\n",
    "    rms_dfg_after = cmd.rms_cur(f\"{pdb_code}_dfg_selection\", f\"{ref_name}_dfg_selection\", matchmaker=-1)\n",
    "    rms_ape_after = cmd.rms_cur(f\"{pdb_code}_ape_selection\", f\"{ref_name}_ape_selection\", matchmaker=-1)\n",
    "    rms_ends_after = cmd.rms_cur(f\"{pdb_code}_ends_selection\", f\"{ref_name}_ends_selection\", matchmaker=-1)\n",
    "\n",
    "    print(f\"After alignment RMSD for {pdb_code}: DFG={rms_dfg_after}, APE={rms_ape_after}, ENDS={rms_ends_after}\")\n",
    "\n",
    "# Load all PDB files and process them\n",
    "fps = glob(pdb_dir + \"/*/*.pdb\")\n",
    "for fp in fps:\n",
    "    process_structure(fp)\n",
    "\n",
    "# Visualize and color\n",
    "cmd.select(\"all_loops\", \"byres 6UAN_chainD_ends_selection\")\n",
    "cmd.color(\"red\", \"6UAN_chainD and 6UAN_chainD_ends_selection\")\n",
    "cmd.color(\"white\", \"not 6UAN_chainD and all_loops\")\n",
    "\n",
    "# Set visualization style\n",
    "cmd.show(\"cartoon\")\n",
    "cmd.hide(\"lines\")\n",
    "\n",
    "# Save the image\n",
    "cmd.png(image_output_path, width=1200, height=800, dpi=300, ray=1)\n",
    "print(f\"Image saved to {image_output_path}\")\n",
    "\n",
    "cmd.quit()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b4f99703",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Same as above, just saving structures that have a sequence aligned to DFG of reference\n",
    "#same functionality as below code just with more structures\n",
    "from glob import glob\n",
    "import mdtraj as md\n",
    "import os\n",
    "from tqdm import tqdm\n",
    "\n",
    "def strip_to_ca(pdb, start_residue, end_residue):\n",
    "    pdb = md.load(pdb)\n",
    "    print(f\"Loaded PDB: {pdb}\")\n",
    "    \n",
    "    # Extract CA atoms within the specified range\n",
    "    atoms_indices = [\n",
    "        atom.index for res in pdb.top._residues[start_residue:end_residue] \n",
    "        for atom in res.atoms if atom.name == \"CA\"\n",
    "    ]\n",
    "    print(f\"Atoms indices for CA: {atoms_indices}\")\n",
    "    \n",
    "    return pdb.atom_slice(atoms_indices)\n",
    "\n",
    "# Set the target directory for saving stripped PDBs\n",
    "kind = 'mustang_dfg_seqAligned'\n",
    "target_dir = f\"Results/activation_segments/CA_segments/{kind}/\"\n",
    "os.makedirs(target_dir, exist_ok=True)\n",
    "print(f\"Target directory: {target_dir}\")\n",
    "\n",
    "# Set the directory containing aligned PDBs\n",
    "pdb_dir = \"Results/activation_segments/mustangs_realigned/\"\n",
    "fps = glob(pdb_dir+\"/*.pdb\")\n",
    "print(f\"Found PDB files: {fps}\")\n",
    "\n",
    "# Process alignments\n",
    "for align_obj in tqdm(new_aligned2):  # Assuming new_aligned2 is a list of alignment objects\n",
    "    print(f\"Processing alignment: {align_obj.name}\")\n",
    "    match_found = False\n",
    "    for fp in fps:\n",
    "        print(fp)\n",
    "        if align_obj.name in fp and \"pdb\" in fp:\n",
    "            match_found = True\n",
    "            print(f\"Match found for {align_obj.name} in file: {fp}\")\n",
    "            break\n",
    "    \n",
    "    if not match_found:\n",
    "        \n",
    "        print(f\"No match found for {align_obj.name}, skipping.\")\n",
    "        continue\n",
    "    \n",
    "    # Check if the DFG and APE motifs are present in seq1\n",
    "    DFG_index = align_obj.seq1.find(\"DFG\")\n",
    "    APE_index = align_obj.seq1.find(\"APE\")\n",
    "    print(f\"DFG index: {DFG_index}, APE index: {APE_index}\")\n",
    "    \n",
    "    if DFG_index == -1 or APE_index == -1:\n",
    "        print(f\"Skipping file {fp} as it does not contain the DFG or APE motifs in seq1.\")\n",
    "        continue\n",
    "    \n",
    "    # Check if the DFG motif is aligned in seq2 (not interrupted by gaps)\n",
    "    if align_obj.seq2[DFG_index:DFG_index + 3] != \"DFG\":\n",
    "        print(f\"Skipping file {fp} as the DFG motif is not aligned properly in seq2.\")\n",
    "        continue\n",
    "    \n",
    "    \n",
    "    # Adjust the DFG and APE indices to account for gaps\n",
    "    DFG_index_adjusted = DFG_index - sum([1 for a in align_obj.seq2[:DFG_index] if a == \"-\"])\n",
    "    APE_index_adjusted = APE_index - sum([1 for a in align_obj.seq2[:APE_index] if a == \"-\"])\n",
    "    print(f\"Adjusted DFG index: {DFG_index_adjusted}, Adjusted APE index: {APE_index_adjusted}\")\n",
    "    \n",
    "    # Strip to CA atoms between DFG and APE\n",
    "    stripped = strip_to_ca(fp, DFG_index_adjusted, APE_index_adjusted + 3)  # +3 to include \"APE\"\n",
    "    \n",
    "    # Construct the file name and save the stripped pdb\n",
    "    new_name = os.path.join(target_dir, os.path.basename(fp))\n",
    "    print(f\"Saving stripped PDB to: {new_name}\")\n",
    "    stripped.save(new_name)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14d358bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#this is right and it works, but renumbers residues meaning that the ones that are missing are ignored\n",
    "import os\n",
    "from Bio import PDB\n",
    "import numpy as np\n",
    "from scipy.interpolate import interp1d\n",
    "import tempfile\n",
    "import mdtraj as md\n",
    "\n",
    "def fitting_code(fp_or_traj, save_path):\n",
    "    # Function to read PDB file or trajectory object and get model\n",
    "    def read_structure(input_data):\n",
    "        if isinstance(input_data, str):\n",
    "            # If input is a string, treat it as a file path\n",
    "            parser = PDB.PDBParser(QUIET=True)\n",
    "            structure = parser.get_structure('structure', input_data)\n",
    "        elif isinstance(input_data, md.Trajectory):\n",
    "            # If input is a trajectory, save to temp PDB and read\n",
    "            with tempfile.NamedTemporaryFile(suffix=\".pdb\", delete=False) as tmpfile:\n",
    "                input_data.save(tmpfile.name)\n",
    "                tmpfile.close()\n",
    "                parser = PDB.PDBParser(QUIET=True)\n",
    "                structure = parser.get_structure('structure', tmpfile.name)\n",
    "            os.unlink(tmpfile.name)\n",
    "        else:\n",
    "            raise ValueError(\"Unsupported input type. Provide a file path or md.Trajectory.\")\n",
    "        return structure[0]\n",
    "\n",
    "    # Read the template for CA atoms configuration\n",
    "    template_model = read_structure('template.pdb')\n",
    "    Nnew = len([atom for atom in template_model.get_atoms() if atom.get_id() == 'CA'])\n",
    "\n",
    "    # Read input PDB file or trajectory\n",
    "    my_model = read_structure(fp_or_traj)\n",
    "    atom_list = [atom for atom in my_model.get_atoms() if atom.get_id() == 'CA']\n",
    "\n",
    "    n = len(atom_list)\n",
    "    avg = np.array([[atom.coord[0], atom.coord[1], atom.coord[2]] for atom in atom_list])\n",
    "\n",
    "    dims = ['x', 'y', 'z']\n",
    "    fits = {}\n",
    "    # Fit cubic interpolation for each axis\n",
    "    for j, dim in enumerate(dims):\n",
    "        fits[dim] = interp1d(np.arange(n), avg[:, j], kind='cubic', fill_value='extrapolate')\n",
    "\n",
    "    # Interpolation scales\n",
    "    X = np.arange(0, n-1, 0.1)\n",
    "\n",
    "    # Get derivative\n",
    "    dYdX = {dim: np.gradient(fits[dim](X)) for dim in dims}\n",
    "\n",
    "    # Calculate path length\n",
    "    Y = np.sqrt(sum(np.square(dYdX[dim]) for dim in dims))\n",
    "    L = np.trapz(Y, X)\n",
    "\n",
    "    # Create an interpolated arc length\n",
    "    Li = np.linspace(0, L, Nnew)\n",
    "\n",
    "    # Calculate the arc length for each sampled point\n",
    "    flen = np.array([np.trapz(Y[:ibig], X[:ibig]) for ibig in range(1, len(X))])\n",
    "\n",
    "    pt = np.zeros(Nnew, dtype=int)\n",
    "    for i in range(Nnew):\n",
    "        pt[i] = np.argmin(np.abs(flen - Li[i]))\n",
    "\n",
    "    new_coords = np.array([[fits[dim](X[pt[i]]) for dim in dims] for i in range(Nnew)])\n",
    "\n",
    "    ca_index = 0\n",
    "    for atom in template_model.get_atoms():\n",
    "        if atom.get_id() == 'CA':\n",
    "            atom.set_coord(new_coords[ca_index])\n",
    "            ca_index += 1\n",
    "\n",
    "    # Now let's write the updated PDB to the file\n",
    "    try:\n",
    "        with open(save_path, \"w\") as file:\n",
    "            io = PDB.PDBIO()\n",
    "            io.set_structure(template_model)\n",
    "            io.save(file)\n",
    "        print(f'Successfully saved the structure to {save_path}')\n",
    "    except Exception as e:\n",
    "        print(f\"Error during file save: {e}\")\n",
    "\n",
    "# Define input and output directories\n",
    "input_dir = 'Results/activation_segments/CA_segments/mustang_dfg_seqAligned/'\n",
    "output_dir = 'Results/fitted_matlab_segments/DFGalignedTrials/'\n",
    "\n",
    "# Ensure output directory exists\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "\n",
    "# Process each PDB file in the input directory\n",
    "for file_name in os.listdir(input_dir):\n",
    "    if file_name.endswith('.pdb'):\n",
    "        input_file_path = os.path.join(input_dir, file_name)\n",
    "        output_file_path = os.path.join(output_dir, file_name)\n",
    "        fitting_code(input_file_path, output_file_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "99814eda",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pymol import cmd\n",
    "from glob import glob\n",
    "import os\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Function to print details of a selection\n",
    "def print_selection_details(selection_name):\n",
    "    print(f\"Details for selection: {selection_name}\")\n",
    "    cmd.iterate(selection_name, \"print(resi, resn, name)\")\n",
    "    count = cmd.count_atoms(selection_name)\n",
    "    print(f\"Number of atoms in {selection_name}: {count}\")\n",
    "\n",
    "# Paths\n",
    "reference_pdb = \"6UAN_chainD.pdb\"\n",
    "fitted_pdb_dir = 'Results/fitted_matlab_segments/DFGalignedTrials/'\n",
    "\n",
    "# Load the reference structure\n",
    "cmd.load(reference_pdb, \"6UAN_chainD\")\n",
    "\n",
    "# Create selections for the reference structure\n",
    "cmd.select(\"6UAN_chainD_first_atom\", \"6UAN_chainD and resi 594 and name CA\")\n",
    "cmd.select(\"6UAN_chainD_last_atom\", \"6UAN_chainD and resi 623 and name CA\")\n",
    "print_selection_details(\"6UAN_chainD_first_atom\")\n",
    "print_selection_details(\"6UAN_chainD_last_atom\")\n",
    "# Create selections for the reference structure\n",
    "cmd.select(\"6UAN_chainD_first3\", \"6UAN_chainD and resi 594-596 and name CA\")\n",
    "cmd.select(\"6UAN_chainD_last3\", \"6UAN_chainD and resi 621-623 and name CA\")\n",
    "print_selection_details(\"6UAN_chainD_first3\")\n",
    "print_selection_details(\"6UAN_chainD_last3\")\n",
    "\n",
    "# Lists to store RMSD values\n",
    "rmsd_first_atom = []\n",
    "rmsd_last_atom = []\n",
    "rmsd_first3_atoms = []\n",
    "rmsd_last3_atoms = []\n",
    "\n",
    "# Load the fitted loop structures into PyMOL\n",
    "fitted_pdb_files = glob(fitted_pdb_dir + '/*.pdb')\n",
    "total_structures = len(fitted_pdb_files)\n",
    "structures_with_large_rmsd_first_atom = []\n",
    "structures_with_large_rmsd_last_atom = []\n",
    "\n",
    "# Load the fitted loop structures into PyMOL\n",
    "fitted_pdb_files = glob(fitted_pdb_dir + '/*.pdb')\n",
    "for fitted_pdb_file in fitted_pdb_files:\n",
    "    print(f\"Processing file: {fitted_pdb_file}\")\n",
    "    pdb_code = os.path.basename(fitted_pdb_file).split('.')[0]\n",
    "    cmd.load(fitted_pdb_file, pdb_code)\n",
    "    # Select the first atom, last atom, first three atoms, and last three atoms of the fitted structure\n",
    "    cmd.select(f\"{pdb_code}_first_atom\", f\"{pdb_code} and resi 593 and name CA\")\n",
    "    cmd.select(f\"{pdb_code}_last_atom\", f\"{pdb_code} and resi 619 and name CA\")\n",
    "    cmd.select(f\"{pdb_code}_first3_atoms\", f\"{pdb_code} and resi 593-595\")\n",
    "    cmd.select(f\"{pdb_code}_last3_atoms\", f\"{pdb_code} and resi 617-619\")\n",
    "    # Print details of each selection\n",
    "    print_selection_details(f\"{pdb_code}_first_atom\")\n",
    "    print_selection_details(f\"{pdb_code}_last_atom\")\n",
    "    print_selection_details(f\"{pdb_code}_first3_atoms\")\n",
    "    print_selection_details(f\"{pdb_code}_last3_atoms\")\n",
    "\n",
    "    # Calculate RMSD for the first atom\n",
    "    try:\n",
    "        rms_first_atom = cmd.rms_cur(f\"{pdb_code}_first_atom\", \"6UAN_chainD_first_atom\", matchmaker=-1)\n",
    "        print(f\"RMSD for first atom of {pdb_code} compared to reference: {rms_first_atom}\")\n",
    "        rmsd_first_atom.append(rms_first_atom)\n",
    "        if rms_first_atom > 2.5:\n",
    "            structures_with_large_rmsd_first_atom.append(pdb_code)\n",
    "    except:\n",
    "        print(f\"Error calculating RMSD for first atom of {pdb_code}\")\n",
    "\n",
    "    # Calculate RMSD for the last atom\n",
    "    try:\n",
    "        rms_last_atom = cmd.rms_cur(f\"{pdb_code}_last_atom\", \"6UAN_chainD_last_atom\", matchmaker=-1)\n",
    "        print(f\"RMSD for last atom of {pdb_code} compared to reference: {rms_last_atom}\")\n",
    "        rmsd_last_atom.append(rms_last_atom)\n",
    "        if rms_last_atom > 2.5:\n",
    "            structures_with_large_rmsd_last_atom.append(pdb_code)\n",
    "    except:\n",
    "        print(f\"Error calculating RMSD for last atom of {pdb_code}\")\n",
    "\n",
    "    # Calculate RMSD for the first three atoms\n",
    "    try:\n",
    "        rms_first3_atoms = cmd.rms_cur(f\"{pdb_code}_first3_atoms\", \"6UAN_chainD_first3\", matchmaker=-1)\n",
    "        print(f\"RMSD for first three atoms of {pdb_code} compared to reference: {rms_first3_atoms}\")\n",
    "        rmsd_first3_atoms.append(rms_first3_atoms)\n",
    "    except:\n",
    "        print(f\"Error calculating RMSD for first three atoms of {pdb_code}\")\n",
    "\n",
    "    # Calculate RMSD for the last three atoms\n",
    "    try:\n",
    "        rms_last3_atoms = cmd.rms_cur(f\"{pdb_code}_last3_atoms\", \"6UAN_chainD_last3\", matchmaker=-1)\n",
    "        print(f\"RMSD for last three atoms of {pdb_code} compared to reference: {rms_last3_atoms}\")\n",
    "        rmsd_last3_atoms.append(rms_last3_atoms)\n",
    "    except:\n",
    "        print(f\"Error calculating RMSD for last three atoms of {pdb_code}\")\n",
    "\n",
    "# List structures with RMSD larger than 2.5  from the first atom of the reference structure\n",
    "print(\"Structures with RMSD larger than 2.5  from the first atom of the reference structure:\")\n",
    "print(structures_with_large_rmsd_first_atom)\n",
    "\n",
    "# List structures with RMSD larger than 2.5  from the last atom of the reference structure\n",
    "print(\"Structures with RMSD larger than 2.5  from the last atom of the reference structure:\")\n",
    "print(structures_with_large_rmsd_last_atom)\n",
    "\n",
    "# Calculate percentage and number of structures with RMSD larger than 2.5 \n",
    "num_large_rmsd_first_atom = len(structures_with_large_rmsd_first_atom)\n",
    "num_large_rmsd_last_atom = len(structures_with_large_rmsd_last_atom)\n",
    "percentage_large_rmsd_first_atom = (num_large_rmsd_first_atom / total_structures) * 100\n",
    "percentage_large_rmsd_last_atom = (num_large_rmsd_last_atom / total_structures) * 100\n",
    "\n",
    "print(f\"Number of structures with RMSD larger than 2.5  from the first atom: {num_large_rmsd_first_atom}\")\n",
    "print(f\"Percentage of structures with RMSD larger than 2.5  from the first atom: {percentage_large_rmsd_first_atom:.2f}%\")\n",
    "print(f\"Number of structures with RMSD larger than 2.5  from the last atom: {num_large_rmsd_last_atom}\")\n",
    "print(f\"Percentage of structures with RMSD larger than 2.5  from the last atom: {percentage_large_rmsd_last_atom:.2f}%\")\n",
    "\n",
    "# Create a violin plot for the RMSD values\n",
    "data = [rmsd_first_atom, rmsd_last_atom, rmsd_first3_atoms, rmsd_last3_atoms]\n",
    "labels = ['First Atom', 'Last Atom', 'First Three Atoms', 'Last Three Atoms']\n",
    "fig, ax = plt.subplots()\n",
    "ax.violinplot(data)\n",
    "ax.set_xticks(np.arange(1, len(labels) + 1))\n",
    "ax.set_xticklabels(labels)\n",
    "ax.set_ylabel('RMSD ()')\n",
    "plt.title('RMSD Comparison')\n",
    "plt.show()\n",
    "\n",
    "\n",
    "cmd.quit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6b5fc504-c470-43da-9076-5d83bac25afc",
   "metadata": {},
   "outputs": [],
   "source": [
    "from glob import glob\n",
    "import os\n",
    "# Start MATLAB\n",
    "#import matlab.engine\n",
    "from tqdm.notebook import tqdm\n",
    "TARGET_DIR = \"Results/fitted_matlab_segments/mustang\"\n",
    "os.makedirs(TARGET_DIR, exist_ok=True)\n",
    "def pdb_id(fp):\n",
    "    return fp.rsplit(\".\",1)[0].rsplit(\"/\",1)[1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6fa60c95",
   "metadata": {},
   "outputs": [],
   "source": [
    "PDB_PATH = \"Results/activation_segments/CA_segments/mustang/\"\n",
    "#xyz = md.load(\"/home/marmatt/Documents/projects/BRAF/results/activation_segments/CA_segments/mustang/1A9U_A.pdb\")\n",
    "files = glob(PDB_PATH+\"/*.pdb\")\n",
    "ids = [pdb_id(f) for f in files]\n",
    "print(ids)\n",
    "out = [TARGET_DIR + f\"/{i}.pdb\" for i in ids]\n",
    "#print(files[0],ids[0],out[0])\n",
    "\n",
    "for f, n in tqdm(zip(files,out),total=len(files)):\n",
    "    fitting_code(f,n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc66c173",
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib import pyplot as plt\n",
    "from mpl_toolkits import mplot3d\n",
    "%matplotlib inline\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "import mdtraj as md\n",
    "\n",
    "\n",
    "# Usage Example\n",
    "xyz = md.load(\"Results/activation_segments/CA_segments/mustang/1AD5_A.pdb\")\n",
    "\n",
    "DFG = xyz.top.to_fasta()[0].find(\"DFG\")\n",
    "APE = xyz.top.to_fasta()[0].find(\"APE\")+2\n",
    "atoms = sum([[atom.index for atom in res.atoms if atom.name == \"CA\"] for res in xyz.top._residues[:]],[])\n",
    "\n",
    "coords = xyz.xyz[0,atoms].T\n",
    "x = coords[0]\n",
    "y = coords[1]\n",
    "z = coords[2]\n",
    "new_coords = md.load(\"Results/fitted_matlab_segments/mustang/1AD5_A.pdb\")\n",
    "\n",
    "atoms = sum([[atom.index for atom in res.atoms if atom.name == \"CA\"] for res in new_coords.top._residues[:]],[])\n",
    "new_coords = new_coords.xyz[0,atoms].T\n",
    "xp = new_coords[0]\n",
    "yp = new_coords[1]\n",
    "zp = new_coords[2]\n",
    "\n",
    "fig = plt.figure(figsize=(10,10))\n",
    "ax = plt.axes(projection='3d')\n",
    "ax.plot3D(x,y,z, 'blue',marker=\"o\")\n",
    "ax.plot3D(xp, yp, zp, 'red')\n",
    "plt.tick_params(bottom=False, top=False, labelbottom=False)\n",
    "ax.set_xticks([])\n",
    "ax.set_yticks([])\n",
    "ax.set_zticks([])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
